
<div>
    <style scoped>
        /*@import url('//maxcdn.bootstrapcdn.com/bootstrap/3.2.0/css/bootstrap.min.css');*/
    </style>

    <div id="home">
        <title>European Variation Archive &lt; EMBL-EBI</title>
        <div class="row">
            <div class="col-md-4">
                <h3><span class="glyphicon glyphicon-tower"></span>&nbsp;Overview</h3>
                <div><p> The European Variation Archive is a database that accepts submission of, and provides access to, all types of genetic variation data from all species.</p></div>
                <div><p> All users are able to download ​ ​data from any​ ​study via our Study Bowser, or ​ ​i​nteractively ​ ​quer​y our ​ ​variant catalogue via our ​ ​EVA Browser. </p></div>
                <div><p> New features ​shall regularly​ be​ added by the EVA development team. Please send all comments and feedback to <a href="mailto:eva-helpdesk@ebi.ac.uk?subject=&amp; target="_top">eva-helpdesk@ebi.ac.uk</a>. </p></div>
                <!--<div><p> Subscribe to our Twitter <a target="_blank" href="https://twitter.com/EBIvariation"><i class="icon-twitter"></i>feed</a> to get updates. </p></div>-->
            </div>

            <div class="col-md-4">
                <h3><span class="glyphicon glyphicon-info-sign"></span>&nbsp; Information</h3>
                <div><p><a href="?Study Browser">Download Data :</a> All of our data is open-access and can be downloaded.</p></div>
                <div><p><a href="?Submit Data">Submit: </a>EVA welcomes direct submission of all types of genetic variation from all species.</p></div>

            </div>

            <div class="col-md-4">
                <h3><span class="glyphicon glyphicon-stats"></span>&nbsp; Statistics</h3>
                <div id="statistics"><eva-statistics height="200" width="200"></eva-statistics></div>
            </div>

        </div>

        <div class="row">
            <div class="col-md-4">
                <h3><span class="glyphicon glyphicon-globe" ></span>&nbsp; News</h3>
                <div id="twitter-widget"></div>
            </div>

            <div class="col-md-4">
                <h3><span class="glyphicon glyphicon-warning-sign" ></span>&nbsp; Note</h3>
                <div><p>This web application makes an intensive use of new web technologies and standards like HTML5,Web Components.</p></div>
                <div><p>Browsers that are fully supported for this site are: Chrome 22+, Firefox 24+, Safari 6+ and Opera 15+.</p></div>
                <div><p> Older browser like Chrome 21-, Firefox 23- or Internet Explorer 10 and 11 may give rise to errors. Internet Explorer 6,7,8 and 9 are not supported at all.</p></div>
            </div>
        </div>
    </div>
    <div  id="submit">
        <title>Submit  &lt; European Variation Archive &lt; EMBL-EBI</title>
        <h2>European Variation Archive submissions</h2>
        <div>
            <p>EVA follows the infrastructure of fellow EMBL-EBI resources European Nucleotide Archive (<a href="http://www.ebi.ac.uk/ena/" target="_blank">ENA</a>) and European Genome-phenome Archive (<a href="http://www.ebi.ac.uk/ega/" target="_blank">EGA</a>)  to accept, archive, and accession <a target="_blank" href="?VCF">VCF files</a>. Submissions consist of VCF file(s) and metadata that describe sample(s), experiment(s), and analysis that produced the variant and/or genotype call(s).</p>
            <p>​Data submitted to EVA is brokered to our collaborating database at <a href="http://www.ncbi.nlm.nih.gov/" target="_blank">NCBI</a>, <a href="http://www.ncbi.nlm.nih.gov/SNP/" target="_blank">dbSNP</a> . It is therefore  unnecessary to submit data to both resources. Please contact <a href="mailto:eva-helpdesk@ebi.ac.uk?subject=Sequence submissionto EVA&amp; target="_top">eva-helpdesk@ebi.ac.uk</a> if you would like any further information on this brokering process or collaboration.</p>
        </div>

        <h3>Key stages of EVA submissions</h3>

        <h4>Contact</h4>
        <div>
            <p><img src="img/Contact.png" alt="" width="48" height="48"> Contact the EVA Helpdesk via <a href="?submission-start">this webform</a> in order to provide details of your submission.</p>
        </div>

        <h4>Receive</h4>
        <div>
            <p>
                <img src="img/Receive.png" alt="" width="48" height="48"> &nbsp;Receive your submission pack, which will include:
                <div style="margin-left:70px;">
                <ul>
                    <li>Details for your submission uploads </li>
                    <li> <a href="?Templates" target="_blank">Templates</a> to capture your associated metadata </li>
                    <li>Key stages for your submission</li>
                </ul>
                </div>
            </p>
        </div>

        <h4>Submit</h4>
        <div>
            <p><img src="img/Upload.png" alt="" width="48" height="48"> Upload your data files to your private submission upload account or directly to the <a href="mailto:eva-helpdesk@ebi.ac.uk?subject=Sequence submissionto EVA&amp;body=**We will provide you with a submission package metadata templates guides for completing templates**" target="_top">eva-helpdesk@ebi.ac.uk</a>.</p>
        </div>

        <!--<h4>Document</h4>-->
        <!--<div>-->
            <!--<p><img src="img/Document.png" alt="" width="48" height="48"> Provide details of your study, samples, experiments, runs/analysis, policy and datasets </p>-->
        <!--</div>-->
    </div>
    <div id="about">
        <title>About  &lt; European Variation Archive &lt; EMBL-EBI</title>
        <h2>About EVA</h2>
        <div>
            <p>The European Variation Archive (EVA) accepts submission of, and provides access to, all types of genetic variations from any species, observed germline or somatic sources, ranging from SNVs to large structural variations.</p>
            <p>All of our data is open access via direct query through web and/or programmatic interfaces.</p>
            <p>The main submission format is a VCF file. VCF files should provide either genotypes from individual samples or aggregate summary information, such as allele frequencies, and be accompanyed by descriptive metadata. Examples of such include: study description, sample origins, experimental details, methodology for variant and/or genotype calls, etc. We strongly encourage the submission of as much metadata possible as this is information is extremely useful for downstream analysis.</p>
            <p>In addition to archiving VCF data, each submission is also standardized to permit variant accessioning, global quering, re-mapping across genome builds and calculation of aggregate summary information based on the individual level genotypes.</p>
        </div>
    </div>

    <div id="contact">
        <title>Contact  &lt; European Variation Archive &lt; EMBL-EBI</title>
        <h2>Contact Us</h2>
        <div>
            <p>If you have any questions related to the European Variation Archive resource, please <a href="mailto:eva-helpdesk@ebi.ac.uk">contact us</a>.</p>
            <p>Follow us on Twitter using <a href="https://twitter.com/EBIvariation" target="_blank">@EBIvariation</a></p>
        </div>
    </div>

    <div id="submissionTemplates">
        <title>Templates  &lt; European Variation Archive &lt; EMBL-EBI</title>
        <h2>Templates</h2>
        <div>
            <h3>EVA Submissions</h3>
            <p>Unless fully described in the header of the VCF file(s), EVA submissions are accompanied by an Excel spreadsheet and/or tab-delimited text files which describe the meta-data. We encourage our users to submit as much metadata as possible.</p>
            <p>Please download our submission template via this <a href="files/EVA_Submission_template.EVA_.V1.0.1_live.xlsx" target="_blank">link</a>.</p>
            <p>A completed template for a fictional submission can be downloaded <a href="files/EVA_Submission_template.EVA_.V1.0.1_live_mockup.xlsx" target="_blank">here</a>.</p>
        </div>
        <h3>EVA Clincal Submissions</h3>
        <p>EVA Clinical submissions are described via an Excel spreadsheet and/or tab-delimited text files. Our submissions templates are fully compliant with the data required for ClinVar allowing us to broker the submission of all EVA Clincal data to our collaborating NCBI resource.</p>
        <p>Please download our EVA Clinical submission template via this link.</p>
        <p>A completed EVA Clinical template for a fictional submission can be downloaded here.</p>
    </div>

    <div id="beacon">
        <title>Beacon  &lt; European Variation Archive &lt; EMBL-EBI</title>
        <h2>A Web Interface for Genomic Variant Queries - EBI</h2>
        <div class="row">
            <div class="col-md-12"><p><b>Example queries:</b></p>
                <div class="row">
                    <div class="col-md-12">
                        <div><p><span><a href="#"  onclick="loadEvaBeacon(this)" chrom="13" coordinate="32888799" allele="C" project="PRJEB6040">Chrom:13&nbsp;Coordinate:32888799 &nbsp;Allele:C&nbsp;Project:PRJEB6040 </a></span></p></div>
                        <div><p><span><a href="#"  onclick="loadEvaBeacon(this)" chrom="1" coordinate="46403" allele="INS" project="PRJEB4019">Chrom:1&nbsp;Coordinate:46403 &nbsp;Allele:INS&nbsp;Project:PRJEB4019</a></span></p></div>
                        <div><p> <span><a href="#"  onclick="loadEvaBeacon(this)" chrom="1" coordinate="1002921" allele="DEL" project="PRJEB4019">Chrom:1&nbsp;Coordinate:1002921 &nbsp;Allele:DEL&nbsp;Project:PRJEB4019</a></span></p><hr/></div>
                    </div>
                </div>
            </div>
        </div>
        <div class="row"><div class="col-md-12"><div id="beacon-element"><eva-beacon></eva-beacon></div></div></div>
        <script>
            function loadEvaBeacon(el) {
                var chrom = el.getAttribute('chrom');
                var coordinate = el.getAttribute('coordinate');
                var allele = el.getAttribute('allele');
                var project = el.getAttribute('project');
                var el = document.querySelector("eva-beacon");
                el.setAttribute('chrom', chrom);
                el.setAttribute('coordinate', coordinate);
                el.setAttribute('allele', allele);
                el.setAttribute('project', project);
            }
        </script>
    </div>
    <div id="clinical">
        <title>Clinical  &lt; European Variation Archive &lt; EMBL-EBI</title>
        <h2>European Variation Archive Clinical</h2>
        <div class="row">
            <div class="col-md-12">
                <p>EVA Clinical welcomes submission of clinically relevant variant data from clinical testing labs, locus-specific databases, expert panels and professional societies.
                   EVA Clinical works in collaboration with the <a href="http://www.ncbi.nlm.nih.gov/clinvar" target="_blank">ClinVar resource at NCBI</a> to provide the most comprehensive dataset of clinically relevant variants available world-wide.
                   Data submitted to EVA Clinical is brokered to ClinVar therefore it is unnecessary to submit to both resources.</p>
                <p>To submit data to EVA Clinical please complete this  <a href="files/EVA_Submission_template.EVA.V1.0.1_live_plus_Clinical_Variants.xlsx" target="_blank">template</a> and submit to <a href="mailto:eva-helpdesk@ebi.ac.uk" target="_top">eva-helpdesk@ebi.ac.uk</a>A completed template for a fictional study can be found <a href="files/EVA_Submission_template.EVA.V1.0.1_live_plus_Clinical_Variants.xlsx" target="_blank">here</a>.</p>
                <p>If you would like to discuss any aspect of EVA Clinical, or if anything is unclear, please contact us directly at <a href="mailto:eva-helpdesk@ebi.ac.uk" target="_top">eva-helpdesk@ebi.ac.uk</a>. </p>
            </div>
        </div>
    </div>

    <div id="vcf-format">
        <title>VCF Format  &lt; European Variation Archive &lt; EMBL-EBI</title>
        <h2>EVA VCF Format</h2>
        <p> EVA only accepts VCF files that conform to <a href="http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41" target="_blank">VCF version 4.1 and above.</a></p>
        <h3>VCF Specification</h3>
        <div>
            <p>All VCF files must be valid, i.e. they must pass validation using a VCF validator, such as vcftools. In addition, all VCF files must include a reference tag (##reference). Any tag that contains white space (e.g. spaces) must be enclosed in double quotes, e.g.</p>
            <p>##EVA_PipelineDescription="This is a testing pipepline!" OK</p>
            <p>##EVA_PipelineDescription=This is a testing pipepline! NOT OK</p>
        </div>
        <h3>EVA Specific headers</h3>
        <div>
            <p>An EVA submission automatically creates the required objects in ENA. A VCF file belongs to an Analysis, which in turn belongs to a Project. Currently each VCF file can only belong to one Analysis, and each Analysis can only contain one VCF file. The Analysis allows the VCF file to be linked to the Samples or SampleGroup (a group of samples).</p>
        </div>
        <h3>Samples</h3>
        <div>
            <p>If a Sample or SampleGroup already exists in ENA, then the VCF file can reference this object. This can be done in 3 ways:</p>
            <p>1) Using the ENA Sample accession in the header of the VCF file.</p>
            <p>2) Use the ##EVA_Sample information line to to link the sample ID in the VCF file to the Sample accession using the following specification:</p>
            <p>##EVA_Sample=&lt;ID=sample_id,ACCESSION=ena_accession&gt;</p>
            <p>3) Supply a sample mapping file(s) along with the VCF files, which links the VCF sample IDs to the Sample accession. The filename for the sample mapping file is given using the ##EVA_SampleMappingFile information line in the VCF file. Multiple files can be used, but each sample ID from the VCF must only occur once in all of the specified sample mapping files. The format to specify sample mapping files is:</p>
            <p>##EVA_SampleMappingFile=&lt;NAME=filename,MD5=md5sum&gt;</p>
            <p dir="ltr">Samples can be linked to external resources, e.g. CORIELL. This is done using the ##EVA_SampleLink tag. e.g.</p>
            <p dir="ltr">##EVA_SampleLink=&lt;ID=HG00097,DB=CORIELL,DB_ID=NA18501,LABEL="Link to Coriell"&gt;</p>
            <p dir="ltr">ID and DB are required (if the VCF ID is the same as the external resource ID). If the external resource ID is different to the VCF ID, this can be specified using the DB_ID sub-tag. LABEL is optional.</p>
            <p>The ##EVA_Scope tag defines the study type. This tag is optional, and if used must correspond to the number of samples listed in the VCF file. If there is 1 sample, the ##EVA_Scope tag is 'single-isolate', otherwise it is 'multi-isolate'.</p>
        </div>
        <h3>Analysis</h3>
        <div>
            <p dir="ltr">Each VCF file must belong to an Analysis object which describes the analysis used to produce the VCF file. Currently and Analysis object can only contain 1 VCF file.</p>
            <p dir="ltr">The following tags are used to describe the analysis:</p>
            <p dir="ltr">##reference--REQUIRED</p>
            <p dir="ltr">##EVA_AnalysisTitle--REQUIRED</p>
            <p dir="ltr">##EVA_PipelineDescription--REQUIRED</p>
            <p dir="ltr">##EVA_AnalysisCenter--OPTIONAL</p>
            <p dir="ltr">##EVA_AnalysisDate--OPTIONAL</p>
            <p dir="ltr">##EVA_CallingAlgorithm--OPTIONAL</p>
            <p dir="ltr">##EVA_Platform--OPTIONAL</p>
            <p dir="ltr">##EVA_ExperimentType--REQUIRED</p>
            <p dir="ltr">##EVA_RunAccession--OPTIONAL</p>
            <p dir="ltr">##EVA_AnalysisAccession--OPTIONAL</p>
            <p dir="ltr">##reference</p>
            <p dir="ltr">This is a 'highly recommended' tag according to VCFv4.1 specifications, and EVA require it for all VCF files. It can refer to either:</p>
            <p dir="ltr">1) An INSDC assembly name or accession,e.g.</p>
            <p dir="ltr">##reference=GRCh37</p>
            <p dir="ltr">or</p>
            <p dir="ltr">##reference=GCA_000001405.1</p>
            <p dir="ltr">2) or a location of the reference file, e.g.</p>
            <p dir="ltr">##reference=file:///seq/references/1000GenomesPilot-NCBI36.fasta</p>
            <p dir="ltr">or</p>
            <p dir="ltr">##reference=<a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.2bit">http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.2bit</a></p>
            <p dir="ltr">EVA also requires INSDC accessions or names for each of the chromosomes (or sequences) referenced in column 1 of the VCF. This should be of the form:</p>
            <p dir="ltr">##contig=&lt;ID=1,accession=<a href="http://www.ncbi.nlm.nih.gov/nuccore/CM000663.1">CM000663.1</a>,length=249,250,621,species=”Homo sapiens”,taxonomy=9606&gt;</p>
            <p dir="ltr">or</p>
            <p dir="ltr">##contig=&lt;ID=1,URL=http://www.ncbi.nlm.nih.gov/nuccore/224384768?report=fasta&gt;</p>
            <p dir="ltr">##EVA_AnalysisTitle</p>
            <p dir="ltr">This is the title displayed for the accession Analysis object, e.g.</p>
            <p dir="ltr">##EVA_AnalysisTitle="1000 Genomes Phase 1 Variation Calls Analysis"</p>
            <p dir="ltr">##EVA_PipelineDescription</p>
            <p dir="ltr">Description of analysis pipeline. Free text field, e.g.</p>
            <p dir="ltr">##EVA_PipelineDescription="Final release set from derived from data from 7 different centers. See PUBMED:23128226"</p>
            <p dir="ltr">##EVA_AnalysisCenter</p>
            <p dir="ltr">The center that performed the analysis</p>
            <p dir="ltr">##EVA_AnalysisCenter="European Bioinformatics Institute"</p>
            <p dir="ltr">##EVA_AnalysisDate</p>
            <p dir="ltr">The date the analysis was performed (in DD/MM/YYYY-hh:min:secs-offest, where offset and hh:min:secs are optional, hr:min:secs defaults to 12:00:00, and offset indicates the hour offset from UTC defaulting to 0), e.g. for an analysis performed at 9.30 am, 6 hours before UTC on 30th May 2011:</p>
            <p dir="ltr">##EVA_AnalysisDate="30/05/2011-09:30-06"</p>
            <p dir="ltr">##EVA_CallingAlgorithm</p>
            <p dir="ltr">Details of the calling algorithm, version, and parameters, e.g.</p>
            <p dir="ltr">##EVA_CallingAlgorithm="GATKv2.3, java -jar GenomeAnalysisTK.jar -T HaplotypeCaller -R reference/human_g1k_v37.fasta -I HG00096.bam -I HG00097.bam --dbsnp dbSNP.vcf -stand_call_conf &nbsp;-stand_emit_conf 10.0 -o output.raw.snps.indels.vcf"</p>
            <p dir="ltr">##EVA_Platform</p>
            <p dir="ltr">The platform used, e.g.</p>
            <p dir="ltr">##EVA_Platform="Illumina Genome Analyser IIe"</p>
            <p dir="ltr">##EVA_ExperimentType</p>
            <p dir="ltr">Tag to indicate the type of experiment, choose from:</p>
            <p dir="ltr">Whole genome, Exome, Array</p><p dir="ltr">e.g.</p>
            <p dir="ltr">##EVA_ExperimentType="Whole Genome"</p>
            <p dir="ltr">##EVA_RunAccession</p>
            <p dir="ltr">Allows the VCF and associated Analysis object to be linked to an existing run accession, e.g.</p>
            <p dir="ltr">##EVA_RunAccession=DRR000003</p>
        </div>
        <h3>Project</h3>
        <div>
            <p dir="ltr">A Project object is required to contain the Sample and Analysis objects that pertain to the submitted VCF files. The tags for Project are:</p>
            <p dir="ltr">##EVA_ProjectName--REQUIRED</p>
            <p dir="ltr">##EVA_ProjectDescription--REQUIRED</p>
            <p dir="ltr">##EVA_Material--REQUIRED</p>
            <p dir="ltr">##EVA_Selection--REQUIRED</p>
            <p dir="ltr">##EVA_Publication--OPTIONAL</p>
            <p dir="ltr">##EVA_RelatedProject--OPTIONAL</p>
            <p dir="ltr">##EVA_ProjectLink--OPTIONAL</p>
            <p dir="ltr">##EVA_ProjectName</p>
            <p dir="ltr">This is a unique name for the Project,e.g.</p>
            <p dir="ltr">##EVA_ProjectName="1000 Genomes Phase 1 Variation Calls"</p>
            <p dir="ltr">##EVA_ProjectDescription</p>
            <p dir="ltr">This is a short description describing the project,e.g.</p>
            <p dir="ltr">##EVA_ProjectDescription="Final release set of the Phase 1 1000 genomes variation data."</p>
            <p dir="ltr">##EVA_Material</p>
            <p dir="ltr">The material used, one of:</p>
            <p dir="ltr">DNA, genomic RNA, transcribed RNA, unknown, other. &nbsp;e.g.</p>
            <p dir="ltr">##EVA_Material=DNA</p>
            <p dir="ltr">##EVA_Selection</p>
            <p dir="ltr">One of the following:</p>
            <p dir="ltr">genome, partial genome, exome, epigenomics, PCR, clone ends, random, CDNA, tag, unknown, other. e.g.</p>
            <p dir="ltr">##EVA_Selection=genome</p>
            <p dir="ltr">##EVA_Publication</p>
            <p dir="ltr">Any associated publications with the Project, e.g.</p>
            <p dir="ltr">##EVA_Publication=&lt;DB=PUBMED,ID=23128226,LABEL="Main 1000 Genomes Publication"&gt;</p>
            <p dir="ltr">DB and ID are required, LABEL is optional.</p>
            <p dir="ltr">##EVA_RelatedProject</p>
            <p dir="ltr">Projects can have parent, peer, or child projects. This tag allows one to specify associated projects, e.g.</p>
            <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=PARENT,ID=28889&gt;</p>
            <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=PEER,ID=28890&gt;</p>
            <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=CHILD,ID=28891&gt;</p>
            <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=CHILD,ID=28892&gt;</p>
            <p dir="ltr">The RELATIVE and ID sub-tags are both required.</p>
            <p dir="ltr">##EVA_ProjectLink</p>
            <p dir="ltr">This tag allows the project to be linked to an external resource, e.g.</p>
            <p dir="ltr">##EVA_ProjectLink=&lt;URL=http://www.1000genomes.org,LABEL="1000 Genomes website"&gt;</p>
            <p dir="ltr">##EVA_ProjectLink=&lt;DB=DGVa,ID=estd199&gt;</p>
            <p dir="ltr">DB and ID can be used for INSDC databases.</p>
            <p dir="ltr">If DB is present, ID is required. If DB is absent, URL is required and ID must not be included. LABEL is OPTIONAL.</p>
        </div>
        <br /><h4>Below is a valid set of tags for a fictional submission:</h4>
        <div>
            <p dir="ltr">##fileformat=VCFv4.1</p>
            <p dir="ltr">##EVA_ProjectName="1000 Genomes Phase 1 Variation Calls"</p>
            <p dir="ltr">##EVA_ProjectTitle="Test 1"</p>
            <p dir="ltr">##EVA_ProjectDescription="Test 2"</p>
            <p dir="ltr">##EVA_Publication=&lt;DB=PUBMED,ID=23128226,LABEL="Main 1000 Genomes Publication"&gt;</p>
            <p dir="ltr">##EVA_Publication=&lt;DB=ePUB,ID=23128226,LABEL="Main 1000 Genomes Publication"&gt;</p>
            <p dir="ltr">##EVA_Scope=multi-isolate</p>
            <p dir="ltr">##EVA_Material=DNA</p>
            <p dir="ltr">##EVA_Selection=other</p>
            <p dir="ltr">##EVA_TaxID=9606</p>
            <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=PARENT,ID=28889&gt;</p>
            <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=PEER,ID=28890&gt;</p>
            <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=CHILD,ID=28891&gt;</p>
            <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=CHILD,ID=28892&gt; ##EVA_ProjectLink=&lt;URL=http://www.1000genomes.org/phase1-analysis-results-directory,LABEL="FTP directory of submitted VCF files"&gt;</p>
            <p dir="ltr">##EVA_ProjectLink=&lt;URL=http://www.1000genomes.org,LABEL="1000 Genomes website"&gt;</p>
            <p dir="ltr">##EVA_AnalysisTitle="1000 Genomes Phase 1 Variation Calls VCF Parse TEST"</p>
            <p dir="ltr">##EVA_FileDescription="Final release set from derived from data from 7 different centers. See PUBMED:23128226"</p>
            <p dir="ltr">##EVA_SampleMappingFile=&lt;Name=test_file1.txt,MD5=123456789123456789123456789000&gt; ##EVA_ChecksumFile=ch1_test_md5.txt ##EVA_PipelineDescription="This is a testing piepline!"</p>
            <p dir="ltr">##EVA_CallingAlgorithm="GATKv2.3, java -jar GenomeAnalysisTK.jar -T HaplotypeCaller -R reference/human_g1k_v37.fasta -I HG00096.bam -I HG00097.bam --dbsnp dbSNP.vcf -stand_call_conf [50.0] -stand_emit_conf 10.0 -o output.raw.snps.indels.vcf" ##EVA_Platform="Illumina GA11"</p>
            <p dir="ltr">##EVA_ExperimentType="Is this a CV"</p>
            <p dir="ltr">##EVA_RunAccession=&lt;ID=SRA00001,NAME=ARunName&gt;</p>
            <p dir="ltr">##EVA_AnalysisAccession=&lt;ID=ENAA00001,NAME=AAnalysisName&gt;</p>
            <p dir="ltr">##EVA_Sample=&lt;ID=HG00096,ACCESSION=ERSAM1&gt;</p>
            <p dir="ltr">##EVA_Sample=&lt;ID=HG00097,GENDER=Male,PHENOTYPE=Ill,SAMPLETYPE=DNA,SUBJECT=Someone,DISEASE_SITE=Pancreas,TAXID=9606,TITLE="1000 Genomes Sample Number 97",DESCRIPTION="This is a description of HG00097"&gt;</p>
            <p dir="ltr">##EVA_SampleLink=&lt;ID=HG00097,DB=CORIELL,DB_ID=HG00097,LABEL="This is a label"&gt;</p>
            <p dir="ltr">&nbsp;</p>
        </div>
        <div>
            <p dir="ltr">These guidlines are also avaialble in <a href="files/EVA_VCF_1.0.pdf" target="_blank">PDF format.</a> For any additional support, please contact us at <a href="mailto:eva-helpdesk@ebi.ac.uk">eva-helpdesk@ebi.ac.uk</a>.</p>
        </div>
    </div>

    <div id="studyBrowser">
        <title>Study Browser  &lt; European Variation Archive &lt; EMBL-EBI</title>
        <ul class="nav nav-pills studyBrowserTabs" id="studyBrowserTabs" >
            <li id="evaLi" class="active"><a href="#eva-browser-tab" role="tab" data-toggle="tab" id="eva" onclick="loadBrowser('eva')">Short Genetic Variations</a></li>
            <li id="dgvaLi"><a href="#dgva-browser-tab" role="tab" data-toggle="tab" id="dgva" onclick="loadBrowser('dgva')">Structural Variation</a></li>
        </ul>
        <!--<div class="btn-group">-->
            <!--<button type="button" class="btn btn-default eva-button" href="#eva-browser-tab" role="tab" data-toggle="tab" id="eva" onclick="loadBrowser('eva')">Single Nucleotide Variation</button>-->
            <!--<button type="button" class="btn btn-default eva-button" href="#dgva-browser-tab" role="tab" data-toggle="tab" id="dgva" onclick="loadBrowser('dgva')">Structural Variation</button>-->
        <!--</div>-->
        <!-- Tab panes -->
        <div class="tab-content">
            <div class="tab-pane active" id="eva-browser-tab"><br />
                <eva-study-browser></eva-study-browser>
            </div>
            <div class="tab-pane active" id="dgva-browser-tab"><br /></div>
        </div>

        <script>
            function loadBrowser(target) {
//                var el = document.querySelector("dgva-study-browser");
//                if(target == "eva") {
//                    el.setAttribute('display', 'false');
//                } else {
//                    el.setAttribute('display', 'false');
//                }
                var browserEl = target+'-study-browser';
                var tabEl = '#'+target+'-browser-tab';
                var el = document.querySelector(browserEl);
                var tab = document.querySelector(tabEl);
                tab.innerHTML = '';
//                if(el == null){
                    var studyBrowser = document.createElement(browserEl);
                    tab.appendChild(studyBrowser);
//                }
            }
        </script>
    </div>
</div>


