<!doctype html>
<!--
  ~
  ~ European Variation Archive (EVA) - Open-access database of all types of genetic
  ~ variation data from all species
  ~
  ~ Copyright 2014-2017 EMBL - European Bioinformatics Institute
  ~
  ~ Licensed under the Apache License, Version 2.0 (the "License");
  ~ you may not use this file except in compliance with the License.
  ~ You may obtain a copy of the License at
  ~
  ~    http://www.apache.org/licenses/LICENSE-2.0
  ~
  ~ Unless required by applicable law or agreed to in writing, software
  ~ distributed under the License is distributed on an "AS IS" BASIS,
  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  ~ See the License for the specific language governing permissions and
  ~ limitations under the License.
  ~
  -->

<html lang="en">
<head>
    <meta charset="utf-8">
    <title>The European Bioinformatics Institute &lt; EMBL-EBI</title>
    <meta name="description" content="EMBL-EBI">
    <meta name="keywords" content="bioinformatics, europe, institute, variation">
    <meta name="author" content="European Variation Archive">
    <meta name="HandheldFriendly" content="true" />
    <meta name="MobileOptimized" content="width" />
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <meta name="theme-color" content="#70BDBD"> <!-- Android Chrome mobile browser tab color -->

    <!-- If you link to any other sites frequently, consider optimising performance with a DNS prefetch -->
    <link rel="dns-prefetch" href="//embl.de" />
    <link rel="dns-prefetch" href="//www.ncbi.nlm.nih.gov" />

    <!-- If you have custom icon, replace these as appropriate.
         You can generate them at realfavicongenerator.net -->
    <link rel="icon" type="image/x-icon" href="../lib/EBI-Framework/images/logos/EMBL-EBI/favicons/favicon.ico" />
    <link rel="icon" type="image/png" href="../lib/EBI-Framework/images/logos/EMBL-EBI/favicons/favicon-32x32.png" />
    <link rel="icon" type="image/png" sizes="192Ã—192" href="../lib/EBI-Framework/images/logos/EMBL-EBI/favicons/android-chrome-192x192.png" /> <!-- Android (192px) -->
    <link rel="apple-touch-icon-precomposed" sizes="114x114" href="../lib/EBI-Framework/images/logos/EMBL-EBI/favicons/apple-touch-icon-114x114.png"> <!-- For iPhone 4 Retina display (114px) -->
    <link rel="apple-touch-icon-precomposed" sizes="72x72" href="../lib/EBI-Framework/images/logos/EMBL-EBI/favicons/apple-touch-icon-72x72.png"> <!-- For iPad (72px) -->
    <link rel="apple-touch-icon-precomposed" sizes="144x144" href="../lib/EBI-Framework/images/logos/EMBL-EBI/favicons/apple-touch-icon-144x144.png"> <!-- For iPad retinat (144px) -->
    <link rel="apple-touch-icon-precomposed" href="../lib/EBI-Framework/images/logos/EMBL-EBI/favicons/apple-touch-icon-57x57.png"> <!-- For iPhone (57px) -->
    <link rel="mask-icon" href="../lib/EBI-Framework/images/logos/EMBL-EBI/favicons/safari-pinned-tab.svg" color="#ffffff"> <!-- Safari icon for pinned tab -->
    <meta name="msapplication-TileColor" content="#2b5797"> <!-- MS Icons -->
    <meta name="msapplication-TileImage" content="../lib/EBI-Framework/images/logos/EMBL-EBI/favicons/mstile-144x144.png">
    <!-- CSS: implied media=all -->
    <!-- CSS concatenated and minified via ant build script-->
    <!-- build:style ebi_framework -->
    <link rel="stylesheet" href="../lib/EBI-Framework/libraries/foundation-6/css/foundation.min.css" type="text/css" media="all">
    <link rel="stylesheet" href="../lib/EBI-Framework/css/ebi-global.css" type="text/css" media="all">
    <link rel="stylesheet" href="../lib/EBI-Framework/css/theme-embl-petrol.css" type="text/css" media="all">
    <!-- /build -->
    <link rel="stylesheet" href="https://www.ebi.ac.uk/web_guidelines/EBI-Icon-fonts/v1.1/fonts.css" type="text/css" media="all">
    <!-- build:style vendor -->
    <link type="text/css" rel="stylesheet" href="../vendor/ext-6.0.1/theme/theme-ebi-embl-all.css"/>
    <!-- /build -->
    <!-- build:style css -->
    <link rel="stylesheet" href="css/eva.css">
    <!-- /build -->
    <meta name="ebi:localmasthead-color" content="#0C6666">
    <meta name="ebi:localmasthead-image" content="img/EVA_homepage_banner_opt.jpg">
    <!-- end CSS-->
    <!-- All JavaScript at the bottom, except for Modernizr -->
    <!-- build:script modernizr -->
    <script src="../lib/EBI-Framework/libraries/modernizr/modernizr.custom.49274.js"></script>
    <!-- /build -->
</head>

<body class="level2 full-width">
<div id="skip-to">
    <ul>
        <li><a href="#content">Skip to main content</a></li>
        <li><a href="#local-nav">Skip to local navigation</a></li>
        <li><a href="#global-nav">Skip to EBI global navigation menu</a></li>
        <li><a href="#global-nav-expanded">Skip to expanded EBI global navigation menu (includes all sub-sections)</a></li>
    </ul>
</div>

<div data-sticky-container>
    <div id="local-masthead" data-sticky data-sticky-on="large" data-top-anchor="180" data-btm-anchor="300000">
        <header>
            <div id="global-masthead" class="clearfix">
                <a href="//www.ebi.ac.uk/" title="Go to the EMBL-EBI homepage"><span class="ebi-logo"></span></a>
                <nav>
                    <div class="row">
                        <ul id="global-nav" class="menu">
                            <!-- set active class as appropriate -->
                            <li id="home-mobile" class=""><a href="//www.ebi.ac.uk"></a></li>
                            <li id="home" class="active"><a href="//www.ebi.ac.uk"><i class="icon icon-generic" data-icon="H"></i> EMBL-EBI</a></li>
                            <li id="services"><a href="//www.ebi.ac.uk/services"><i class="icon icon-generic" data-icon="("></i> Services</a></li>
                            <li id="research"><a href="//www.ebi.ac.uk/research"><i class="icon icon-generic" data-icon=")"></i> Research</a></li>
                            <li id="training"><a href="//www.ebi.ac.uk/training"><i class="icon icon-generic" data-icon="t"></i> Training</a></li>
                            <li id="about"><a href="//www.ebi.ac.uk/about"><i class="icon icon-generic" data-icon="i"></i> About us</a></li>
                            <li id="search">
                                <a data-toggle="search-global-dropdown"><i class="icon icon-functional" data-icon="1"></i> <span class="show-for-small-only">Search</span></a>
                                <div id="search-global-dropdown" class="dropdown-pane" data-dropdown data-options="closeOnClick:true;">
                                    <form id="global-search" name="global-search" action="/ebisearch/search.ebi" method="GET">
                                        <fieldset>
                                            <div class="input-group">
                                                <input type="text" name="query" id="global-searchbox" class="input-group-field" placeholder="Search all of EMBL-EBI">
                                                <div class="input-group-button">
                                                    <input type="submit" name="submit" value="Search" class="button">
                                                    <input type="hidden" name="db" value="allebi" checked="checked">
                                                    <input type="hidden" name="requestFrom" value="global-masthead" checked="checked">
                                                </div>
                                            </div>
                                        </fieldset>
                                    </form>
                                </div>
                            </li>
                            <li class="float-right show-for-medium embl-selector">
                                <button class="button float-right" type="button" data-toggle="embl-dropdown">Hinxton</button>
                                <!-- The dropdown menu will be programatically added by script.js -->
                            </li>
                        </ul>
                    </div>
                </nav>
            </div>

            <div class="masthead row">
                <!-- local-title -->
                <div class="columns medium-12 " id="local-title">
                    <h1><img src="img/eva_logo.png" style="margin-left:-10px;"/><a href="?Home" title="Back to European Variation Archive homepage homepage">European Variation Archive</a></h1>
                </div>
                <!-- /local-title -->
                <!-- local-nav -->
                <nav>
                    <ul class="dropdown menu float-left eva-primary-menu" data-dropdown-menu data-description="navigational">
                        <li><a>Home</a></li>
                        <li><a>Submit Data</a></li>
                        <li><a>Study Browser</a></li>
                        <li><a>Variant Browser</a></li>
                        <li><a>GA4GH</a></li>
                        <li><a>API</a></li>
                        <li><a>RS Release</a></li>
                        <li><a>Help</a></li>
                        <li><a>Feedback</a></li>
                    </ul>
                </nav>
                <!-- /local-nav -->
            </div>
        </header>
    </div>
</div>


<div id="content" role="main" class="row">
    <section>
        <div id="main-content-area" class="columns">
            <div id="home-page" class="hide-div">
                <nav aria-label="You are here:" role="navigation">
                    <ul class="breadcrumbs">
                        <li><a href="?Home">EVA</a></li>
                        <li>
                            <span class="show-for-sr">Current: </span> Home
                        </li>
                    </ul>
                </nav>
                <div class="row overview-main">
                    <div class="small-12 medium-8 columns">
                        <div>
                            <h3><span class="icon icon-generic" data-icon="&"> </span>Overview</h3>
                            <div><p> The European Variation Archive is an open-access database of all types of genetic
                                variation data from all species.</p></div>
                            <div><p> All users can <a href="?Study Browser "> download data </a> from any study, or <a
                                    href="?Submit Data"> submit their own data</a> to the archive. You can also query
                                all variants in the EVA by study, gene, chromosomal location or dbSNP identifier using
                                our <a href="?Variant Browser"> Variant Browser</a>. </p></div>
                            <div><p> We will be adding new features to the EVA on a regular basis, and welcome <a
                                    href="mailto:eva-helpdesk@ebi.ac.uk" target="_top">your comments and feedback</a>.
                            </p></div>
                        </div>
                        <br/><br/>
                        <div class="row">
                            <div class="search-snps">
                                <h3><span class="icon icon-functional" data-icon="1"> </span>Search for SNPs</h3>
                                <form id="snpSearch" name="local-search" action="./variant" method="GET">
                                    <fieldset>
                                        <div>
                                            <div class="small-10 medium-10 columns">
                                                <input type="text" name="accessionID" id="accession-search-box"
                                                       placeholder="ex: rs123 or ss567 or comma/space separated RS or SS IDs">
                                            </div>

                                            <div class="small-2 medium-2 columns">
                                                <div class="input-group-button">
                                                    <input class="button" name="accessionID"
                                                           id="accession-search-button" value="Search" type="submit">
                                                </div>
                                            </div>
                                        </div>
                                    </fieldset>
                                </form>
                                <p>The RS ID release <span id="release-version"></span> is available in our
                                    <a id="release-ftp-link" target="_blank">FTP</a> or through our
                                    <a href="https://www.ebi.ac.uk/eva/webservices/identifiers/swagger-ui.html"
                                       target="_blank">API</a>
                                    . See <a href="?RS-Release" target="_blank">release page</a> for details.
                                </p>
                            </div>
                        </div>
                    </div>

                    <div class="small-12 medium-4 columns">
                        <h3><span class="icon icon-generic" data-icon="N"> </span>News</h3>
                        <div id="twitter-widget"></div>
                    </div>
                </div>
                <br/>

                    <div class="padding-bottom-large"></div>
                    <div class="row">
                        <div class="columns"><h3><span class="icon icon-generic" data-icon="g"> </span>Statistics</h3></div>
                        <div class="small-12 medium-6 columns">
                            <div class="row">
                                <div class="small-10 medium-10 medium-offset-2">
                                    <h4>Short genetic variants studies (&lt;50bp)</h4>
                                </div>
                            </div>
                            <div id="eva-statistics"></div>
                        </div>
                        <div class="small-12 medium-6 columns">
                            <div class="row">
                                <div class="small-10 medium-10 medium-offset-2">
                                    <h4 class="padding-bottom">Structural variants studies (&gt;50bp)</h4>
                                </div>
                             </div>
                            <div id="dgva-statistics"></div>
                        </div>
                    </div>

                    <div class="row">
                        <div class="small-12 medium-12 columns">
                            <div><p style="font-size:11px;">This web application makes an intensive use of new web technologies and standards like HTML5. Please see <a href="?Help#dataAccessPanel&link=supported-browsers" target="_blank">FAQs</a> for further browser compatibility notes.</p></div>
                        </div>
                    </div>
                </div>

                <div  id="submit" class="hide-div">
                    <nav aria-label="You are here:" role="navigation">
                        <ul class="breadcrumbs">
                            <li><a href="?Home">EVA</a></li>
                            <li>
                                <span class="show-for-sr">Current: </span> Submit
                            </li>
                        </ul>
                    </nav>
                    <h2>Submit</h2>
                    <div class="padding-bottom-large">
                        <p>The EVA is based entirely on direct user submissions. If you are interested in submitting your data, please read our <a href="?Submit-Data#data-requirements" target="_blank">Data Requirements</a> and the <a href="?Submit-Data#key-stages" target="_blank">Key stages of submission</a> below.</p>
                    </div>

                    <h3 id="data-requirements" class="padding-bottom-large">Data requirements</h3>
                    <div class="padding-bottom-large">
                        <p>EVA accepts <b>all</b> types of <b>precise</b> genetic variants, in <b>any species</b>, provided the following requirements are met:</p>
                        <ol>
                            <li>Data is described in <b>valid VCF file(s)</b>. This can be tested prior to submission using the <a href="https://github.com/EBIvariation/eva-sub-cli" target="_blank">eva-sub-cli</a> tool. For help with converting variation data to VCF, please see our <a href="?Help#submissionPanel&link=how-to-gen-vcf" target="_blank">help</a> pages.</li>
                            <li>Data includes <b>sample genotypes</b> in the FORMAT fields and/or <b>allele frequencies</b> in the INFO field</li>
                            <li>The reference sequence used is <a href="http://www.insdc.org/" target="_blank">INSDC</a> registered, or will be at point of submission. The type of sequence should be an assembly, e.g. <a href="https://www.ebi.ac.uk/ena/browser/view/GCA_000002285.4" target="_blank">GCA_000002285.4</a></li>
                        </ol>
                        <p>More details on whether your data is suitable can be found <a href="?Help#submissionPanel&link=is-my-data-suitable" target="_blank">here</a>.</p>
                        <p><b><u>PLEASE NOTE:</u></b> We support multiple sequence identifiers (CHROM column in VCF), such as INSDC or Refseq identifiers and ENA and Genbank chromosome names. For more details, please see our <a href="?Help#submissionPanel&link=how-do-we-deal-with-chromosome-name-diversity" target="_blank">help</a> pages.</p>
                    </div>

                    <h3 class="padding-bottom-large">Alternative resources for data not accepted by EVA</h3>
                    <div>
                        <ul>
                            <li>Submit variations with sensitive clinical data to <a href="https://www.ebi.ac.uk/ega/" target="_blank">EGA</a>.</li>
                            <li>Submit variations with clinically relevant genetic variant data, i.e. data that relates genetic variation(s) with clinical significance values (e.g. pathogenic, benign, etc.), to the <a href="https://www.ncbi.nlm.nih.gov/clinvar/" target="_blank">ClinVar</a> archive at <a href="https://www.ncbi.nlm.nih.gov/clinvar/" target="_blank">NCBI</a>.</li>
                        </ul>
                    </div>

                    <h3 id="key-stages" class="padding-bottom-large">Key stages of EVA submissions</h3>

                    <div class="submit-data-diagram">
                        <div class="box green">
                            <h3><span class="icon icon-generic" data-icon="'" aria-hidden="true"></span> Prepare your data and metadata</h3>
                            <ul>
                                <li>Prepare valid VCF file(s), which can be validated prior to submission using the <a href="https://github.com/EBIvariation/eva-sub-cli" target="_blank">eva-sub-cli tool</a>.</li>
                                <li>Complete a <a href="https://raw.githubusercontent.com/EBIvariation/eva-sub-cli/main/eva_sub_cli/etc/EVA_Submission_template.xlsx" target="_blank">metadata template</a> describing the samples and analyses in your study. This can be validated alongside your VCF files with <a href="https://github.com/EBIvariation/eva-sub-cli" target="_blank">eva-sub-cli</a>. Please provide as much metadata as possible since this information is useful for downstream analysis and leads to increased dataset reuse. For reference, here is an <a href="https://raw.githubusercontent.com/EBIvariation/eva-sub-cli/main/eva_sub_cli/etc/EVA_Submission_Example.xlsx" target="_blank">example</a> of a completed metadata template.</li>
                                <li>Please also note that the template requires the submitter to fill in some personal data, which will be used as described in our <a href="https://www.ebi.ac.uk/data-protection/privacy-notice/embl-ebi-public-website/" target="_blank">privacy notice</a>.</li>
                                <li>If your submission contains human genotype data, then please provide a signed copy of our <a href="https://docs.google.com/document/d/1UaRmimAe919IZYIH55mDXMe-4sHsRVqaP4moqx2IYE4" target="_blank">Consent Statement</a> via email to <a href="mailto:eva-helpdesk@ebi.ac.uk">eva-helpdesk@ebi.ac.uk</a>. See our <a href="?Help#submissionPanel&link=consent-statement-for-human-genotype-data" target="_blank">help</a> section for more info.</li>
                            </ul>
                        </div>
                        <div class="arrow"></div>
                        <div class="row">
                            <div class="box blue">
                                <h3><span class="icon icon-functional" data-icon="_" aria-hidden="true"></span> Submit via eva-sub-cli</h3>
                                <p>This is our command-line tool for validating and uploading your data and metadata. For instructions on how to install and use <a href="https://github.com/EBIvariation/eva-sub-cli" target="_blank">eva-sub-cli</a>, please see our <a href="https://github.com/EBIvariation/eva-sub-cli/blob/main/README.md" target="_blank">documentation</a>.</p>
                                <p>This method has faster turnaround times, but may be more suitable for more technical users.</p>
                                <p>Before submitting data through eva-sub-cli, please make sure you have a <a href="https://www.ebi.ac.uk/ena/submit/webin/login" target="_blank">Webin</a> Account.</p>
                            </div>
                            <div class="or-label"><h3>OR</h3></div>
                            <div class="box blue">
                                <h3><span class="icon icon-functional" data-icon="n" aria-hidden="true"></span> Submit via email</h3>
                                <p>Contact <a href="mailto:eva-helpdesk@ebi.ac.uk" target="_top">eva-helpdesk@ebi.ac.uk</a> to request a submission. You will receive a custom private FTP account to deposit your data and metadata files, as well as further instructions.</p>
                                <p>This submission method is less technical but may involve longer processing times as we help you validate your data.</p>
                                <p>Please make sure to read the instructions given in the metadata spreadsheet before submitting.</p>
                            </div>
                        </div>
                        <div class="arrow"></div>
                        <div class="box yellow">
                            <h3><span class="icon icon-functional" data-icon="k" aria-hidden="true"></span> Access your data</h3>
                            <p>The EVA aims to process submission requests within two business days. Accession numbers will be sent via email to the submitter upon successful archival of the deposited data.</p>
                            <p>All data that is valid for EVA submission shall be made available via the <a href="?Study Browser" target="_blank">Study Browser</a> and will be browsable using both the <a href="?Variant Browser" target="_blank">Variant Browser</a> and the <a href="?API" target="_blank">EVA API</a>. Our periodic <a href="?RS-Release" target="_blank">RS Releases</a> will also incorporate the submitted variants.</p>
                        </div>
                    </div>

                </div>

                <div id="feedback" class="hide-div">
                    <nav aria-label="You are here:" role="navigation">
                        <ul class="breadcrumbs">
                            <li><a href="?Home">EVA</a></li>
                            <li>
                                <span class="show-for-sr">Current: </span> Feedback
                            </li>
                        </ul>
                    </nav>
                    <h2>Feedback</h2>
                    <div>
                        <p>If you have any questions related to the European Variation Archive resource, please <a href="mailto:eva-helpdesk@ebi.ac.uk">contact us</a>.</p>
                        <p>Follow us on Twitter using <a href="https://twitter.com/evarchive" target="_blank">@evarchive</a></p>
                    </div>
                    <div id="feedback-form">
                        <div class="row">
                            <div class="medium-12 columns feedback-form">
                                <iframe src="https://www.surveymonkey.co.uk/r/eva-feedback" frameborder="0" width="100%"  height="1490"></iframe>
                            </div>
                        </div>
                    </div>
                </div>
                <div id="api" class="hide-div">
                    <nav aria-label="You are here:" role="navigation">
                        <ul class="breadcrumbs">
                            <li><a href="?Home">EVA</a></li>
                            <li>
                                <span class="show-for-sr">Current: </span> API
                            </li>
                        </ul>
                    </nav>
                    <h2>API</h2>
                    <div>
                        <h3>The general structure of a EVA REST web service URL is one of:</h3>
                        <ul>
                            <li><p><a href="https://www.ebi.ac.uk/eva/webservices/rest/swagger-ui.html">Queries to retrieve variant information (locus, genotypes, evidence etc.,) - http://www.ebi.ac.uk/eva/webservices/rest/{version}/{category}/IDs/{resource}?{filters}</a></p></li>
                            <li><p><a href="https://www.ebi.ac.uk/eva/webservices/identifiers/swagger-ui.html">Queries to retrieve locus information by accession ID - http://www.ebi.ac.uk/eva/webservices/identifiers/{version}/{category}/ID[/submitted]</a></p></li>
                        </ul>
                        <h3>Where:</h3>
                        <p>
                            * <span class="api-span">version</span>: indicates the version of the API, this defines the available filters and JSON schema to be returned. Currently there is only version 'v1'.
                            <br/> * <span class="api-span">category</span>: this defines what objects we want to query. Currently there are five different categories for variant information queries: variants, segments, genes, files and studies and two categories for accessioning queries: submitted-variants (to query by SS ID) and clustered-variants (to query by RS IDs)
                            <br/> * <span class="api-span">resource</span>: specifies the resource to be returned, therefore the JSON data model.
                            <br/> * <span class="api-span">filters</span>: each specific endpoint allows different filters.
                        </p>
                        <div>
                            <p>REST web services have been implemented using GET protocol since only queries are allowed so far. Several IDs can be concatenated using comma as separator.<br/>For more detailed information about the API and filters you can visit the <a href="https://github.com/ebivariation/eva/wiki" target="_blank">project wiki</a> and Swagger documentation for <a href="https://www.ebi.ac.uk/eva/webservices/rest/swagger-ui.html" target="_blank">variant information queries</a> and <a href="https://www.ebi.ac.uk/eva/webservices/identifiers/swagger-ui.html" target="_blank">accessioning locus queries</a>.</p>
                        </div>
                        <div>
                            <h3>Some example of queries include:</h3>
                            <p>* To fetch all the variants in a segment region:
                                <br/> <a href="http://www.ebi.ac.uk/eva/webservices/rest/v1/segments/11:128446-128446/variants?species=hsapiens_grch37" target="_blank">http://www.ebi.ac.uk/eva/webservices/rest/v1/segments/11:128446-128446/variants?species=hsapiens_grch37</a></p>

                            <p><br/> * To fetch all the info for a variant by ID:
                                <br/> <a href="http://www.ebi.ac.uk/eva/webservices/rest/v1/variants/rs666/info?species=hsapiens_grch37" target="_blank">http://www.ebi.ac.uk/eva/webservices/rest/v1/variants/rs666/info?species=hsapiens_grch37</a></p>

                            <p><br/> * To fetch locus info for a variant by <a href="https://www.ncbi.nlm.nih.gov/books/NBK44374/#Content.what_is_the_ncbi_assay_id_or__ss" target="_blank">SubSNP ID (SS ID)</a>:
                                <br/> <a href="https://www.ebi.ac.uk/eva/webservices/identifiers/v1/submitted-variants/99308221" target="_blank">https://www.ebi.ac.uk/eva/webservices/identifiers/v1/submitted-variants/99308221</a></p>

                            <p><br/> * To fetch locus and type info for a variant by <a href="https://www.ncbi.nlm.nih.gov/books/NBK44417/#Content.what_is_a_reference_snp_or__rs_i" target="_blank">RefSNP ID (RS ID)</a>:
                                <br/> <a href="https://www.ebi.ac.uk/eva/webservices/identifiers/v1/clustered-variants/17870277" target="_blank">https://www.ebi.ac.uk/eva/webservices/identifiers/v1/clustered-variants/17870277</a></p>

                            <p><br/> * To fetch associated SubSNP IDs (SS IDs) for a variant by RefSNP IDs (RS IDs):
                                <br/> <a href="https://www.ebi.ac.uk/eva/webservices/identifiers/v1/clustered-variants/17870277/submitted" target="_blank">https://www.ebi.ac.uk/eva/webservices/identifiers/v1/clustered-variants/17870277/submitted</a></p>

                            <p><br/> * To fetch all the Short Genetics Variations studies:
                                <br/> <a href="http://www.ebi.ac.uk/eva/webservices/rest/v1/meta/studies/all" target="_blank">http://www.ebi.ac.uk/eva/webservices/rest/v1/meta/studies/all</a></p>

                            <p><br/> * To fetch all the Structural Variations studies:
                                <br/> <a href="https://www.ebi.ac.uk/dgva/webservices/rest/v1/meta/studies/all" target="_blank">https://www.ebi.ac.uk/dgva/webservices/rest/v1/meta/studies/all</a></p>

                            <p><br/> * To fetch all  info of a study:
                                <br/> <a href="http://www.ebi.ac.uk/eva/webservices/rest/v1/studies/PRJEB4019/summary" target="_blank">http://www.ebi.ac.uk/eva/webservices/rest/v1/studies/PRJEB4019/summary</a></p>

                            <p><br/> * To fetch all file information of a study:
                                <br/> <a href="http://www.ebi.ac.uk/eva/webservices/rest/v1/studies/PRJEB4019/files?species=hsapiens_grch37" target="_blank">http://www.ebi.ac.uk/eva/webservices/rest/v1/studies/PRJEB4019/files?species=hsapiens_grch37</a></p>
                        </div>

                        <h3>Rate Limiting for Variant Region queries</h3>
                        <p>
                            Rate limiting has been implemented on Variant Region queries in order to ensure fairness
                            when serving multiple client requests. Therefore, please limit Variant Region API request rates to <b>5 requests/second</b>.
                            Higher rates of request might result in a HTTP 429 (Too many requests) response.

                            <br/><br/>
                            Also, when specifying the &quot;limit&quot; parameter in these region queries, please restrict it to <b>10000 or lower</b>. Greater values for this parameter are disallowed and will result in a HTTP 500 (Bad Request) response. Please note that this restriction on the &quot;limit&quot; parameter does not mean that large queries are forbidden altogether. It just means that only a maximum of 10,000 records will be served in a <b>single</b> request. Therefore, a client program can employ the limit parameter in conjunction with the "skip" parameter to "page through" the results from a large region. For example: the following queries can be used to page through the 20,259 results in the variant region 105000001-105500000 in chromosome 1 of the Mouse grcm38 assembly.
                            <ul>
                                <li>
                                    <a href="https://www.ebi.ac.uk/eva/webservices/rest/v1/segments/1:105000001-105500000/variants?species=mmusculus_grcm38&limit=10000" target="_blank">
                                    https://www.ebi.ac.uk/eva/webservices/rest/v1/segments/1:105000001-105500000/variants?species=mmusculus_grcm38&limit=10000
                                    </a>
                                </li>
                                <li>
                                    <a href="https://www.ebi.ac.uk/eva/webservices/rest/v1/segments/1:105000001-105500000/variants?species=mmusculus_grcm38&limit=10000&skip=10000" target="_blank">
                                    https://www.ebi.ac.uk/eva/webservices/rest/v1/segments/1:105000001-105500000/variants?species=mmusculus_grcm38&limit=10000&skip=10000
                                    </a>
                                </li>
                                <li>
                                    <a href="https://www.ebi.ac.uk/eva/webservices/rest/v1/segments/1:105000001-105500000/variants?species=mmusculus_grcm38&limit=10000&skip=20000" target="_blank">
                                    https://www.ebi.ac.uk/eva/webservices/rest/v1/segments/1:105000001-105500000/variants?species=mmusculus_grcm38&limit=10000&skip=20000
                                    </a>
                                </li>
                            </ul>
                        </p>
                    </div>
                </div>
                <div id="help-section" class="hide-div">
                    <nav aria-label="You are here:" role="navigation">
                        <ul class="breadcrumbs">
                            <li><a href="?Home">EVA</a></li>
                            <li>
                                <span class="show-for-sr">Current: </span> Help
                            </li>
                        </ul>
                    </nav>
                    <h2>Help</h2>
                    <ul class="tabs faq-tabs small-block-grid-4" data-tabs id="faq-tabs">
                        <li class="tabs-title is-active"><a href="#generalPanel" aria-selected="true">General</a></li>
                        <li class="tabs-title"><a href="#dataAccessPanel">Data Access</a></li>
                        <li class="tabs-title"><a href="#submissionPanel">Submission</a></li>
                        <li class="tabs-title"><a href="#accessionPanel">Accessions</a></li>
                        <li class="tabs-title"><a href="#citationPanel">Citation</a></li>
                    </ul>
                    <div class="tabs-content faq-tabs-content" data-tabs-content="faq-tabs">
                        <div class="tabs-panel is-active" id="generalPanel">
                            <ul class="padding-top-large padding-bottom-large">
                                <li><a href="#what-is-eva">What is the European Variation Archive (EVA)?</a></li>
                                <li><a href="#eva-normalization-variant-processing-steps">What are the EVA normalisation and variant processing steps?</a></li>
                                <li><a href="#eva-colaborates-with">With whom does the EVA collaborate?</a></li>
                                <li><a href="#how-to-follow-eva-development">How can I follow the development of the EVA?</a></li>
                            </ul>
                            <ul class="accordion" data-accordion>
                                <div id="what-is-eva">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>What is the European Variation Archive (EVA)?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <p>The European Variation Archive (EVA) is a primary deposition archive for genetic variation. The EVA is part of the EMBL-EBI infrastructure and accepts submission of <b>all</b> types of <b>precise</b> genetic variants, ranging from single nucleotide polymorphisms to large structural variants, observed in germline or somatic sources, from any organism. The EVA provide access to these data at multiple levels:</p>
                                            <p> i) The original variation data in VCF format as was submitted to the EVA, via the <a href="http://www.ebi.ac.uk/eva/?Study%20Browser&browserType=sgv" target="_blank">EVA Study Browser</a></p>
                                            <p> ii) Each individual variant and their genotypes, via the <a href="http://www.ebi.ac.uk/eva/?Variant%20Browser" target="_blank">EVA Variant Browser</a> and <a href="http://www.ebi.ac.uk/eva/?API" target="_blank">EVA API</a></p>
                                            <p> iii) All variant loci associated with RS ids found in one assembly, via the <a href="https://www.ebi.ac.uk/eva/?RS-Release" target="_blank">EVA RS release</a></p>

                                        </div>
                                    </li>
                                </div>
                                <div id="eva-normalization-variant-processing-steps">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>What are the EVA normalisation and variant processing steps? </h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <div>
                                                <img class="faq-images" src="img/optimized/EVA_Variant_Level_Normalisation.svg" alt="EVA Variant Level Processing: Submitted data from the EVA Study Browser -> Variants are merged, normalized and annotated for functional consequences and statistical values
                                        -> Non-human variants are accessioned by the EVA. Human variants are brokered to dbSNP at NCBI and resulting SS/RS accessions are ingested by EVA -> Data are exposed as JSON objects either via the EVA website GUI or API" data-svg-source="https://drive.google.com/file/d/1zam0DBdg00b8GnmtWR0LLy6SnHqUGPWP/view"><br /><br />
                                                <h5>Normalisation</h5>
                                                <p>
                                                    Variants submitted to the EVA have been determined by a number of different algorithms and software packages. As a result, the VCF files generated by these differing methodologies describe variants in a number of different ways. The primary processing step of the EVA is to normalise variant representation following two basic rules:
                                                    <ol>
                                                        <li>Each variant is shifted to be left-aligned</li>
                                                        <li>The Start and End positions represent exactly the range where the variation occurs (which could, in the case of insertions, result in the reference allele being recorded as 'empty')</li>
                                                    </ol>
                                                </p>
                                                <p>Examples of our variant normalisation process can be seen <a href="https://docs.google.com/presentation/d/1WqSiT5AEEQF9jdIewdYIp-I0G5ozkFP3IikfCJZO1dc/edit#slide=id.ge1548f905_0_592" target="_blank">here</a></p>

                                            </div>
                                            <div>
                                                <h5>Annotation</h5>
                                                <p>
                                                    Once variants have been normalised, the EVA uses the <a href="http://www.ensembl.org/info/docs/tools/vep/" target="_blank">Variant Effect Predictor (VEP) </a>  of Ensembl to annotate variant consequences.
                                                    The variant consequences are described using Sequence Ontology terms and both the VEP version and Ensembl gene build used are described via the "i" help bubbles on the <a href="http://www.ebi.ac.uk/eva/?Variant%20Browser" target="_blank">EVA Variant Browser</a>.
                                                    <br /><br /> N.B. Variants that have been mapped to a reference genome sequence that is not supported by Ensembl are not annotated.
                                                </p>
                                            </div>
                                            <div>
                                                <h5>Statistical calculations</h5>
                                                <p>The EVA adopts the classical definition of allele frequency (AF): 'a measure of the relative frequency of an allele at a genetic locus in a given population'.
                                                    The AF value(s) stored by the EVA for each variant is (are) study specific - i.e. the same variant reported in two distinct studies shall be given two allele frequencies, one for each study. There are two methodologies by which the EVA is able to determine allele frequency values, dependent on the datatype of the study in question:</p>
                                                <p><span style="font-size:14px;font-weight:bold;">Variants associated with genotypes:</span></p>
                                                <p>
                                                    For variants associated with genotypes, the EVA determines the AF values via the calculation:
                                                    <br /><br />AF = (number of alternate allele observations (AC)) / (number of observations (AN))
                                                    <br /><br />The result of this calculation allows the EVA to also store the minor allele frequency (MAF) for each variant (defined as the minimum of the reference or alternative allele frequency) and the MAF allele (the allele associated with the MAF).
                                                </p>
                                                <p><span style="font-size:14px;font-weight:bold;">Variants not associated with genotypes:</span></p>
                                                <p>
                                                    For variants that are not associated with genotypes, the EVA is dependent on the AF value(s) estimated from the primary data and provided in the submitted VCF files(s).
                                                    AF values that are specifically provided in the submitted aggregated VCF file(s) are directly stored. In cases where no AF is provided then the EVA uses the AC and AN values in the submitted aggregated VCF file(s) to calculate AF value(s) via the calculation:
                                                    <br /><br />AF = AC / AN
                                                </p>
                                                <p><span style="font-size:14px;font-weight:bold;">Population / sample cohort allele frequency values:</span></p>
                                                <p>
                                                    The EVA accepts submission of <a href="http://pngu.mgh.harvard.edu/~purcell/plink/data.shtml#ped" target="_blank">pedigree </a> files, or structured samples (using "derived_from" and/or "subject" layers), to define populations and cohorts within studies.
                                                    In cases where such information is associated with variants that have genotypes then the EVA calculates intra-study population/cohort specific AF values via the method described above, with the caveat that the (total number of populations/cohorts):(total number of samples) ratio must be less than 1:10.
                                                    For studies that do not contain genotypes but instead provide intra-study population/cohort AF values in the submitted aggregated VCF file(s), or AC and AN values, then these are directly stored, or calculated by the EVA using the method described above, again with the caveat that a ratio of 1:10 (total number of populations/cohorts):(total number of samples) must not be exceeded.

                                                    <br /><br />*NB: there are a low number of variants for which the EVA is unable to determine any allele frequency value(s) as the submitted VCF file(s) contain neither genotypes nor AF or AC and AN values. The EVA discourages submission of variants that cannot be associated with an AF.
                                                </p>
                                            </div>

                                        </div>
                                    </li>
                                </div>
                                <div id="eva-colaborates-with">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>With whom does the EVA collaborate?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <div><img class="faq-images" src="img/EVA_Related_Projects.png" alt="The EVA receives data from direct submissions, and exchanges it with international collaborators"><br/><br/><br/></div>
                                            <p><span style="font-size:14px;font-weight:bold;">dbSNP and dbVar</span></p>
                                            <div>
                                                <p>
                                                    dbSNP and dbVar maintain the accessions for human variants, while the EVA accessions any non-human species. Human data is routinely exchanged between resources.
                                                </p>
                                            </div>
                                            <p><span style="font-size:14px;font-weight:bold;">Database of Genomic Variants Archive</span></p>
                                            <div>
                                                <p>
                                                    The sister database of the EVA was initially solely responsible for storing structural variants. This data can now be archived in the EVA, as long as it can be represented in Variant Call Format (VCF).
                                                </p>
                                            </div>
                                            <p><span style="font-size:14px;font-weight:bold;">Europe PubMed Central</span></p>
                                            <div>
                                                <p>
                                                    An archive for life science literature, which can be directly linked to archived data within the EVA using unique study accessions.
                                                </p>
                                            </div>
                                            <p><span style="font-size:14px;font-weight:bold;">Open Targets</span></p>
                                            <div>
                                                <p>
                                                    A private-public partnership aiming to systematically identify and prioritize drug targets. The EVA aids in the automation and manual curation of ClinVar traits for the Open Targets platform.
                                                </p>
                                            </div>
                                            <p><span style="font-size:14px;font-weight:bold;">Accelerating Medicines Partnership - Type 2 Diabetes</span></p>
                                            <div>
                                                <p>
                                                    A knowledge portal enabling observation of human genetic information linked to type 2 diabetes. With the EVA as one of the resources leading the federation component, the software architecture is designed and developed between both resources, to allow submission and consumption across multiple nodes.
                                                </p>
                                            </div>
                                            <p><span style="font-size:14px;font-weight:bold;">European Nucleotide Archive and Biosamples</span></p>
                                            <div>
                                                <p>
                                                    Data archived at the EVA is first brokered to BioSamples and the ENA to provide unique accessions for each submission. Reference sequences used by submitters must also be present within the ENA, which is part of the International Nucleotide Sequence Database Collaboration.
                                                </p>
                                            </div>
                                            <p><span style="font-size:14px;font-weight:bold;">Ensembl</span></p>
                                            <div>
                                                <p>
                                                    Ensembl is a genome browser that provides support for a large diversity of species. The EVA provides non-human variant and genotype information, which can then be viewed and downloaded via the Ensembl web browser and API. Ensembl provide the Variant Effect Predictor (VEP) tool, which the EVA uses to functionally annotate archived variants.
                                                </p>
                                            </div>
                                            <p><span style="font-size:14px;font-weight:bold;">Global Alliance for Genomics and Health</span></p>
                                            <div>
                                                <p>
                                                    An international organization that defines standards for genomic data sharing, with the EVA being one of the driver projects that guide GA4GH development efforts, as well as an active contributor to specifications like the Variant Call Format (VCF).
                                                </p>
                                            </div>
                                            <p><span style="font-size:14px;font-weight:bold;">Elixir</span></p>
                                            <div>
                                                <p>
                                                    An organization that brings together infrastructure and various scientific communities across Europe. Elixir both supports the EVA as a recommended deposition database, and FAIR data principles for the accessibility and reusablity of genetic data.
                                                </p>
                                            </div>
                                        </div>
                                    </li>
                                </div>
                                <div id="how-to-follow-eva-development">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>How can I follow the development of the EVA?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <h5>The following are our GitHub repositories:</h5>
                                            <div>
                                                <ul style="list-style-type:disc">
                                                    <li>
                                                        <p>
                                                            The <a href="https://github.com/EBIvariation/vcf-validator/wiki/User's-Guide">EVA VCF validator</a> checks that a file is compliant with the VCF specification.
                                                            It includes and expands the validations supported by the vcftools suite. It supports versions 4.1, 4.2 and 4.3 of the specification.
                                                        </p>
                                                    </li>
                                                    <li>
                                                        <p>
                                                            The <a href="https://github.com/EBIvariation/eva-pipeline">EVA pipeline</a> processes VCF files, stores the variation data in a database and post-processes it, in a way that can be later consumed via web services.
                                                        </p>
                                                    </li>
                                                    <li>
                                                        <p>
                                                            The <a href="https://github.com/EBIvariation/eva-ws"> EVA web services</a> serve the data generated and stored by the EVA pipeline. They follow the REST paradigm and can be consumed by any external application.
                                                        </p>
                                                    </li>
                                                    <li>
                                                        <p>
                                                            The <a href="https://github.com/EBIvariation/eva-web">EVA website</a> displays the data served by the EVA REST web services API in a user-friendly way.
                                                        </p>
                                                    </li>
                                                </ul>
                                            </div>
                                            <h5>Acknowledgement</h5>
                                            <p>We would like to acknowledge the following software support.</p>
                                            <div>
                                                <a href="https://github.com/" target="_blank" style="border-bottom-style: none;">
                                                    <img src="img/github.png" alt="GitHub" class="acknowledgement-logo">
                                                </a>
                                                <a href="http://www.ej-technologies.com/products/jprofiler/overview.html" target="_blank" style="border-bottom-style: none;">
                                                    <img src="img/jprofiler_large.png" alt="JProfiler" class="acknowledgement-logo">
                                                </a>
                                                <a href="http://www.jetbrains.com/idea/" target="_blank" style="border-bottom-style: none;">
                                                    <img src="img/intelliJ.png" alt="IntelliJ" class="acknowledgement-logo">
                                                </a>
                                            </div>
                                        </div>
                                    </li>
                                </div>
                            </ul>
                        </div>
                        <div class="tabs-panel is-active" id="dataAccessPanel">
                            <ul class="padding-top-large padding-bottom-large">
                                <li><a href="#how-to-consume-variant-data">How can I consume variant data from the EVA?</a></li>
                                <li><a href="#supported-browsers">Which browsers does the EVA website support?</a></li>
                                <li><a href="#how-to-search-ss-rs-id">How can I search EVA with RS and SS identifiers?</a></li>
                                <li><a href="#can-download-accesioned-variants">Can I download accessioned variants from the EVA?</a></li>
                                <li><a href="#how-to-track-variants">How can I track variants that have been generated, merged and/or deprecated over time?</a></li>
                            </ul>
                            <ul class="accordion" data-accordion>
                                <div id="how-to-consume-variant-data">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>How can I consume variant data from the  EVA?</h5></a>

                                        <div class="accordion-content" data-tab-content>
                                            <p>
                                                Links are shown on the <a href="?Study%20Browser&browserType=sgv">EVA Study Browser</a>  to both the raw submitted files and the 'EVA browsable files'.
                                                Not all submitted VCF files for a study are browsable due to overlapping information. For example, we have studies where the submitted VCF files contain the sample and variant data split by population, but also merged together. It would be redundant for the EVA to load all of these data to our variant warehouse. Please note that due to processing time required not all studies have 'EVA browsable files' just yet.
                                            </p>
                                            <p> Additionally users can download query results from the <a href="?Variant%20Browser&species=hsapiens_grch37&selectFilter=region&studies=PRJEB4019%2CPRJEB6930%2CPRJEB5439%2CPRJEB5829%2CPRJEB8652%2CPRJEB8650%2CPRJEB8639%2CPRJEB6042%2CPRJX00001%2CPRJEB8705%2CPRJEB8661%2CPRJEB7895%2CPRJEB7217%2CPRJEB7218%2CPRJEB6041&region=1%3A3000000-3100000">EVA variant browser</a> directly and all of our loaded data is available programmatically via our <a href="?API">API</a>.</p>
                                        </div>
                                    </li>
                                </div>
                                <div id="supported-browsers">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>Which browsers does the EVA website support?</h5></a>

                                        <div class="accordion-content" data-tab-content>
                                            <p>
                                                The EVA website employs HTML5 technologies and standards and is
                                                optimized for use with modern browsers. In particular, Chrome
                                                (version 73 or higher) and Firefox (version 66 or higher) are fully
                                                supported, however older versions of these browsers may give rise
                                                to errors.<br />
                                                The EVA website also supports mobile versions of web browsers, with
                                                limited functionality. Please report all errors to the
                                                <a href="mailto:eva-helpdesk@ebi.ac.uk" target="_top">EVA helpdesk</a>.
                                            </p>
                                        </div>
                                    </li>
                                </div>
                                <div id="how-to-search-ss-rs-id">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>How can I search EVA with RS and SS identifiers?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <p>
                                                The EVA offers users the ability to search 'SS' and 'RS' identifiers generated by the EVA via
                                                the <a href="https://www.ebi.ac.uk/eva/?Home">Home page</a>,

                                                the <a href="?Variant%20Browser&species=btaurus_umd31&selectFilter=region&studies=PRJEB6119%2CPRJEB7061%2CPRJEB14878&region=1%3A3000000-3100000">Variant Browser</a>
                                                and the <a href="?API">API</a>
                                            </p>
                                        </div>
                                    </li>
                                </div>
                                <div id="can-download-accesioned-variants">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>Can I download accessioned variants from the EVA?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <p>
                                                Yes. Each non-human study archived in EVA will contain a VCF file with the accessioned variants (SS ids).
                                                After an RS release the EVA will provide a VCF dump for each species/assembly combination.

                                                These files contain the rs ID positional information, but not the genotype nor annotation data.
                                            </p>
                                        </div>
                                    </li>
                                </div>
                                <div id="how-to-track-variants">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>How can I track variants that have been generated, merged and/or deprecated over time?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <p>
                                                In addition to outputting a VCF file for each rs build the EVA shall also generate a file that will allow users to track changes to variant numbers over time.
                                                These files shall include details of all newly generated identifiers for each rs build, a list of those variants that have been deprecated as well as information on any variants that have been merged together.
                                                The EVA shall follow the same rules as dbSNP for generating, deprecating and merging variant identifiers.
                                            </p>
                                        </div>
                                    </li>
                                </div>
                            </ul>
                        </div>
                        <div class="tabs-panel is-active" id="submissionPanel">
                            <ul class="padding-top-large padding-bottom-large">
                                <li><a href="#why-submit-data-to-eva">Why submit data to the EVA?</a></li>
                                <li><a href="#is-my-data-suitable">Is my data suitable for submission to the EVA?</a></li>
                                <li><a href="#how-to-gen-vcf">How do I generate a VCF file?</a></li>
                                <li><a href="#what-happens-to-submitted-data">What happens to my data once submitted?</a></li>
                                <li><a href="#accept-submission-same-variation-file-format">Do EVA and dbSNP accept submission of the same genetic variation file format?</a></li>
                                <li><a href="#do-eva-and-dbsnp-collect-metadata-same-way">Do EVA and dbSNP collect metadata in the same way?</a></li>
                                <li><a href="#submitted-data-held-privately">Can my submitted data be held privately at the EVA?</a></li>
                                <li><a href="#what-data-is-brokered-to-dbsnp">Which data is brokered to DBSNP?</a></li>
                                <li><a href="#how-do-we-deal-with-chromosome-name-diversity">How do we deal with chromosome or contig name diversity?</a></li>
                                <li><a href="#consent-statement-for-human-genotype-data">Consent statement for human genotype data</a></li>
                            </ul>
                            <ul class="accordion" data-accordion>
                                <div id="why-submit-data-to-eva">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>Why submit data to the EVA?</h5></a>

                                        <div class="accordion-content" data-tab-content>
                                            <p>The EVA provides to the community a completely free, secure and permanent solution to data sharing. Each project, VCF file and sample that is submitted to the EVA is assigned a unique identifier that is accessible in perpetuity and is therefore able to be referenced in publication, for example.
                                                The <a href="mailto:eva-helpdesk@ebi.ac.uk" target="_top">EVA helpdesk</a> provides support to submitters, and users, to ensure accurate represention and proper integration of the submitted data with other EMBL-EBI resources such as the <a href="https://www.ebi.ac.uk/ega/" target="_blank">EGA</a> and <a href="http://www.ensembl.org/index.html" target="_blank">Ensembl</a>. A final advantage of submitting to the EVA is that variants are brokered to the National Centre for Biotechnology Information (<a href="http://www.ncbi.nlm.nih.gov/" target="_blank">NCBI</a>) on the submitter's behalf, negating the requirement for independent submissions.</p>
                                        </div>
                                    </li>
                                </div>
                                <div id="is-my-data-suitable">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>Is my data suitable for submission to the EVA?</h5></a>

                                        <div class="accordion-content" data-tab-content>
                                            <b>Is your data open access?</b>
                                            <p>The most important consideration is that all data archived at the EVA is open access. As such, there are no restrictions as to who can access the data or how such data is reused. It is the submitter's responsibility to ensure that the data archived at the EVA complies to this open access policy.<br/>
                                                <b>
                                                    If consent was gathered for any individual genotype data in the study then we will require a <a href="https://docs.google.com/document/d/1UaRmimAe919IZYIH55mDXMe-4sHsRVqaP4moqx2IYE4/edit" target="_blank">consent statement</a> to be completed prior to submission.
                                                </b>
                                            </p>


                                            <br/>

                                            <b>Is your data represented in a valid VCF file?</b>
                                            <p>
                                                Genetic variants submitted to the EVA must be described in the <a href="https://samtools.github.io/hts-specs/VCFv4.3.pdf" target="_blank">Variant Call Format (VCF)</a>.
                                                The EVA only accepts VCF files that adhere to the specification and match a reference sequence (Versions <a href="https://samtools.github.io/hts-specs/VCFv4.1.pdf" target="_blank">v4.1</a>, <a href="https://samtools.github.io/hts-specs/VCFv4.2.pdf" target="_blank">v4.2</a>, & <a href="https://samtools.github.io/hts-specs/VCFv4.3.pdf" target="_blank">v4.3</a> accepted)
                                                Submitters can make use of the <a href="https://github.com/EBIvariation/vcf-validator/wiki/User's-Guide" target="_blank">EVA VCF validation suite</a> to ensure they fulfill these requirements.
                                                Furthermore, VCF files submitted to the EVA should provide either sample genotypes and/or aggregated sample summary-level allele frequencies.
                                                Variant lines should also be unique and not specify duplicate loci as this causes issues during steps such as calculation of population frequencies.
                                                Although both per-sample VCFs and merged VCFs are acceptable, merged VCF files containing sample information through the FORMAT field are preferred over separate VCF files for each sample. Single VCF files allow data to be easily reusable and reproducible. Therefore, we recommend merging per-sample VCF files before submission.
                                                This allows the data to be easily reusable/reproducible.
                                            <ul>
                                                <li><b>Population allele frequencies</b>: Alleles typically exist at different frequencies in different populations and this information provides useful supporting evidence for all submitted variants. Population-level allele frequencies can be provided in the INFO field of the VCF (see section 1.6.1 of the <a href="http://samtools.github.io/hts-specs/VCFv4.3.pdf" target="_blank">VCF specification</a>).</li>
                                                <li><b>Individual genotypes</b>: EVA accepts individual genotypes from samples provided by donors that have consented to having their DNA sequence housed in a public database. Genotypes can be provided in the FORMAT field of the VCF (see section 1.6.2 of the <a href="http://samtools.github.io/hts-specs/VCFv4.3.pdf" target="_blank">VCF specification</a>).</li>
                                            </ul>
                                            </p>

                                            <br/>

                                            <b>Is the reference sequence INSDC registered?</b>
                                            <p>
                                                The EVA requires that all variants are submitted with an asserted position on an <a href="https://www.insdc.org/" target="_blank">INSDC</a> sequence. This means that the reference allele submitted for every variant must match the position in a sequence that has been accessioned in either GenBank or ENA database. An â€œINSDC sequenceâ€ can be any of, but not restricted to, the following assembled sequences:
                                            <ul>
                                                <li>Genome assembly</li>
                                                <li>Transcriptome</li>
                                                <li>Gene sequence</li>
                                            </ul>
                                            </p>
                                            <p>
                                                Aligning all of our submitted data with INSDC sequences enables integration with other EMBL-EBI resources including Ensembl and is an important step in maintaining standardisation at the EVA.
                                                <b>Importantly this requires all sequence identifiers in your VCF to match those in the reference FASTA file.</b>
                                            </p>
                                        </div>

                                    </li>
                                </div>

                                <div id="how-to-gen-vcf">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>How do I generate a VCF file?</h5></a>

                                        <div class="accordion-content" data-tab-content>
                                            <p>
                                                The basic template of a minimal VCF file, where no genotypes are given, looks like this:
                                            </p>
                                            <pre>
                                                ##fileformat=VCFv4.3
                                                #CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO
                                            </pre>
                                            <p>
                                                We advise submitters to consult the <a href="http://samtools.github.io/hts-specs/VCFv4.3.pdf" target="_blank">VCF specification</a> (specifically sections 1.1 and 5) when converting data to VCF in order to ensure that files generated are valid, as this is a mandatory requirement.
                                            </p>

                                            <p><b>Conversion of spreadsheet to VCF</b></p>
                                            <p>
                                                Representing variant data in a spreadsheet can be convenient when you have a small number of variants to report. This file can be used as an example or template: <a href="https://docs.google.com/spreadsheets/d/13MdtQdfbexoVNQIuBeRtZUdw1NUzYh0wXXtUy9d0TB4/edit?usp=sharing">spreadsheet_to_vcf.xlsx</a> (you will need to create a copy). When reporting a large number of variants, the use of specialised software may be a better approach.
                                            </p>
                                            <p>
                                                Using a spreadsheet editor, you can use the 'Save as' feature to convert the given file to a tab delimited text file. This can also be done using an online editor such as Google Sheets with the 'Download as' option. A tab character will be applied between values entered into separate cells. The result should be a file named 'spreadsheet_to_vcf.tsv'.
                                            </p>
                                            <p>
                                                The file can be renamed as appropriate although the extension must be changed from '.tsv' to '.vcf' to create a VCF file.
                                            </p>
                                            <p>
                                                The data entered will remain the same, and the file can be manually checked to ensure spaces between entries are tabulated and there are no extra spaces at the end of lines.
                                            </p>

                                            <p><b>Conversion from PED / genotyping array files to VCF</b></p>
                                            <p>
                                                You can use <a href="https://www.cog-genomics.org/plink/1.9/" target="_blank">PLINK1.9</a> to convert from PLINK text format to VCF, using the command <i>--recode vcf</i>, along with <i>--reference-allele</i> to select the reference alleles. Find more details <a href="https://www.cog-genomics.org/plink/1.9/data#recode" target="_blank">here</a>.
                                            </p>

                                            <p><b>Useful tools for handling VCF files</b></p>
                                            <p>
                                                <ul>
                                                    <li>
                                                    VCF Validation: <a href="https://github.com/EBIvariation/vcf-validator/wiki/User's-Guide" target="_blank">EVA VCF validation suite</a>
                                                    </li>

                                                    <li>
                                                    Sorting, merging and manipulation of VCF files: <a href="https://github.com/samtools/bcftools" target="_blank">bcftools</a>
                                                    </li>

                                                    <li>Variant callers: <a href="https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeCaller.php" target="_blank">GATK HaplotypeCaller</a>, <a href="http://www.well.ox.ac.uk/platypus-doc" target="_blank">Platypus</a>, <a href="http://varscan.sourceforge.net/" target="_blank">VarScan</a>, <a href="http://csb5.github.io/lofreq/" target="_blank">LoFreq</a>, <a href="https://github.com/ekg/freebayes" target="_blank">FreeBayes</a>, <a href="http://snver.sourceforge.net/" target="_blank">SNVer</a>, <a href="http://www.htslib.org/" target="_blank">SAMtools</a> and <a href="https://github.com/AstraZeneca-NGS/VarDict" target="_blank">VarDict</a>
                                                    </li>
                                                </ul>
                                            </p>

                                            <p><b>Examples of valid VCF submissions</b></p>
                                            <p>
                                                <ul>
                                                    <li><a href="http://ftp.ebi.ac.uk/pub/databases/eva/PRJEB27443/two_indels-CAU_100718.vcf.gz" target="_blank">PRJEB27443</a> - Simple INDEL submission with allele frequencies</li>
                                                    <li><a href="http://ftp.ebi.ac.uk/pub/databases/eva/PRJEB27177/07_06_2018_pine.vcf.gz" target="_blank">PRJEB27177</a> - Single sample VCF with genotype</li>
                                                    <li><a href="http://ftp.ebi.ac.uk/pub/databases/eva/PRJEB27093/68_Osativa_lines.vcf.gz" target="_blank">PRJEB27093</a> - Multiple sample VCF with genotypes</li>
                                                    <li><a href="http://ftp.ebi.ac.uk/pub/databases/eva/PRJEB4163/Sratti_lanes.recode_filtered.normalized_merged_noPL.vcf.gz" target="_blank">PRJEB4163</a> - VCF with population allele frequencies and genotypes</li>
                                                    <li><a href="http://ftp.ebi.ac.uk/pub/databases/eva/PRJEB6041/CAR_Batch123.vcf.gz" target="_blank">PRJEB6041</a> - VCF with allele counts (for calculating AF)</li>
                                                </ul>
                                            </p>
                                            <p>Further examples can be found on our <a href="http://ftp.ebi.ac.uk/pub/databases/eva" target="_blank">FTP</a> site.</p>
                                        </div>
                                    </li>
                                </div>
                                <div id="what-happens-to-submitted-data">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>What happens to my data once submitted?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <img class="faq-images" src="img/EVA_FIle_Level_Processing.svg" alt="EVA File Level Processing: Submitter brings VCF File(s) and metadata template to EVA -> Submitted data are validated against VCF specification and reference genome sequence ->
                                    Accessions are sent to submitter,for referencing in publication for example -> EVA provide links to download the submitted data via the Study Browser"><br /><br />
                                            <h5>Submission validation processes</h5>
                                            <p>The <a href="https://github.com/EBIvariation/vcf-validator/wiki/User's-Guide" target="_blank">EVA VCF validation suite</a> provides tools to check file validity previous to submission.</p>
                                            <p><span style="font-size:14px;font-weight:bold;">VCF Specification:</span></p>
                                            <p>
                                                All VCF files submitted to EVA are validated for adherence to the <a href="https://samtools.github.io/hts-specs/" target="_blank">format standard</a> using the <a href="https://github.com/EBIvariation/vcf-validator/wiki/User's-Guide" target="_blank">EVA VCF validation suite</a> which includes all the checks from the vcftools suite, and some more that involve lexical, syntactic and semantic analysis of the VCF input.
                                                The EVA VCF validation suite also includes a debugging tool to automatically correct many of the common errors found in files. To improve processing time, submitters are encouraged to pre-validate VCF files prior to submission.
                                            </p>
                                            <p><span style="font-size:14px;font-weight:bold;">Genome Assembly:</span></p>
                                            <p>
                                                To improve interoperability of variant data submitted to EVA with other resources at EMBL-EBI, and the wider open-access community, all submitted VCF files are subject to validation against the INSDC accessioned reference sequence indicated by the submitter in their metadata. The EVA VCF validation suite provides an assembly checker tool to validate the reference alleles in the VCF files against the reference sequence. EVA is able to only accept files that match a known sequence at 100%.
                                            </p>
                                            <h5>Archive process</h5>
                                            <p> All  VCF files and novel samples that are submitted to EVA are permanently and securely archived at the <a href="http://www.ebi.ac.uk/ena" target="_blank">European Nucleotide Archive </a> and <a href="http://www.ebi.ac.uk/biosamples/" target="_blank">BioSamples </a>, respectively. The EVA provides access to all submitted data via the <a href="http://www.ebi.ac.uk/eva/?Study%20Browser&browserType=sgv" target="_blank">Study Browser</a>.</p>
                                            <p> Variants within VCF files submitted to EVA are normalised, annotated and used for statistical calculations (via methodologies described below) and these EVA processed data are available via the <a href="http://www.ebi.ac.uk/eva/?Study%20Browser&browserType=sgv" target="_blank">Study Browser</a>, <a href="http://www.ebi.ac.uk/eva/?Variant%20Browser" target="_blank">Variant Browser</a> and the <a href="http://www.ebi.ac.uk/eva/?API" target="_blank">REST API</a>.
                                            <p> Finally, the EVA brokers all submitted data to the NCBI: <a href="http://www.ncbi.nlm.nih.gov/projects/SNP/" target="_blank">dbSNP</a>, for short variants (&lt;50bp) <a href="http://www.ncbi.nlm.nih.gov/dbvar" target="_blank">dbVar</a> for large variants (&gt;50bp)</p>
                                        </div>
                                    </li>
                                </div>
                                <div id="accept-submission-same-variation-file-format">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>Do EVA and dbSNP accept submission of the same genetic variation file format?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <p>
                                                Yes. EVA, like dbSNP, accepts submission of genetic variation data that is described in Variant Call Format (VCF) files.
                                                However, unlike dbSNP, EVA does not have a custom VCF specification. Instead EVA only accepts submission of VCF files that conform to the <a href="https://samtools.github.io/hts-specs/VCFv4.4.pdf" target="_blank">specification guidelines</a>.
                                                Conformity to these guidelines is checked at time of submission, but can also be checked beforehand using our <a href="https://github.com/EBIvariation/eva-sub-cli/blob/main/README.md" target="_blank">submission command line tool</a>. Further guidelines for submission to EVA can be found <a href="?Submit%20Data">here</a>.
                                            </p>
                                        </div>
                                    </li>
                                </div>
                                <div id="do-eva-and-dbsnp-collect-metadata-same-way">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>Do EVA and dbSNP collect metadata in the same way?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <p>
                                                No. Although there are many similarities, the EVA and dbSNP metadata templates, which are submitted along with VCF files, are different.
                                                The EVA metadata template asks the submitter to provide information about the project, analyses and samples so that these can be archived simultaneously along with the VCF file(s), whereas this is a stepwise process during dbSNP submission.
                                                The EVA metadata template can be found <a href="https://raw.githubusercontent.com/EBIvariation/eva-sub-cli/main/eva_sub_cli/etc/EVA_Submission_template.xlsx">here</a> and a mocked up version for a fictional study is <a href="https://raw.githubusercontent.com/EBIvariation/eva-sub-cli/main/eva_sub_cli/etc/EVA_Submission_Example.xlsx">here</a>.
                                                Detailed information on how to submit to the EVA can be found <a href="?Submit%20Data">here</a>.

                                            </p>
                                        </div>
                                    </li>
                                </div>
                                <div id="submitted-data-held-privately">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>Can my submitted data be held privately at the EVA?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <p>
                                                Yes. Data submitted to the EVA can be held privately for up to two years.
                                                The date of publication is set by the submitter using the "Hold Date" field of the <a href="https://raw.githubusercontent.com/EBIvariation/eva-sub-cli/main/eva_sub_cli/etc/EVA_Submission_template.xlsx">EVA metadata template</a>.
                                            </p>
                                        </div>
                                    </li>
                                </div>
                                <div id="what-data-is-brokered-to-dbsnp">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>Which data is brokered to DBSNP?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <p>Any variation dataset containing human genotype or allele frequency will be brokered to dbSNP for variant accessioning an assignment of RS ids.</p>
                                        </div>
                                    </li>
                                </div>
                                <div id="how-do-we-deal-with-chromosome-name-diversity">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>How do we deal with chromosome or contig name diversity?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <p>Contig naming convention can vary depending on how the reference genome was obtained. We strive to support multiple chromosome names, such as INSDC or RefSeq identifiers and ENA and Genbank chromosome names. In case we do not support the naming convention you have used, please contact <a href="mailto:eva-helpdesk@ebi.ac.uk">eva-helpdesk@ebi.ac.uk</a>.</p>
                                        </div>
                                    </li>
                                </div>
                                <div id="consent-statement-for-human-genotype-data">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>Consent statement for human genotype data</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <p>The EVA is an open access resource, and it is the submitter's responsibility to ensure their data is compliant with this policy.
                                                If human genotype data is to be submitted then we request a <a href="https://docs.google.com/document/d/1UaRmimAe919IZYIH55mDXMe-4sHsRVqaP4moqx2IYE4" target="_blank">Consent Statement</a> to be completed prior to submission via email to <a href="mailto:eva-helpdesk@ebi.ac.uk">eva-helpdesk@ebi.ac.uk</a>.
                                                Your submission will only be processed after receiving the consent statement. If your data contains allele frequencies and no genotypes then no consent statement is needed.
                                                Please consider that if your data requires controlled access, an alternative EMBL-EBI service, the European Genome-phenome Archive (EGA; www.ebi.ac.uk/ega), will be more suitable for your submission.</p>
                                        </div>
                                    </li>
                                </div>
                            </ul>
                        </div>
                        <div class="tabs-panel is-active" id="accessionPanel">
                            <ul class="accordion" data-accordion>
                                <div>
                                    <h3 class="padding-bottom-large">Frequently Asked Questions</h3>
                                    <ul class="padding-bottom-large">
                                        <li><a href="#accessions-administred-by-eva">What metadata accessions are administered by the EVA?</a></li>
                                        <li><a href="#variant-accession-administred-by-eva">What variant accessions will be administered by the EVA?</a></li>
                                        <li><a href="#retrieve-existing-dbsnp-ids">Will it be possible to retrieve existing dbSNP identifiers from the EVA?</a></li>
                                        <li><a href="#why-variants-in-different-strand">Why are some variants in a different strand than the one dbSNP reports?</a></li>
                                        <li><a href="#cant-find-dbsnp-ids-in-eva-website">Why can't I find some of the dbSNP identifiers on the EVA website?</a></li>
                                        <li><a href="#eva-builds">Will there be EVA builds or releases like the dbSNP ones?</a></li>
                                    </ul>
                                    <br>
                                    <ul class="accordion" data-accordion>
                                        <div id="accessions-administred-by-eva">
                                            <li class="accordion-item is-active" data-accordion-item>
                                                <a class="accordion-title"><h5>What metadata accessions are administered by the EVA?</h5></a>
                                                <div class="accordion-content" data-tab-content>
                                                    <p>
                                                        The EVA follows the SRA model for accessioning metadata:
                                                    <ul>
                                                        <ul>
                                                            <li>A submitted project is administered a 'project' accession. These begin "PRJEB" followed by a numerical sequence.</li>
                                                            <li>Each analysis object within a project is administered an 'analysis' accession. These begin "ERZ" followed by a numerical sequence.</li>
                                                        </ul>
                                                    </ul>
                                                    </p>
                                                    <p>
                                                        Both the project and analysis accessions are sent to the submitter once the data has been validated and fully archived at EMBL-EBI. The project and analysis accessions are stable identifiers that are suitable for publication.
                                                    </p>
                                                </div>
                                            </li>
                                        </div>
                                        <div id="variant-accession-administred-by-eva">
                                            <li class="accordion-item is-active" data-accordion-item>
                                                <a class="accordion-title"><h5>What variant accessions will be administered by the EVA?</h5></a>
                                                <div class="accordion-content" data-tab-content>
                                                    <p>
                                                        The EVA shall follow the same accessioning principles employed by dbSNP: non-human variants that are submitted to the EVA are issued 'Submitted SNP' (SS) identifiers, and these are periodically clustered to form 'Reference SNP' (RS) identifiers.
                                                    </p>

                                                    <h5>Submitted SNP (SS)</h5>
                                                    <p>
                                                        A Submitted SNP (SS) identifies a variant submitted in the context of a genomic study. New SS IDs issued by the EVA will always be linked to a change from a reference allele to a single alternate allele. This means that a SS ID uniquely identifies a combination of:
                                                        <ul>
                                                            <li>Taxonomy (e.g. 9615 - dog)</li>
                                                            <li>Reference sequence (e.g. CanFam3.1)</li>
                                                            <li>EVA study or dbSNP batch (e.g. PRJEB24066)</li>
                                                            <li>Contig or chromosome</li>
                                                            <li>Start position</li>
                                                            <li>Reference allele</li>
                                                            <li>Alternate allele</li>
                                                        </ul>
                                                    </p>
                                                    <p>
                                                        Most pre-existing dbSNP identifiers also follow this rule but, in some cases, multiple alternate alleles in the same location could be assigned a single SS ID depending on the representation at the time of submission.
                                                    </p>
                                                    <p>
                                                        An SS ID is linked to a variant in a particular study. Therefore, the same variant reported across multiple files in the study will still be assigned only one SS ID.
                                                    </p>
                                                    <h6>Submissions</h6>
                                                    <p>
                                                        The following restrictions apply to EVA submissions, although legacy dbSNP identifiers are preserved even if they donâ€™t:
                                                        <ul>
                                                            <li>Variants must have a specific location in a sequence accessioned by INSDC in order to be archived and accessioned.</li>
                                                            <li>Variants must have supporting evidence (genotypes and/or allele frequencies) in order to be archived and accessioned.</li>
                                                            <li>Non-variants (as in gVCF reference blocks) will be archived, but will neither be assigned an SS ID nor displayed in the Variant Browser.</li>
                                                        </ul>
                                                    </p>
                                                    <h6>Reporting</h6>
                                                    <p>
                                                        The list of SS IDs issued for a submission are provided to the submitter in VCF format, as in <a href="files/ss_ids.vcf">this example</a>; please note the SS IDs listed there are mere examples and do not refer to real variants. Multi-allelic variants are split in multiple records, one per alternate allele. Variants are always reported in the forward strand.
                                                    </p>

                                                    <h5>Reference SNP (RS)</h5>
                                                    <p>
                                                        A Reference SNP (RS) identifies a variant class in a genomic location, which can be reported by one or more studies. RS IDs are preserved when remapped to a newer version of the reference sequence they were initially mapped against; this allows to track how they have evolved with the passing of time. This means that a RS ID can be associated with one or more combinations of:
                                                        <ul>
                                                            <li>Taxonomy (e.g. 9615 - dog)</li>
                                                            <li>Reference sequence (e.g. CanFam3.1)</li>
                                                            <li>Contig or chromosome</li>
                                                            <li>Start position</li>
                                                            <li>Variant class</li>
                                                        </ul>
                                                    </p>
                                                    <p>
                                                        The following table lists the mapping between variant classes supported by dbSNP and the EVA, and the Sequence Ontology terms used by the latter.
                                                    </p>

                                                    <table id="help-variant-classes" class="responsive-table hover tablesorter">
                                                        <thead>
                                                            <tr>
                                                                <th>dbSNP class</th>
                                                                <th>EVA class</th>
                                                                <th>Sequence Ontology term</th>
                                                            </tr>
                                                        </thead>
                                                        <tbody>
                                                            <tr>
                                                                <td>SNV</td>
                                                                <td>SNV</td>
                                                                <td><a href="http://www.sequenceontology.org/browser/current_release/term/SO:0001483">SO:0001483</a></td>
                                                            </tr>
                                                            <tr>
                                                                <td rowspan="3">DIV</td>
                                                                <td>deletion</td>
                                                                <td><a href="http://www.sequenceontology.org/browser/current_release/term/SO:0000159">SO:0000159</a></td>
                                                            </tr>
                                                            <tr>
                                                                <td>insertion</td>
                                                                <td><a href="http://www.sequenceontology.org/browser/current_release/term/SO:0000667">SO:0000667</a></td>
                                                            </tr>
                                                            <tr>
                                                                <td>indel</td>
                                                                <td><a href="http://www.sequenceontology.org/browser/current_release/term/SO:1000032">SO:1000032</a></td>
                                                            </tr>
                                                            <tr>
                                                                <td>HETEROZYGOUS</td>
                                                                <td></td>
                                                                <td></td>
                                                            </tr>
                                                            <tr>
                                                                <td>STR / microsatellite</td>
                                                                <td>tandem repeat</td>
                                                                <td><a href="http://www.sequenceontology.org/browser/current_release/term/SO:0000705">SO:0000705</a></td>
                                                            </tr>
                                                            <tr>
                                                                <td>NAMED</td>
                                                                <td>sequence alteration</td>
                                                                <td><a href="http://www.sequenceontology.org/browser/current_release/term/SO:0001059">SO:0001059</a></td>
                                                            </tr>
                                                            <tr>
                                                                <td>NO VARIATION</td>
                                                                <td>no sequence alteration</td>
                                                                <td><a href="http://www.sequenceontology.org/browser/current_release/term/SO:0002073">SO:0002073</a></td>
                                                            </tr>
                                                            <tr>
                                                                <td>MIXED</td>
                                                                <td></td>
                                                                <td></td>
                                                            </tr>
                                                            <tr>
                                                                <td>MNV</td>
                                                                <td>MNV</td>
                                                                <td><a href="http://www.sequenceontology.org/browser/current_release/term/SO:0002007">SO:0002007</a></td>
                                                            </tr>
                                                        </tbody>
                                                    </table>
                                                    <p>
                                                        Please note that due to the different granularities, dbSNP DIV variants will be assigned the appropriate EVA class when imported.<br />
                                                        HETEROZYGOUS and MIXED variants were initially defined but no longer supported in any of the non-human dbSNP databases, thus why the EVA hasnâ€™t defined a mapping.<br />
                                                        Variants of class NO VARIATION will no longer be accessioned by the EVA, but legacy identifiers will be preserved.
                                                    </p>

                                                    <h6>Reporting</h6>
                                                    <p>
                                                        The list of RS IDs issued for a species in a release will be provided in VCF format, as in <a href="files/rs_ids.vcf">this example</a>; multi-allelic variants are split in multiple records, one per alternate allele, even if several of them have the same variant class. Coordinates are mandatory in VCF format, so RefSNP's without them won't be included in the dump. Please note the example file is the actual RS release for the barrel medic species (Medicago truncatula).
                                                    </p>
                                                </div>
                                            </li>
                                        </div>
                                        <div id="retrieve-existing-dbsnp-ids">
                                            <li class="accordion-item is-active" data-accordion-item>
                                                <a class="accordion-title"><h5>Will it be possible to retrieve existing dbSNP identifiers from the EVA?</h5></a>
                                                <div class="accordion-content" data-tab-content>
                                                    <p>
                                                        Yes. The EVA is committed to the continuation of existing dbSNP identifiers for Submitted SNP (SS) and Reference SNP (RS).
                                                    </p>
                                                    <p>
                                                        The identifiers of those dbSNP variants that satisfy the EVA submission requirements will be retrievable via the
                                                        <a href="?Variant%20Browser&species=mmusculus_grcm38&selectFilter=snp&snp=rs585444580">EVA variant browser</a> and
                                                        <a href="?API">web services API</a>. If you want to learn more about EVA submission requirements, please click <a href="?Submit%20Data">here</a>.
                                                    </p>
                                                    <p>
                                                        dbSNP variants that don't satisfy these requirements <b>will still be imported</b> and searchable via a specific API and web view. We will work to make this experience as intuitive as possible, while keeping our commitment to only make high-quality variants part of the core EVA database.
                                                    </p>
                                                </div>
                                            </li>
                                        </div>
                                        <div id="why-variants-in-different-strand">
                                            <li class="accordion-item is-active" data-accordion-item>
                                                <a class="accordion-title"><h5>Why are some variants in a different strand than the one dbSNP reports?</h5></a>
                                                <div class="accordion-content" data-tab-content>
                                                    <p>
                                                        The EVA accepts variant submission in the forward strand only, whereas dbSNP is not so restrictive. In addition to this, dbSNP registers multiple orientations:
                                                    </p>
                                                    <p>
                                                        <ul>
                                                            <ul>
                                                                <li>Contig to chromosome</li>
                                                                <li>SNP to contig</li>
                                                                <li>SubSNP (ss) to RefSNP (rs)</li>
                                                            </ul>
                                                        </ul>
                                                    </p>
                                                    <p>
                                                        As a result, variants with a 'reverse' orientation will be displayed differently despite the underlying data being equivalent.
                                                    </p>
                                                </div>
                                            </li>
                                        </div>
                                        <div id="cant-find-dbsnp-ids-in-eva-website">
                                            <li class="accordion-item is-active" data-accordion-item>
                                                <a class="accordion-title"><h5>Why can't I find some of the dbSNP identifiers on the EVA website?</h5></a>
                                                <div class="accordion-content" data-tab-content>
                                                    <p>
                                                        We will gradually import the dbSNP data, and your species of interest may not have been yet processed. Please check our <a href="?dbSNP-Import-Progress">status report</a> on the transition process.
                                                    </p>
                                                    <p>
                                                        If a species has been fully imported and your variant of interest is not displayed in our browser, please communicate the issue via <a href="mailto:eva-helpdesk@ebi.ac.uk" target="_top">eva-helpdesk@ebi.ac.uk</a>
                                                    </p>
                                                </div>
                                            </li>
                                        </div>
                                        <div id="eva-builds">
                                            <li class="accordion-item is-active" data-accordion-item>
                                                <a class="accordion-title"><h5>Will there be EVA builds or releases like the dbSNP ones?</h5></a>
                                                <div class="accordion-content" data-tab-content>
                                                    <p>
                                                        Yes. The EVA will create RefSNP dumps in VCF format every 6 months, starting in 2019. These dumps shall contain the most basic information about each RefSNP:
                                                    </p>
                                                    <p>
                                                        <ul>
                                                            <ul>
                                                                <li>Genomic coordinates</li>
                                                                <li>Reference and alternate alleles in the forward strand</li>
                                                                <li>Identifiers of the studies that reported the variant</li>
                                                            </ul>
                                                        </ul>
                                                    </p>
                                                    <p>
                                                        The release files will only contain this information for the following reasons:
                                                        <ul>
                                                            <ul>
                                                                <li>Reduce their size</li>
                                                                <li>Distribute only information that will not get outdated, as new studies may come in very often</li>
                                                                <li>Avoid merging genotypes from multiple studies conducted with different technologies</li>
                                                            </ul>
                                                        </ul>
                                                    </p>
                                                    <p>
                                                        The list of RS IDs issued for a species in a release will be provided in VCF format; multi-allelic variants are split in multiple records, one per alternate allele, even if several of them have the same variant class. Coordinates are mandatory in VCF format, so RefSNP's that could not be mapped against an assembly will be reported separately. Variants with the unspecified allele &lt;*&gt; or breakend notation will be excluded from the VCF files. An example file containing the actual RS release for the barrel medic species (Medicago truncatula) can be found <a href="files/rs_ids.vcf">here</a>.
                                                    </p>
                                                    <p>
                                                        The details of study submissions, reported genotypes and frequencies, variant annotation, etc, can be retrieved from the EVA website through one of the following methods: FTP, VCF download from the Variant Browser or <a href="?API">REST API</a>.
                                                    </p>
                                                </div>
                                            </li>
                                        </div>
                                    </ul>
                                </div>
                            </ul>
                        </div>
                        <div class="tabs-panel is-active" id="citationPanel">
                            <ul class="padding-top-large padding-bottom-large">
                                <li><a href="#archive-data-reference">How to reference archived data?</a></li>
                                <li><a href="#citation-reference">How to cite EVA?</a></li>
                            </ul>
                            <ul class="accordion" data-accordion>
                                <div id="archive-data-reference">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>How to reference archived data?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <p>Archived data can be referenced using the project accession & associated URL
                                                e.g. The variant data for this study have been deposited in the European
                                                Variation Archive (EVA) at EMBL-EBI under accession number PRJEBxxxx
                                                (<a>https://www.ebi.ac.uk/eva/?eva-study=PRJEB*****</a>)</p>
                                        </div>
                                    </li>
                                </div>
                            </ul>
                            <ul class="accordion" data-accordion>
                                <div id="citation-reference">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>How to cite EVA?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <p>The EVA can be cited directly using the associated literature:</br></br><i>
                                                Cezard T, Cunningham F, Hunt SE, Koylass B, Kumar N, Saunders G, Shen A,
                                                Silva AF, Tsukanov K, Venkataraman S, Flicek P, Parkinson H, Keane TM.
                                                The European Variation Archive: a FAIR resource of genomic variation for
                                                all species. Nucleic Acids Res. 2021 Oct 28:gkab960.
                                                doi: 10.1093/nar/gkab960. PMID: 34718739.</i></p>
                                        </div>
                                    </li>
                                </div>
                            </ul>
                        </div>
                    </div>
                </div>
                <div id="studyView" class="hide-div">
                    <nav aria-label="You are here:" role="navigation">
                        <ul class="breadcrumbs">
                            <li><a href="?Home">EVA</a></li>
                            <li><a href="?Study Browser">Study Browser</a></li>
                            <li>
                                <span class="show-for-sr">Current: </span> Study View
                            </li>
                        </ul>
                    </nav>
                    <div id="studyViewContent"></div>
                </div>
                <div id="variantView" class="hide-div">
                    <nav aria-label="You are here:" role="navigation">
                        <ul class="breadcrumbs">
                            <li><a href="?Home">EVA</a></li>
                            <li><a href="?Variant Browser">Variant Browser</a></li>
                            <li>
                                <span class="show-for-sr">Current: </span> Variant View
                            </li>
                        </ul>
                    </nav>
                    <div data-sticky-container id = "navigation-strip">
                        <div class="sticky" data-sticky data-options="marginTop:2.65;" style="width:100%" data-top-anchor="summary-grid-div:top">
                                <div data-magellan class="megallan-div expanded button-group">
                                    <a id="summary-button" href="#summary-grid-div" class="button">Summary</a>
                                    <a id="consequence-types-button" href="#consequence-types-grid-div" class="button">Consequence Type</a>
                                    <a id="studies-button" href="#studies-grid-div" class="button">Files</a>
                                    <a id="genotypes-button" href="#genotypes-grid-div" class="button">Genotypes</a>
                                    <a id="population-stats-button" href="#population-stats-grid-div" class="button">Population Statistics</a>
                                </div>
                        </div>
                    </div>
                    <div class="columns medium-12 large-12">
                        <div id="summary-grid-div" class="padding-bottom-large" data-magellan-target="summary-grid-div"></div><br/>
                        <div id="summary-grid"></div>
                        <div id="consequence-types-grid-div"  data-magellan-target="consequence-types-grid-div"></div><br/><br/>
                        <div id="consequence-types-grid"></div>
                        <div id="studies-grid-div"  data-magellan-target="studies-grid-div"></div><br/>
                        <div id="studies-grid" style="margin-left:-20px;"></div>
                        <div id="genotypes-grid-div"  data-magellan-target="genotypes-grid-div"></div><br/>
                        <div id="genotypes-grid" style="margin-left:-20px;"></div>
                        <div id="population-stats-grid-div"  data-magellan-target="population-stats-grid-div"></div><br/>
                        <div id="population-stats-grid-view" style="margin-left:-20px;"></div>
                    </div>
                    <div id="variantViewContent"></div>
                </div>
                <div id="dbSNPImport" class="hide-div">
                    <nav aria-label="You are here:" role="navigation">
                        <ul class="breadcrumbs">
                            <li><a href="?Home">EVA</a></li>
                            <li>
                                <span class="show-for-sr">Current: </span> dbSNP Import Progress
                            </li>
                        </ul>
                    </nav>
                    <div id="dbSNPImportContent"></div>
                </div>
                <div id="rsRelease" class="hide-div">
                    <nav aria-label="You are here:" role="navigation">
                        <ul class="breadcrumbs">
                            <li><a href="?Home">EVA</a></li>
                            <li>
                                <span class="show-for-sr">Current: </span> Rs Release
                            </li>
                        </ul>
                    </nav>
                    <div id="rsReleaseContent"></div>
                </div>
        </div>
    </section>
</div>


<footer>

    <div id="elixir-banner" data-color="grey" data-name="This service" data-description="The European Variation Archive is an ELIXIR Deposition Database" data-more-information-link="https://www.elixir-europe.org/platforms/data/elixir-deposition-databases" data-use-basic-styles="false"></div>
    <div id="global-footer">

        <nav id="global-nav-expanded" class="row">
            <!-- Footer will be automatically inserted by footer.js -->
        </nav>

        <section id="ebi-footer-meta" class="row">
            <!-- Footer meta will be automatically inserted by footer.js -->
        </section>

    </div>

</footer>
</div> <!--! end of #wrapper -->


<!-- JavaScript -->
<!-- JavaScript at the bottom for fast page loading -->
<!-- build:script vendor -->
<script type="text/javascript" src="../vendor/ext-6.0.1/js/ext-all.js"></script>
<script type="text/javascript" src="../bower_components/underscore/underscore.js"></script>
<script type="text/javascript" src="../bower_components/backbone/backbone.js"></script>
<script type="text/javascript" src="../bower_components/jquery/dist/jquery.min.js"></script>
<!-- These dependencies must go AFTER jquery -->
<script type="text/javascript" src="../bower_components/linqjs/linq.min.js"></script>
<script type="text/javascript" src="../bower_components/vkbeautify/vkbeautify.js"></script>
<!-- /build -->
<script type="text/javascript" src="https://www.gstatic.com/charts/loader.js"></script>
<!-- build:script lib -->
<script type="text/javascript" src="../lib/jsorolla/build/1.1.9/genome-viewer/genome-viewer.js"></script>
<!-- /build -->
<!-- build:script eva-js -->
<script type="text/javascript" src="js/eva-manager.js"></script>
<script type="text/javascript" src="js/eva-annotation-model.js"></script>
<script type="text/javascript" src="js/eva-config.js"></script>
<script type="text/javascript" src="js/eva-menu.js"></script>
<script type="text/javascript" src="js/eva-adapter.js"></script>
<script type="text/javascript" src="js/variant-widget/eva-variant-widget-panel.js"></script>
<script type="text/javascript" src="js/study-browser/eva-study-browser-widget-panel.js"></script>
<script type="text/javascript" src="js/study-browser/eva-study-browser-filter-form-panel.js"></script>
<script type="text/javascript" src="js/study-browser/eva-study-browser-type-filter-form-panel.js"></script>
<script type="text/javascript" src="js/study-browser/eva-study-browser-text-search-form-panel.js"></script>
<script type="text/javascript" src="js/study-browser/eva-study-browser-grid.js"></script>
<script type="text/javascript" src="js/beacon-form/eva-beacon-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/eva-variant-widget.js"></script>
<script type="text/javascript" src="js/variant-widget/eva-variant-browser-grid.js"></script>
<script type="text/javascript" src="js/variant-widget/eva-variant-files-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/eva-variant-genotype-grid-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/eva-variant-population-stats-panel.js"></script>
<script type="text/javascript" src="js/views/eva-study-view.js"></script>
<script type="text/javascript" src="js/views/eva-variant-view.js"></script>
<script type="text/javascript" src="js/views/eva-multi-variant-view.js"></script>
<script type="text/javascript" src="js/statistics/eva-statistics.js"></script>
<script type="text/javascript" src="js/statistics/dgva-statistics.js"></script>
<script type="text/javascript" src="js/beacon-form/eva-beacon-form.js"></script>
<script type="text/javascript" src="js/beacon-form/eva-variant-search-form.js"></script>
<script type="text/javascript" src="js/variant-widget/filters/eva-form-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/filters/eva-species-filter-form-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/filters/eva-study-filter-form-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/filters/eva-position-filter-form-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/filters/eva-annotation-version-filter-form-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/filters/eva-population-frequency-filter-form-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/filters/eva-protein-substitutions-score-filter-form-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/filters/eva-consequence-type-filter-form-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/filters/eva-conservation-score-filter-form-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/eva-variant-annotation-panel.js"></script>
<script type="text/javascript" src="js/dbSNP-import/dbSNP-import-progress.js"></script>
<script type="text/javascript" src="js/rs-release/rs-release.js"></script>
<script type="text/javascript" src="js/eva.js"></script>
<script type="text/javascript" src="js/eva-google-analytics.js"></script>
<!-- /build -->
<!-- build:script ebi_framework -->
<script src="../lib/EBI-Framework/js/cookiebanner.js"></script>
<script src="../lib/EBI-Framework/js/script.js"></script>
<!--Footer overwrite-->
<script type="text/javascript" src="js/footer/footer.js"></script>
<!-- The Foundation theme JavaScript -->
<script src="../lib/EBI-Framework/libraries/foundation-6/js/foundation.js"></script>
<script src="../lib/EBI-Framework/libraries/foundation-6/js/foundation.magellan.js"></script>
<script src="../lib/EBI-Framework/js/foundationExtendEBI.js"></script>
<script src="../lib/EBI-Framework/libraries/tablesorter/dist/js/jquery.tablesorter.min.js"></script>
<!-- /build -->
<script type="text/JavaScript">$(document).foundation();</script>
<script type="text/JavaScript">$(document).foundationExtendEBI();</script>
<script defer="defer" src="https://ebi.emblstatic.net/web_guidelines/EBI-Framework/v1.3/js/elixirBanner.js"></script>

<script type="text/javascript">
    var eva;
    var start = function () {
        populateLatestReleaseInfo();

        // Bring in the import content.
        var homeDiv = document.querySelector('#home-page');
        var submitDiv = document.querySelector('#submit');
        var feedbackDiv = document.querySelector('#feedback');
        var studyViewDiv = document.querySelector('#studyView');
        var variantViewDiv = document.querySelector('#variantView');
        var apiDiv =  document.querySelector('#api');
        var helpDiv = document.querySelector('#help-section');
        var dbSNPImportDiv = document.querySelector('#dbSNPImport');
        var rsReleaseDiv = document.querySelector('#rsRelease');

        var tab = getUrlParameters("");
        var tabId =  decodeURI(tab.id);
        if(tabId == 'eva study'){
            var evastudyView = new EvaStudyView({
                target:'studyViewContent',
                type: 'eva',
                projectId:tab.value
            });

        }
        if(tabId == 'dgva study'){
            var dgvastudyView = new EvaStudyView({
                target:'studyViewContent',
                type: 'dgva',
                projectId:tab.value
            });
        }

        if(tabId == 'variant'){
            var annotationVersion = $.urlParam('annotationVersion');
            var accessionID = $.urlParam('accessionID');
            accessionID = accessionID ? accessionID.toLowerCase(): "";
            var requestedPage = $.urlParam('page');
            var isMultiLookup = accessionID ? (accessionID.split(",").length > 1): false;

            if (isMultiLookup) {
                var evaMultiVariantView = new EvaMultiVariantView({
                    target:'variantViewContent',
                    position:tab.value,
                    accessionID: accessionID,
                    annotationVersion:annotationVersion,
                    page: parseInt(requestedPage)
                });
                evaMultiVariantView.render();
            }
            else {
                var evaVariantView = new EvaVariantView({
                    target:'variantViewContent',
                    position:tab.value,
                    accessionID:accessionID,
                    annotationVersion:annotationVersion
                });
                evaVariantView.render();
            }
            $('.sticky').foundation('_calc', true);
            $(window).trigger('resize');
        }

        Ext.getBody().removeCls('x-body');
        var ulLocalNav = document.querySelector('.eva-primary-menu');

        eva = new Eva({
            target: 'content',
            targetMenu: ulLocalNav,
            width: $('#content').width(),
            homeDiv:homeDiv,
            submitDiv:submitDiv,
            feedbackDiv:feedbackDiv,
            studyView:studyViewDiv,
            variantView:variantViewDiv,
            apiDiv:apiDiv,
            helpDiv:helpDiv,
            dbSNPImportDiv:dbSNPImportDiv,
            rsReleaseDiv:rsReleaseDiv
        });


        if(tab.id){
            eva.draw(tabId);
        }else{
            eva.draw('Home');
        }

        window.onpopstate = function(event) {
            if (event.state) {
                location.reload();
            }
        }

    };

    function populateLatestReleaseInfo() {
        var latestReleaseVersion;
        EvaManager.get({
            host:EVA_RELEASE_HOST,
            version: EVA_VERSION,
            category: 'info',
            resource: 'latest',
            async: false,
            success: function (response) {
                try {
                    latestReleaseVersion = response;
                } catch (e) {
                    console.log(e);
                }
            }
        });

        var releaseLink = document.getElementById("release-ftp-link");
        releaseLink.href = latestReleaseVersion.releaseFtp;

        var releaseVersion = document.getElementById("release-version");
        releaseVersion.textContent = latestReleaseVersion.releaseVersion;
    }

    function isValidAccessionID (accessionID) {
        return /^(rs|ss)[0-9]+$/.test(accessionID);
    }

    function areValidAccessionIDs (accessionIDs) {
        return accessionIDs.every(function(accessionID) { return isValidAccessionID(accessionID.toLowerCase()); });
    }

    function areSameTypeAccessionIDs (accessionIDs) {
        return accessionIDs.every(function(x) {return x.toLowerCase().startsWith("ss");}) ||
               accessionIDs.every(function(x) {return x.toLowerCase().startsWith("rs");});
    }

    $("#snpSearch").submit(function(event) {
        /* stop form from submitting normally */
        event.preventDefault();

        var accessionID = document.getElementById("accession-search-box").value.trim();

        /* Some times more than one accession ID can be submitted. Ex: rs123, ss234 */
        accessionID = accessionID.replace(/(, | )+/, ',').replace(/\s/g, '')
        var accessionIDs = accessionID.split(',');

        if (!areValidAccessionIDs(accessionIDs)) {
            Ext.Msg.alert('Invalid Accession ID', 'Accession ID must have rs or ss followed by a number (ex: rs123 or ss567)');
            return;
        }

        if (!areSameTypeAccessionIDs(accessionIDs)) {
            Ext.Msg.alert('Multiple Accession ID types', 'Accession IDs must either be all RS IDs or all SS IDs');
            return;
        }

        window.open(window.location.origin + window.location.pathname + "?variant&accessionID=" + accessionID, "_blank");
    });

    function getUrlParameters(staticURL) {
        var data = [];

        var currLocation = (staticURL.length) ? staticURL : window.location.search;
        if (currLocation.indexOf("?") >= 0) {
            parArr = currLocation.split("?")[1].split("&");
            if (parArr) {
                for (var i = 0; i < parArr.length; i++) {
                    parr = parArr[i].split("=");
                    data.id = parr[0].replace(/-/g, " ");
                    data.value = parr[1];
                    return data;
                }
            }
        } else {
            return false
        }
    }

    $.urlParam = function(name){

        var results = new RegExp('[?&]' + name + '=([^&#]*)').exec(window.location.href);
        if (results==null){
            return null;
        }
        else{
            return results[1] || 0;
        }
    }

    //Start point
    window.addEventListener('load', function (e) {
        // all imports loaded
        Ext.QuickTips.init();
        Ext.onReady(start);
    });

</script>
</body>
</html>
