<!doctype html>
<!--
  ~
  ~ European Variation Archive (EVA) - Open-access database of all types of genetic
  ~ variation data from all species
  ~
  ~ Copyright 2014-2017 EMBL - European Bioinformatics Institute
  ~
  ~ Licensed under the Apache License, Version 2.0 (the "License");
  ~ you may not use this file except in compliance with the License.
  ~ You may obtain a copy of the License at
  ~
  ~    http://www.apache.org/licenses/LICENSE-2.0
  ~
  ~ Unless required by applicable law or agreed to in writing, software
  ~ distributed under the License is distributed on an "AS IS" BASIS,
  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  ~ See the License for the specific language governing permissions and
  ~ limitations under the License.
  ~
  -->

<html lang="en">
<head>
    <meta charset="utf-8">
    <title>The European Bioinformatics Institute &lt; EMBL-EBI</title>
    <meta name="description" content="EMBL-EBI">
    <meta name="keywords" content="bioinformatics, europe, institute, variation">
    <meta name="author" content="European Variation Archive">
    <meta name="HandheldFriendly" content="true" />
    <meta name="MobileOptimized" content="width" />
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <meta name="theme-color" content="#70BDBD"> <!-- Android Chrome mobile browser tab color -->

    <!-- If you link to any other sites frequently, consider optimising performance with a DNS prefetch -->
    <link rel="dns-prefetch" href="//embl.de" />
    <link rel="dns-prefetch" href="//www.ncbi.nlm.nih.gov" />

    <!-- If you have custom icon, replace these as appropriate.
         You can generate them at realfavicongenerator.net -->
    <link rel="icon" type="image/x-icon" href="../lib/EBI-Framework/images/logos/EMBL-EBI/favicons/favicon.ico" />
    <link rel="icon" type="image/png" href="../lib/EBI-Framework/images/logos/EMBL-EBI/favicons/favicon-32x32.png" />
    <link rel="icon" type="image/png" sizes="192×192" href="../lib/EBI-Framework/images/logos/EMBL-EBI/favicons/android-chrome-192x192.png" /> <!-- Android (192px) -->
    <link rel="apple-touch-icon-precomposed" sizes="114x114" href="../lib/EBI-Framework/images/logos/EMBL-EBI/favicons/apple-touch-icon-114x114.png"> <!-- For iPhone 4 Retina display (114px) -->
    <link rel="apple-touch-icon-precomposed" sizes="72x72" href="../lib/EBI-Framework/images/logos/EMBL-EBI/favicons/apple-touch-icon-72x72.png"> <!-- For iPad (72px) -->
    <link rel="apple-touch-icon-precomposed" sizes="144x144" href="../lib/EBI-Framework/images/logos/EMBL-EBI/favicons/apple-touch-icon-144x144.png"> <!-- For iPad retinat (144px) -->
    <link rel="apple-touch-icon-precomposed" href="../lib/EBI-Framework/images/logos/EMBL-EBI/favicons/apple-touch-icon-57x57.png"> <!-- For iPhone (57px) -->
    <link rel="mask-icon" href="../lib/EBI-Framework/images/logos/EMBL-EBI/favicons/safari-pinned-tab.svg" color="#ffffff"> <!-- Safari icon for pinned tab -->
    <meta name="msapplication-TileColor" content="#2b5797"> <!-- MS Icons -->
    <meta name="msapplication-TileImage" content="../lib/EBI-Framework/images/logos/EMBL-EBI/favicons/mstile-144x144.png">
    <!-- CSS: implied media=all -->
    <!-- CSS concatenated and minified via ant build script-->
    <!-- build:style ebi_framework -->
    <link rel="stylesheet" href="../lib/EBI-Framework/libraries/foundation-6/css/foundation.min.css" type="text/css" media="all">
    <link rel="stylesheet" href="../lib/EBI-Framework/css/ebi-global.css" type="text/css" media="all">
    <link rel="stylesheet" href="../lib/EBI-Framework/css/theme-embl-petrol.css" type="text/css" media="all">
    <!-- /build -->
    <link rel="stylesheet" href="https://www.ebi.ac.uk/web_guidelines/EBI-Icon-fonts/v1.1/fonts.css" type="text/css" media="all">
    <!-- build:style vendor -->
    <link type="text/css" rel="stylesheet" href="../vendor/ext-6.0.1/theme/theme-ebi-embl-all.css"/>
    <!-- /build -->
    <!-- build:style css -->
    <link rel="stylesheet" href="css/eva.css">
    <!-- /build -->
    <meta name="ebi:localmasthead-color" content="#0C6666">
    <meta name="ebi:localmasthead-image" content="img/EVA_homepage_banner_opt.jpg">
    <!-- end CSS-->
    <!-- All JavaScript at the bottom, except for Modernizr -->
    <!-- build:script modernizr -->
    <script src="../lib/EBI-Framework/libraries/modernizr/modernizr.custom.49274.js"></script>
    <!-- /build -->
</head>

<body class="level2 full-width">
<div id="skip-to">
    <ul>
        <li><a href="#content">Skip to main content</a></li>
        <li><a href="#local-nav">Skip to local navigation</a></li>
        <li><a href="#global-nav">Skip to EBI global navigation menu</a></li>
        <li><a href="#global-nav-expanded">Skip to expanded EBI global navigation menu (includes all sub-sections)</a></li>
    </ul>
</div>

<div data-sticky-container>
    <div id="local-masthead" data-sticky data-sticky-on="large" data-top-anchor="180" data-btm-anchor="300000">
        <header>
            <div id="global-masthead" class="clearfix">
                <a href="//www.ebi.ac.uk/" title="Go to the EMBL-EBI homepage"><span class="ebi-logo"></span></a>
                <nav>
                    <div class="row">
                        <ul id="global-nav" class="menu">
                            <!-- set active class as appropriate -->
                            <li id="home-mobile" class=""><a href="//www.ebi.ac.uk"></a></li>
                            <li id="home" class="active"><a href="//www.ebi.ac.uk"><i class="icon icon-generic" data-icon="H"></i> EMBL-EBI</a></li>
                            <li id="services"><a href="//www.ebi.ac.uk/services"><i class="icon icon-generic" data-icon="("></i> Services</a></li>
                            <li id="research"><a href="//www.ebi.ac.uk/research"><i class="icon icon-generic" data-icon=")"></i> Research</a></li>
                            <li id="training"><a href="//www.ebi.ac.uk/training"><i class="icon icon-generic" data-icon="t"></i> Training</a></li>
                            <li id="about"><a href="//www.ebi.ac.uk/about"><i class="icon icon-generic" data-icon="i"></i> About us</a></li>
                            <li id="search">
                                <a data-toggle="search-global-dropdown"><i class="icon icon-functional" data-icon="1"></i> <span class="show-for-small-only">Search</span></a>
                                <div id="search-global-dropdown" class="dropdown-pane" data-dropdown data-options="closeOnClick:true;">
                                    <form id="global-search" name="global-search" action="/ebisearch/search.ebi" method="GET">
                                        <fieldset>
                                            <div class="input-group">
                                                <input type="text" name="query" id="global-searchbox" class="input-group-field" placeholder="Search all of EMBL-EBI">
                                                <div class="input-group-button">
                                                    <input type="submit" name="submit" value="Search" class="button">
                                                    <input type="hidden" name="db" value="allebi" checked="checked">
                                                    <input type="hidden" name="requestFrom" value="global-masthead" checked="checked">
                                                </div>
                                            </div>
                                        </fieldset>
                                    </form>
                                </div>
                            </li>
                            <li class="float-right show-for-medium embl-selector">
                                <button class="button float-right" type="button" data-toggle="embl-dropdown">Hinxton</button>
                                <!-- The dropdown menu will be programatically added by script.js -->
                            </li>
                        </ul>
                    </div>
                </nav>
            </div>

            <div class="masthead row">
                <!-- local-title -->
                <div class="columns medium-12 " id="local-title">
                    <h1><img src="img/eva_logo.png" style="margin-left:-10px;"/><a href="?Home" title="Back to European Variation Archive homepage homepage">European Variation Archive</a></h1>
                </div>
                <!-- /local-title -->
                <!-- local-nav -->
                <nav>
                    <ul class="dropdown menu float-left eva-primary-menu" data-dropdown-menu data-description="navigational">
                        <li><a>Home</a></li>
                        <li><a>Submit Data</a></li>
                        <li><a>Study Browser</a></li>
                        <li><a>Variant Browser</a></li>
                        <li><a>Clinical Browser</a></li>
                        <li><a>GA4GH</a></li>
                        <li><a>API</a></li>
                        <li><a>dbSNP Import Progress</a></li>
                        <li><a>Help</a></li>
                        <li><a>Feedback</a></li>
                    </ul>
                </nav>
                <!-- /local-nav -->
            </div>
        </header>
    </div>
</div>


<div id="content" role="main" class="row">
    <section>
        <div id="main-content-area" class="columns">
                <div id="home-page"  class="hide-div">
                     <nav aria-label="You are here:" role="navigation">
                       <ul class="breadcrumbs">
                         <li><a href="?Home">EVA</a></li>
                         <li>
                           <span class="show-for-sr">Current: </span> Home
                         </li>
                       </ul>
                     </nav>
                    <div class="callout alert">
                        <p>Support for non-human variant data archival and accessioning is transitioning from dbSNP to EVA from September 2017. <a href="?Help#accession-Panel">[Read more]</a></p>
                    </div>
                    <div class="row">
                        <div class="small-12 medium-8 columns">
                            <h3><span class="icon icon-generic" data-icon="&"> </span>Overview</h3>
                            <div><p> The European Variation Archive is an open-access database of all types of genetic variation data from all species.</p></div>
                            <div><p> All users can <a href="?Study Browser "> download data </a> from any study, or <a href="?Submit Data"> submit their own data</a> to the archive. You can also query all variants in the EVA by study, gene, chromosomal location or dbSNP identifier using our <a href="?Variant Browser"> Variant Browser</a>. </p></div>
                            <div><p> We will be adding new features to the EVA on a regular basis, and welcome <a href="mailto:eva-helpdesk@ebi.ac.uk" target="_top">your comments and feedback</a>. </p></div>
                        </div>
                        <div class="small-12 medium-4 columns">
                            <h3><span class="icon icon-generic" data-icon="N"> </span>News</h3>
                            <div id="twitter-widget"></div>
                        </div>
                    </div>
                    <div class="padding-bottom-large"></div>
                    <div class="row">
                        <div class="columns"><h3><span class="icon icon-generic" data-icon="g"> </span>Statistics</h3></div>
                        <div class="small-12 medium-6 columns">
                            <div class="row">
                                <div class="small-10 medium-10 medium-offset-2">
                                    <h4>Short genetic variants studies (&lt;50bp)</h4>
                                </div>
                            </div>
                            <div id="eva-statistics"></div>
                        </div>
                        <div class="small-12 medium-6 columns">
                            <div class="row">
                                <div class="small-10 medium-10 medium-offset-2">
                                    <h4 class="padding-bottom">Structural variants studies (&gt;50bp)</h4>
                                </div>
                             </div>
                            <div id="dgva-statistics"></div>
                        </div>
                    </div>

                    <div class="row">
                        <div class="small-12 medium-12 columns">
                            <div><p style="font-size:11px;">This web application makes an intensive use of new web technologies and standards like HTML5. Please see <a href="?Help#collapseOne">FAQs</a> for further browser compatibility notes.</p></div>
                        </div>
                    </div>
                </div>

                <div  id="submit" class="hide-div">
                    <nav aria-label="You are here:" role="navigation">
                        <ul class="breadcrumbs">
                            <li><a href="?Home">EVA</a></li>
                            <li>
                                <span class="show-for-sr">Current: </span> Submit
                            </li>
                        </ul>
                    </nav>
                    <h2>Submit</h2>
                    <div class="padding-bottom-large">
                        <p>Please read our Data Requirements and the <a href="https://www.ebi.ac.uk/eva/?Submit-Data#submit-prepare" target="_blank">Key stages of submission</a> below.</p>
                        <p>Data submitted to the EVA is brokered to our collaborating databases at <a href="http://www.ncbi.nlm.nih.gov/" target="_blank">NCBI</a>, <a href="http://www.ncbi.nlm.nih.gov/SNP/" target="_blank">dbSNP</a> and <a href="http://www.ncbi.nlm.nih.gov/dbvar" target="_blank">dbVar</a>. It is therefore unnecessary to submit data to multiple resources.</p>
                        <p>All data valid for EVA submission shall be made available via the <a href="?Study Browser">Study Browser</a> and will be browsable using both the <a href="?Variant Browser">Variant Browser</a> and the <a href="?API">EVA API</a>. <a href="http://www.ensembl.org/info/docs/tools/vep/index.html" target="_blank">Variant Effect Predictor</a> annotations shall be available for variants mapped to genome assemblies that are known to <a href="http://www.ensembl.org/index.html" target="_blank">Ensembl</a>.</p>
                    </div>

                    <h3 class="padding-bottom-large">Data requirements</h3>
                    <div class="padding-bottom-large">
                        <p>EVA accepts <b>all</b> types of short sequence variation &le;50bp, in <b>any species</b> providing the following requirements are met:</p>
                        <ol>
                            <li>Data is described in valid VCF file(s).<br/> This can be tested prior to submission using the EVA VCF Validation Suite found <a href="https://github.com/EBIvariation/vcf-validator" target="_blank">here</a>. For help with converting variation data to VCF, please see our <a href="https://www.ebi.ac.uk/eva/?Help#what-happens-to-submitted-data" target="_blank">help</a> pages.</li>
                            <li>Data includes:
                                <ol type="i">
                                    <li><b>sample genotypes</b> and/or</li>
                                    <li><b>allele frequencies</b></li>
                                </ol>
                            </li>
                            <li>
                                The reference sequence used is <a href="http://www.insdc.org/" target="_blank">INSDC</a> registered, or will be at point of submission. A “reference” can be any of the following, but not restricted to:
                                <ul>
                                    <li>Assembly, e.g. <a href="https://www.ebi.ac.uk/ena/data/view/GCA_000002285.2" target="_blank">GCA_000002285.2</a></li>
                                    <li>Transcriptome/Transcript, e.g. <a href="https://www.ebi.ac.uk/ena/data/view/GCJV01000000" target="_blank">GCJV01000000</a>, <a href="https://www.ebi.ac.uk/ena/data/view/KY286086" target="_blank">KY286086</a></li>
                                    <li>Gene sequence, e.g. <a href="https://www.ebi.ac.uk/ena/data/view/X76482" target="_blank">X76482</a></li>
                                </ul>
                                <b><u>PLEASE NOTE:</u></b> Sequence identifiers in VCF must match those in the reference FASTA file.
                            </li>
                        </ol>
                        <p><a href="https://www.ebi.ac.uk/eva/?Help#variant-accession-administred-by-eva" target="_blank">Variant accessions</a> (ss# and rs#) and <a href="https://www.ebi.ac.uk/eva/?Help#accessions-administred-by-eva" target="_blank">study accessions</a> will only be provided for data which satisfies all 3 data requirements.</p>
                        <p>To find out if your data is suitable, see <a href="https://www.ebi.ac.uk/eva/?Help#is-my-data-suitable" target="_blank">here</a>.</p>
                        <p>Variation data <b>not</b> accepted by EVA can be submitted to one of the following resources:
                        <ul>
                            <li>Submit large-scale structural variations &gt;50bp to <a href="http://www.ebi.ac.uk/dgva" target="_blank">DGVa</a>.</li>
                            <li>Submit variations with clinical assertions to <a href="https://www.ncbi.nlm.nih.gov/clinvar/" target="_blank">ClinVar</a>.</li>
                            <li>Submit variations with sensitive clinical data to <a href="https://www.ebi.ac.uk/ega/home" target="_blank">EGA</a>.</li>
                        </ul>
                        </p>
                    <p><u>Consent requirements for Human data</u>: It is the submitter’s responsibility to ensure that the data submitted to EVA is compliant with our open access policy, and if consent was gathered for any individual genotype data then we will require a <a href="https://docs.google.com/document/d/1UaRmimAe919IZYIH55mDXMe-4sHsRVqaP4moqx2IYE4/edit" target="_blank">consent statement</a> to be completed prior to submission.</p>

                    </div>

                    <h3 class="padding-bottom-large">Key stages of EVA submissions</h3>

                    <h4 id="submit-prepare"><span class="icon icon-generic" data-icon="'" aria-hidden="true"></span> Prepare</h4>
                    <div>
                        <ul>
                            <li>Prepare <a href="https://www.ebi.ac.uk/eva/?Help#is-my-data-suitable" target="_blank">valid VCF file</a>(s), which can be validated prior to submission using the <a href="https://github.com/EBIvariation/vcf-validator" target="_blank">EVA VCF validation suite</a>.</li>
                            <li>Complete a <a href="https://www.ebi.ac.uk/eva/files/EVA_Submission_template.V1.1.2.xlsx" target="_blank">metadata template</a> describing the samples and analyses in your study. Please provide as much metadata as possible since this information is extremely useful for downstream analysis and is directly related to the frequency at which datasets archived at EVA are reused. An example of a completed metadata template is available <a href="https://www.ebi.ac.uk/eva/files/EVA_Submission_template.V1.1.1_mockup.xlsx" target="_blank">here</a>.</li>
                        </ul>
                        <p>Please also note that the template requires the submitter to fill in some personal data, which will be used as described in our <a href="https://www.ebi.ac.uk/data-protection/privacy-notice/eva-submission-service" target="_blank">privacy notice</a>.</p>
                    </div>

                    <h4 id="submit-contact"><span class="icon icon-generic" data-icon="C" aria-hidden="true"></span> Contact</h4>
                    <div>
                        <p>Contact  <a href="mailto:eva-helpdesk@ebi.ac.uk" target="_top">eva-helpdesk@ebi.ac.uk</a> in order to provide details of your submission. You shall receive a custom private FTP location for you to deposit files.</p>
                    </div>

                    <h4 id="submit-submit"><span class="icon icon-functional" data-icon="_"></span> Submit</h4>
                    <div>
                        <p>Upload your VCF file(s), associated data file(s) and EVA metadata template to your private EVA FTP location.</p>
                    </div>

                    <h4 id="submit-receive"><span class="icon icon-functional" data-icon="k" aria-hidden="true"></span> Receive</h4>
                    <div class="padding-bottom-large">
                      <p>The EVA aims to process submissions within two business days. Accession numbers shall be sent via email to the submitter upon successful archival of the deposited data.</p>
                    </div>

                    <h3 class="padding-bottom-large">Clinically relevant genetic variation data</h3>
                    <div>
                        <p>The EVA strongly recommends submission of clinically relevant genetic variant data, i.e. data that relates genetic variation(s) with clinical significance values (e.g. pathogenic, benign, etc.), to the ClinVar resource at <a href="https://www.ncbi.nlm.nih.gov/clinvar/" target="_blank">NCBI</a>.</p>
                        <p>Submitters unsure of the most relevant resource to archive their genetic variation data are encouraged to first contact <a href="mailto:eva-helpdesk@ebi.ac.uk?subject=Clinical submission to EVA" target="_top">eva-helpdesk@ebi.ac.uk</a> to discuss their situation.</p>
                    </div>
                </div>

                <div id="feedback" class="hide-div">
                    <nav aria-label="You are here:" role="navigation">
                        <ul class="breadcrumbs">
                            <li><a href="?Home">EVA</a></li>
                            <li>
                                <span class="show-for-sr">Current: </span> Feedback
                            </li>
                        </ul>
                    </nav>
                    <h2>Feedback</h2>
                    <div>
                        <p>If you have any questions related to the European Variation Archive resource, please <a href="mailto:eva-helpdesk@ebi.ac.uk">contact us</a>.</p>
                        <p>Follow us on Twitter using <a href="https://twitter.com/EBIvariation" target="_blank">@EBIvariation</a></p>
                    </div>
                    <div id="feedback-form">
                        <div class="row">
                            <div class="medium-12 columns feedback-form">
                                <iframe src="https://www.surveymonkey.co.uk/r/eva-feedback" frameborder="0" width="100%"  height="1490"></iframe>
                            </div>
                        </div>
                    </div>
                </div>
                <div id="api" class="hide-div">
                    <nav aria-label="You are here:" role="navigation">
                        <ul class="breadcrumbs">
                            <li><a href="?Home">EVA</a></li>
                            <li>
                                <span class="show-for-sr">Current: </span> API
                            </li>
                        </ul>
                    </nav>
                    <h2>API</h2>
                    <div>
                        <h3>The general structure of a EVA REST web service URL is:</h3><p><a href="#">http://www.ebi.ac.uk/eva/webservices/rest/{version}/{category}/IDs/{resource}?{filters}</a></p>
                        <h3>Where:</h3>
                        <p>
                            * <span class="api-span">version</span>: indicates the version of the API, this defines the available filters and JSON schema to be returned. Currently there is only version 'v1'.
                            <br/> * <span class="api-span">category</span>: this defines what objects we want to query. Currently there are five different categories: variants, segments, genes, files and studies.
                            <br/> * <span class="api-span">resource</span>: specifies the resource to be returned, therefore the JSON data model.
                            <br/> * <span class="api-span">filters</span>: each specific endpoint allows different filters.
                        </p>
                        <div>
                            <p>REST web services have been implemented using GET protocol since only queries are allowed so far. Several IDs can be concatenated using comma as separator.<br/>For more detailed information about the API and filters you can visit the <a href="https://github.com/ebivariation/eva/wiki" target="_blank">project wiki</a> and <a href="https://www.ebi.ac.uk/eva/webservices/rest/swagger-ui.html" target="_blank">Swagger documentation</a>.</p>
                        </div>
                        <div>
                            <h3>Some example of queries include:</h3>
                            <p>* To fetch all the variants in a segment region:
                                <br/> <a href="http://www.ebi.ac.uk/eva/webservices/rest/v1/segments/11:128446-128446/variants?species=hsapiens_grch37" target="_blank">http://www.ebi.ac.uk/eva/webservices/rest/v1/segments/11:128446-128446/variants?species=hsapiens_grch37</a></p>

                            <p><br/> * To fetch all the info of a variant:
                                <br/> <a href="http://www.ebi.ac.uk/eva/webservices/rest/v1/variants/rs666/info?species=hsapiens_grch37" target="_blank">http://www.ebi.ac.uk/eva/webservices/rest/v1/variants/rs666/info?species=hsapiens_grch37</a></p>

                            <p><br/> * To fetch all the Short Genetics Variations studies:
                                <br/> <a href="http://www.ebi.ac.uk/eva/webservices/rest/v1/meta/studies/all" target="_blank">http://www.ebi.ac.uk/eva/webservices/rest/v1/meta/studies/all</a></p>

                            <p><br/> * To fetch all the Structural Variations studies:
                                <br/> <a href="http://www.ebi.ac.uk/eva/webservices/rest/v1/meta/studies/all?structural=true" target="_blank">http://www.ebi.ac.uk/eva/webservices/rest/v1/meta/studies/all?structural=true</a></p>

                            <p><br/> * To fetch all  info of a study:
                                <br/> <a href="http://www.ebi.ac.uk/eva/webservices/rest/v1/studies/PRJEB4019/summary" target="_blank">http://www.ebi.ac.uk/eva/webservices/rest/v1/studies/PRJEB4019/summary</a></p>

                            <p><br/> * To fetch all file information of a study:
                                <br/> <a href="http://www.ebi.ac.uk/eva/webservices/rest/v1/studies/PRJEB4019/files?species=hsapiens_grch37" target="_blank">http://www.ebi.ac.uk/eva/webservices/rest/v1/studies/PRJEB4019/files?species=hsapiens_grch37</a></p>
                        </div>
                    </div>
                </div>
                <div id="help-section" class="hide-div">
                    <nav aria-label="You are here:" role="navigation">
                        <ul class="breadcrumbs">
                            <li><a href="?Home">EVA</a></li>
                            <li>
                                <span class="show-for-sr">Current: </span> Help
                            </li>
                        </ul>
                    </nav>
                    <h2>Help</h2>
                    <ul class="tabs faq-tabs small-block-grid-4" data-tabs id="faq-tabs">
                        <li class="tabs-title is-active"><a href="#generalPanel" aria-selected="true">General</a></li>
                        <li class="tabs-title"><a href="#dataAccessPanel">Data Access</a></li>
                        <li class="tabs-title"><a href="#submissionPanel">Submission</a></li>
                        <li class="tabs-title"><a href="#accessionPanel">Accessions</a></li>
                    </ul>
                    <div class="tabs-content faq-tabs-content" data-tabs-content="faq-tabs">
                        <div class="tabs-panel is-active" id="generalPanel">
                            <ul class="padding-top-large padding-bottom-large">
                                <li><a href="#what-is-eva">What is the European Variation Archive (EVA)?</a></li>
                                <li><a href="#eva-normalization-variant-processing-steps">What are the EVA normalisation and variant processing steps?</a></li>
                                <li><a href="#eva-colaborates-with">With whom does the EVA collaborate?</a></li>
                                <li><a href="#how-to-follow-eva-development">How can I follow the development of the EVA?</a></li>
                            </ul>
                            <ul class="accordion" data-accordion>
                                <div id="what-is-eva">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>What is the European Variation Archive (EVA)?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <p>The European Variation Archive (EVA) is EMBL-EBI's open-access genetic variation archive. The EVA accepts submission of all types of genetic variants, ranging from single nucleotide polymorphisms to large structural variants, observed in germline or somatic sources, from any eukaryotic organism. The EVA permits access to these data at two distinct levels:</p>
                                            <p> i) The raw variant data as was submitted to the EVA, via the <a href="http://www.ebi.ac.uk/eva/?Study%20Browser&browserType=sgv" target="_blank">EVA Study Browser</a></p>
                                            <p> ii) The normalised and processed variant data, via the <a href="http://www.ebi.ac.uk/eva/?Variant%20Browser" target="_blank">EVA Variant Browser</a> and <a href="http://www.ebi.ac.uk/eva/?API" target="_blank">EVA API</a></p>
                                        </div>
                                    </li>
                                </div>
                                <div id="eva-normalization-variant-processing-steps">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>What are the EVA normalisation and variant processing steps? </h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <div>
                                                <img class="faq-images" src="img/EVA_Variant_Level_Normalisation.svg" alt="EVA Variant Level Processing: Submitted data from the EVA Study Browser -> Variants are merged, normalized and annotated for functional consequences and statistical values
                                        -> Processed variants are brokered to dbSNP at NCBI and resulting 'ss' and 'rs' accessions are ingested by EVA -> Data are exposed as JSON objects either via the EVA website GUI or API"><br /><br />
                                                <h5>Normalisation</h5>
                                                <p>
                                                    Variants submitted to the EVA have been determined by a number of different algorithms and software packages. As a result, the VCF files generated by these differing methodologies describe variants in a number of different ways. The primary processing step of the EVA is to normalise variant representation following two basic rules:
                                                    <ol>
                                                        <li>Each variant is shifted to be left-aligned</li>
                                                        <li>The Start and End positions represent exactly the range where the variation occurs (which could, in the case of insertions, result in the reference allele being recorded as 'empty')</li>
                                                    </ol>
                                                </p>
                                                <p>Examples of our variant normalisation process can be seen <a href="https://docs.google.com/presentation/d/1WqSiT5AEEQF9jdIewdYIp-I0G5ozkFP3IikfCJZO1dc/edit#slide=id.ge1548f905_0_592" target="_blank">here</a></p>

                                            </div>
                                            <div>
                                                <h5>Annotation</h5>
                                                <p>
                                                    Once variants have been normalised, the EVA uses the <a href="http://www.ensembl.org/info/docs/tools/vep/" target="_blank">Variant Effect Predictor (VEP) </a>  of Ensembl to annotate variant consequences.
                                                    The variant consequences are described using Sequence Ontology terms and both the VEP version and Ensembl gene build used are described via the "i" help bubbles on the <a href="http://www.ebi.ac.uk/eva/?Variant%20Browser" target="_blank">EVA Variant Browser</a>.
                                                    <br /><br /> N.B. Variants that have been mapped to a reference genome sequence that is not supported by Ensembl are not annotated.
                                                </p>
                                            </div>
                                            <div>
                                                <h5>Statistical calculations</h5>
                                                <p>The EVA adopts the classical definition of allele frequency (AF): 'a measure of the relative frequency of an allele at a genetic locus in a given population'.
                                                    The AF value(s) stored by the EVA for each variant is (are) study specific - i.e. the same variant reported in two distinct studies shall be given two allele frequencies, one for each study. There are two methodologies by which the EVA is able to determine allele frequency values, dependent on the datatype of the study in question:</p>
                                                <p><span style="font-size:14px;font-weight:bold;">Variants associated with genotypes:</span></p>
                                                <p>
                                                    For variants associated with genotypes, the EVA determines the AF values via the calculation:
                                                    <br /><br />AF = (number of alternate allele observations (AC)) / (number of observations (AN))
                                                    <br /><br />The result of this calculation allows the EVA to also store the minor allele frequency (MAF) for each variant (defined as the minimum of the reference or alternative allele frequency) and the MAF allele (the allele associated with the MAF).
                                                </p>
                                                <p><span style="font-size:14px;font-weight:bold;">Variants not associated with genotypes:</span></p>
                                                <p>
                                                    For variants that are not associated with genotypes, the EVA is dependent on the AF value(s) estimated from the primary data and provided in the submitted VCF files(s).
                                                    AF values that are specifically provided in the submitted aggregated VCF file(s) are directly stored. In cases where no AF is provided then the EVA uses the AC and AN values in the submitted aggregated VCF file(s) to calculate AF value(s) via the calculation:
                                                    <br /><br />AF = AC / AN
                                                </p>
                                                <p><span style="font-size:14px;font-weight:bold;">Population / sample cohort allele frequency values:</span></p>
                                                <p>
                                                    The EVA accepts submission of <a href="http://pngu.mgh.harvard.edu/~purcell/plink/data.shtml#ped" target="_blank">pedigree </a> files, or structured samples (using "derived_from" and/or "subject" layers), to define populations and cohorts within studies.
                                                    In cases where such information is associated with variants that have genotypes then the EVA calculates intra-study population/cohort specific AF values via the method described above, with the caveat that the (total number of populations/cohorts):(total number of samples) ratio must be less than 1:10.
                                                    For studies that do not contain genotypes but instead provide intra-study population/cohort AF values in the submitted aggregated VCF file(s), or AC and AN values, then these are directly stored, or calculated by the EVA using the method described above, again with the caveat that a ratio of 1:10 (total number of populations/cohorts):(total number of samples) must not be exceeded.

                                                    <br /><br />*NB: there are a low number of variants for which the EVA is unable to determine any allele frequency value(s) as the submitted VCF file(s) contain neither genotypes nor AF or AC and AN values. The EVA discourages submission of variants that cannot be associated with an AF.
                                                </p>
                                            </div>

                                        </div>
                                    </li>
                                </div>
                                <div id="eva-colaborates-with">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>With whom does the EVA collaborate?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <div><img class="faq-images" src="img/EVA_Related_Projects.png" alt="Collaborators:GEUVADIS European Exome Variant Server and Príncipe Felipe Research Center. Past Collaborators: The Genomics and Bioinformatics Platform of Andalusia"><br/><br/><br/></div>
                                            <h5> The EVA & GEUVADIS European Exome Variant Server </h5>
                                            <div>
                                                <p>
                                                    The EVA & GEUVADIS European Exome Variant Server (<a href="http://www.ncbi.nlm.nih.gov/pubmed/24037378" target="_blank">GEEVS;</a>) work in collaboration to coordinate common data formats for data exchange.
                                                    As part of this collaboration, we fully endorse the variant calling protocol detailed on the <a href="http://geevs.crg.eu/home" target="_blank">GEEVS</a> website as adherence to this protocol for variant calling permits direct comparison and/or aggregation of results from different datasets.
                                                </p>
                                            </div>
                                            <h5>The EVA & Príncipe Felipe Research Center</h5>
                                            <p>Some of the technical and analytical features of the EVA were developed in collaboration with the department of Computational Genomics led by Joaquin Dopazo at the Principe Felipe Research Centre <a href="http://bioinfo.cipf.es/" target="_blank">Computational Genomics Department (CIPF)</a>.</p>
                                            <h4 style="color:#996a44;">Past Collaborations</h4>
                                            <h5>EVA & The Genomics and Bioinformatics Platform of Andalusia</h5>
                                            <div><p>Early development of the EVA was carried out in collaboration with the Bioinformatics Department at the <a href="http://www.gbpa.es" target="_blank">Genomics and Bioinformatics Platform of Andalusia</a>.</p></div>
                                        </div>
                                    </li>
                                </div>
                                <div id="how-to-follow-eva-development">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>How can I follow the development of the EVA?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <h5>The following are our GitHub repositories:</h5>
                                            <div>
                                                <ul style="list-style-type:disc">
                                                    <li>
                                                        <p>
                                                            The <a href="https://github.com/EBIvariation/vcf-validator">EVA VCF validator</a> checks that a file is compliant with the VCF specification.
                                                            It includes and expands the validations supported by the vcftools suite. It supports versions 4.1, 4.2 and 4.3 of the specification.
                                                        </p>
                                                    </li>
                                                    <li>
                                                        <p>
                                                            The <a href="https://github.com/EBIvariation/eva-pipeline">EVA pipeline</a> processes VCF files, stores the variation data in a database and post-processes it, in a way that can be later consumed via web services.
                                                        </p>
                                                    </li>
                                                    <li>
                                                        <p>
                                                            The <a href="https://github.com/EBIvariation/eva-ws"> EVA web services</a> serve the data generated and stored by the EVA pipeline. They follow the REST paradigm and can be consumed by any external application.
                                                        </p>
                                                    </li>
                                                    <li>
                                                        <p>
                                                            The <a href="https://github.com/EBIvariation/eva-web">EVA website</a> displays the data served by the EVA REST web services API in a user-friendly way.
                                                        </p>
                                                    </li>
                                                </ul>
                                            </div>
                                            <h5>Acknowledgement</h5>
                                            <p>We would like to acknowledge the following software support.</p>
                                            <div>
                                                <a href="https://github.com/" target="_blank" style="border-bottom-style: none;">
                                                    <img src="img/github.png" alt="GitHub" class="acknowledgement-logo">
                                                </a>
                                                <a href="http://www.ej-technologies.com/products/jprofiler/overview.html" target="_blank" style="border-bottom-style: none;">
                                                    <img src="img/jprofiler_large.png" alt="JProfiler" class="acknowledgement-logo">
                                                </a>
                                                <a href="http://www.jetbrains.com/idea/" target="_blank" style="border-bottom-style: none;">
                                                    <img src="img/intelliJ.png" alt="IntelliJ" class="acknowledgement-logo">
                                                </a>
                                            </div>
                                        </div>
                                    </li>
                                </div>
                            </ul>
                        </div>
                        <div class="tabs-panel is-active" id="dataAccessPanel">
                            <ul class="padding-top-large padding-bottom-large">
                                <li><a href="#how-to-consume-variant-data">How can I consume variant data from the EVA?</a></li>
                                <li><a href="#supported-browsers">Which browsers does the EVA website support?</a></li>
                                <li><a href="#how-to-search-existing-dbsnp">How can I search EVA with existing dbSNP accessions?</a></li>
                                <li><a href="#can-download-accesioned-variants">Can I download accessioned variants from the EVA?</a></li>
                                <li><a href="#how-to-track-variants">How can I track variants that have been generated, merged and/or deprecated over time?</a></li>
                                <li><a href="#data-shown-clinical-browser">What data is shown in the EVA Clinical Browser?</a></li>
                            </ul>
                            <ul class="accordion" data-accordion>
                                <div id="how-to-consume-variant-data">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>How can I consume variant data from the  EVA?</h5></a>

                                        <div class="accordion-content" data-tab-content>
                                            <p>
                                                Links are shown on the <a href="?Study%20Browser&browserType=sgv">EVA Study Browser</a>  to both the raw submitted files and the 'EVA browsable files'.
                                                Not all submitted VCF files for a study are browsable due to overlapping information. For example, we have studies where the submitted VCF files contain the sample and variant data split by population, but also merged together. It would be redundant for the EVA to load all of these data to our variant warehouse. Please note that due to processing time required not all studies have 'EVA browsable files' just yet.
                                            </p>
                                            <p> Additionally users can download query results from the <a href="?Variant%20Browser&species=hsapiens_grch37&selectFilter=region&studies=PRJEB4019%2CPRJEB6930%2CPRJEB5439%2CPRJEB5829%2CPRJEB8652%2CPRJEB8650%2CPRJEB8639%2CPRJEB6042%2CPRJX00001%2CPRJEB8705%2CPRJEB8661%2CPRJEB7895%2CPRJEB7217%2CPRJEB7218%2CPRJEB6041&region=1%3A3000000-3100000">EVA variant browser</a> directly and all of our loaded data is available programmatically via our <a href="?API">API</a>.</p>
                                        </div>
                                    </li>
                                </div>
                                <div id="supported-browsers">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>Which browsers does the EVA website support?</h5></a>

                                        <div class="accordion-content" data-tab-content>
                                            <p>
                                                The EVA website employs HTML5 technologies and standards. Chrome (version 18 or higher), Firefox (version 12 or higher), Safari (version 6 or higher) , Opera (version 12 or higher) and Internet Explorer (version 10 or higher) are fully supported, however older versions of these browsers may give rise to errors.<br />
                                                The EVA website also supports mobile versions of web browsers, with limited functionality. Please report all errors to the <a href="mailto:eva-helpdesk@ebi.ac.uk" target="_top">EVA helpdesk</a>.
                                            </p>
                                        </div>
                                    </li>
                                </div>
                                <div id="how-to-search-existing-dbsnp">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>How can I search EVA with existing dbSNP identifiers?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <p>
                                                The EVA offers users the ability to search both the <a href="?Variant%20Browser&species=btaurus_umd31&selectFilter=region&studies=PRJEB6119%2CPRJEB7061%2CPRJEB14878&region=1%3A3000000-3100000">EVA Variant Browser</a> and
                                                <a href="?API">API</a> using existing dbSNP 'SS' and 'RS' identifiers, as well as those that are newly generated by the EVA.
                                            </p>
                                        </div>
                                    </li>
                                </div>
                                <div id="can-download-accesioned-variants">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>Can I download accessioned variants from the EVA?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <p>
                                                Yes. After an rs build the EVA will provide a VCF dump for each species/assembly combination.
                                                These files shall be in the same style as the current VCFs that dbSNP generates (i.e. contain the rs ID positional information, but not the genotypic nor annotation data).
                                                Furthermore we are developing a 'VCF Dumper' tool which shall allow users to generate a custom VCF file based on filtering options available at the
                                                <a href="?Variant%20Browser&species=btaurus_umd31&selectFilter=region&studies=PRJEB6119%2CPRJEB7061%2CPRJEB14878&region=1%3A3000000-3100000">EVA Variant Browser</a>.
                                            </p>
                                        </div>
                                    </li>
                                </div>
                                <div id="how-to-track-variants">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>How can I track variants that have been generated, merged and/or deprecated over time?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <p>
                                                In addition to outputting a VCF file for each rs build the EVA shall also generate a file that will allow users to track changes to variant numbers over time.
                                                These files shall include details of all newly generated identifiers for each rs build, a list of those variants that have been deprecated as well as information on any variants that have been merged together.
                                                The EVA shall follow the same rules as dbSNP for generating, deprecating and merging variant identifiers.
                                            </p>
                                        </div>
                                    </li>
                                </div>
                                <div id="data-shown-clinical-browser">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>What data is shown in the EVA Clinical Browser?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <p>
                                                The EVA Clinical Browser displays variant data imported from the NCBI resource <a href="http://www.ncbi.nlm.nih.gov/clinvar/" target="_blank">ClinVar</a>, where each variant is associated with both a phenotype and a clinical significance assigned using the guidlines from the <a href="http://www.ncbi.nlm.nih.gov/pubmed/18414213" target="_blank">American College of Medical Genetics and Genomics</a>.
                                                Importantly, the variants shown via the EVA Clinical Browser have been annotated against the human GRCh37 genome using both the VEP and GENCODE Basic gene build of Ensembl version 78.
                                            </p>
                                        </div>
                                    </li>
                                </div>
                            </ul>
                        </div>
                        <div class="tabs-panel is-active" id="submissionPanel">
                            <ul class="padding-top-large padding-bottom-large">
                                <li><a href="#why-submit-data-to-eva">Why submit data to the EVA??</a></li>
                                <li><a href="#is-my-data-suitable">Is my data suitable for submission to the EVA?</a></li>
                                <li><a href="#what-happens-to-submitted-data">What happens to my data once submitted?</a></li>
                                <li><a href="#accept-submission-same-variation-file-format">Do EVA and dbSNP accept submission of the same genetic variation file format?</a></li>
                                <li><a href="#do-eva-and-dbsnp-collect-metadata-same-way">Do EVA and dbSNP collect metadata in the same way?</a></li>
                                <li><a href="#submitted-data-held-privately">Can my submitted data be held privately at the EVA?</a></li>
                            </ul>
                            <ul class="accordion" data-accordion>
                                <div id="why-submit-data-to-eva">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>Why submit data to the EVA?</h5></a>

                                        <div class="accordion-content" data-tab-content>
                                            <p>The EVA provides to the community a completely free, secure and permanent solution to data sharing. Each project, VCF file and sample that is submitted to the EVA is assigned a unique identifier that is accessible in perpetuity and is therefore able to be referenced in publication, for example.
                                                The <a href="mailto:eva-helpdesk@ebi.ac.uk" target="_top">EVA helpdesk</a> provides support to submitters, and users, to ensure accurate represention and proper integration of the submitted data with other EMBL-EBI resources such as the <a href="https://www.ebi.ac.uk/ega/" target="_blank">EGA</a> and <a href="http://www.ensembl.org/index.html" target="_blank">Ensembl</a>. A final advantage of submitting to the EVA is that variants are brokered to the National Centre for Biotechnology Information (<a href="http://www.ncbi.nlm.nih.gov/" target="_blank">NCBI</a>) on the submitter's behalf, negating the requirement for independent submissions.</p>
                                        </div>
                                    </li>
                                </div>
                                <div id="is-my-data-suitable">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>Is my data suitable for submission to the EVA?</h5></a>

                                        <div class="accordion-content" data-tab-content>
                                            <p>The most important consideration is that all data archived at the EVA is open access. As such, there are no restrictions as to who can access the data or how such data is reused. It is the submitter's responsibility to ensure that the data archived at the EVA complies to this open access policy.</p>
                                            <p>
                                                Genetic variants submitted to the EVA must be described in the <a href="http://samtools.github.io/hts-specs/" target="_blank">Variant Call Format (VCF)</a>.
                                                The EVA has developed a custom <a href="https://github.com/ebivariation/vcf-validator" target="_blank">VCF file validator</a> and accepts submission of only VCF files that pass <a href="https://docs.google.com/document/d/1Wcr2fjzwQ7NmxvqZEbiT7GVck8dLyWG2KvzmOxXAVfs/edit" target="_blank">these validation steps</a>.
                                                Furthermore, VCF files submitted to the EVA should provide either genotypes from the individual samples analysed, or aggregated sample summary information, such as allele frequencies.
                                            </p>
                                            <p>Finally, each submission to the EVA is accompanied by a completed metadata template. This metadata template captures the study description, details of the sample(s) analysed and the experimental methodology. Submission of as much metadata as possible is strongly encouraged as this information is extremely useful for downstream analysis and is directly related to the frequency at which datasets archived at EVA are reused.</p>
                                        </div>
                                    </li>
                                </div>
                                <div id="what-happens-to-submitted-data">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>What happens to my data once submitted?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <img class="faq-images" src="img/EVA_FIle_Level_Processing.svg" alt="EVA File Level Processing: Submitter brings VCF File(s) and metadata template to EVA -> Submitted data are validated against VCF specification and reference genome sequence ->
                                    Accessions are sent to submitter,for referencing in publication for example -> EVA provide links to download the submitted data via the Study Browser"><br /><br />
                                            <h5>Submission validation processes</h5>
                                            <p><span style="font-size:14px;font-weight:bold;">VCF Specification:</span></p>
                                            <p>
                                                All VCF files submitted to EVA are validated for adherence to the <a href="https://samtools.github.io/hts-specs/" target="_blank">format specification</a> using the <a href="https://github.com/EBIvariation/vcf-validator" target="_blank">EVA VCF validation suite</a> which includes all the checks from the vcftools suite, and some more that involve lexical, syntactic and semantic analysis of the VCF input.
                                                The EVA VCF validation suite also includes a debugging tool to automatically correct many of the common errors found in files. In order to improve processing time, submitters are encouraged to prevalidate VCF files prior to submission.
                                            </p>
                                            <p><span style="font-size:14px;font-weight:bold;">Genome Assembly:</span></p>
                                            <p>
                                                To improve interoperability of variant data submitted to EVA with other resources at EMBL-EBI, and the wider open-access community, all VCF files submitted are subject to validation against the INSDC accessioned genome assembly that is referenced in the associated EVA metadata template. EVA is able to only accept files that match a known assembly at 100%. VCF files that fail this validation step shall be archived at the <a href="http://www.ebi.ac.uk/ena" target="_blank">European Nucleotide Archive</a> only.
                                            </p>
                                            <h5>Submission summary</h5>
                                            <p> All  VCF files and novel samples that are submitted to EVA are permanently and securely archived at the <a href="http://www.ebi.ac.uk/ena" target="_blank">European Nucleotide Archive </a> and <a href="http://www.ebi.ac.uk/biosamples/" target="_blank">BioSamples </a>, respectively. The EVA provides access to all submitted data via the <a href="http://www.ebi.ac.uk/eva/?Study%20Browser&browserType=sgv" target="_blank">EVA Study Browser</a>.</p>
                                            <p> Variants within VCF files submitted to EVA are normalised, annotated and used for statistical calculations (via methodologies described below) and these EVA processed data are available via the <a href="http://www.ebi.ac.uk/eva/?Variant%20Browser" target="_blank">EVA Variant Browser</a> and <a href="http://www.ebi.ac.uk/eva/?API" target="_blank">EVA API</a>.
                                            <p> Finally, the EVA brokers all submitted data to the NCBI: <a href="http://www.ncbi.nlm.nih.gov/projects/SNP/" target="_blank">dbSNP</a>, for short variants (<50bp); <a href="http://www.ncbi.nlm.nih.gov/dbvar" target="_blank">dbVar</a> for large variants (>50bp) or <a href="http://www.ncbi.nlm.nih.gov/clinvar/" target="_blank">ClinVar</a>, for variants associated with both a phenotype and clinical significance.</p>
                                        </div>
                                    </li>
                                </div>
                                <div id="accept-submission-same-variation-file-format">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>Do EVA and dbSNP accept submission of the same genetic variation file format?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <p>
                                                Yes. EVA, like dbSNP, accepts submission of genetic variation data that is described in Variant Call Format (VCF) files.
                                                However, unlike dbSNP, EVA does not have a custom VCF specification. Instead EVA only accepts submission of VCF files that conform to the <a href="https://samtools.github.io/hts-specs/VCFv4.3.pdf" target="_blank">specification guidelines</a>.
                                                Conformity to these guidelines is checked at time of submission, but can also be checked beforehand using the <a href="https://github.com/EBIvariation/vcf-validator" target="_blank">EVA Validation Suite</a>. Further guidelines for submission to EVA can be found <a href="?Submit%20Data">here</a>.
                                            </p>
                                        </div>
                                    </li>
                                </div>
                                <div id="do-eva-and-dbsnp-collect-metadata-same-way">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>Do EVA and dbSNP collect metadata in the same way?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <p>
                                                No. Although there are many similarities, the EVA and dbSNP metadata templates, which are submitted along with VCF files, are different.
                                                The EVA metadata template asks the submitter to provide information about the project and samples so that these can be archived simultaneously along with the VCF file(s), whereas this is a stepwise process during dbSNP submission.
                                                The EVA metadata template can be found <a href="files/EVA_Submission_template.METADATA_TEMPLATE_VERSION.xlsx">here</a> and a mocked up version for a fictional study is <a href="files/EVA_Submission_template.METADATA_TEMPLATE_VERSION_mockup.xlsx">here</a>.
                                                Detailed information on how to submit to the EVA can be found <a href="?Submit%20Data">here</a>.

                                            </p>
                                        </div>
                                    </li>
                                </div>
                                <div id="submitted-data-held-privately">
                                    <li class="accordion-item is-active" data-accordion-item>
                                        <a class="accordion-title"><h5>Can my submitted data be held privately at the EVA?</h5></a>
                                        <div class="accordion-content" data-tab-content>
                                            <p>
                                                Yes. Data submitted to the EVA can be held privately for up to one year.
                                                The date of publication is set by the submitter using the "Hold Date" field of the <a href="files/EVA_Submission_template.METADATA_TEMPLATE_VERSION.xlsx">EVA metadata template</a>.
                                            </p>
                                        </div>
                                    </li>
                                </div>
                            </ul>
                        </div>
                        <div class="tabs-panel is-active" id="accessionPanel">
                            <ul class="accordion" data-accordion>
                                <h3 class="padding-bottom-large">EVA & dbSNP Transition</h3>
                                <div>
                                    <p>
                                        EMBL-EBI will maintain reliable accessions for non-human genetic variation data through the European Variation Archive (EVA). NCBI's dbSNP database will continue to maintain stable identifiers for human genetic variation data only. This change will enable a more rapid turnaround for data sharing in this burgeoning field.
                                    </p>
                                    <p>
                                        For more information, please subscribe to <a href="mailto:eva-announce@ebi.ac.uk" target="_top">eva-announce@ebi.ac.uk</a> and follow us at <a href="https://twitter.com/evarchive" target="_blank">@evarchive</a>, where we shall continue to post updates. If anything remains unclear please contact us at <a href="mailto:eva-helpdesk@ebi.ac.uk" target="_top">eva-helpdesk@ebi.ac.uk</a>.
                                    </p>
                                </div>
                                <h3 class="padding-bottom-large">Press releases</h3>
                                <div class="row">
                                    <div class="small-5 medium-5 columns">
                                        <div class="img-format">
                                            <img src="img/dbSNP/EMBL_EBI-logo.png" style="width:150px;" alt="EMBL-EBI-logo">
                                            <br>
                                            <a href="https://goo.gl/gvDGbm" target="_blank">EVA issues long-term IDs for non-human variants</a>
                                        </div>
                                    </div>
                                    <div class="small-7 medium-7 columns">
                                        <div class="img-format">
                                            <img src="img/dbSNP/ncbi.jpg" style="width:100px;" alt="NCBI-logo">
                                            <br>
                                            <a href="https://goo.gl/buoWFD" target="_blank">Phasing out support for non-human genome organism data in dbSNP and dbVar</a>
                                        </div>
                                    </div>
                                </div>
                                <br>
                                <div>
                                    <h3 class="padding-bottom-large">Frequently Asked Questions</h3>
                                    <ul class="padding-bottom-large">
                                        <li><a href="#key-steps-transitional-process">What are the key steps for the dbSNP - EVA transitional process?</a></li>
                                        <li><a href="#accessions-administred-by-eva">What metadata accessions are administered by the EVA?</a></li>
                                        <li><a href="#variant-accession-administred-by-eva">What variant accessions will be administered by the EVA?</a></li>
                                        <li><a href="#retrieve-existing-dbsnp-ids">Will it be possible to retrieve existing dbSNP identifiers from the EVA?</a></li>
                                        <li><a href="#why-variants-in-different-strand">Why are some variants in a different strand than the one dbSNP reports?</a></li>
                                        <li><a href="#cant-find-dbsnp-ids-in-eva-website">Why can't I find some of the dbSNP identifiers on the EVA website?</a></li>
                                        <li><a href="#eva-builds">Will there be EVA builds or releases like the dbSNP ones?</a></li>
                                    </ul>
                                    <br>
                                    <ul class="accordion" data-accordion>
                                        <div id="key-steps-transitional-process">
                                            <li class="accordion-item is-active" data-accordion-item>
                                                <a class="accordion-title"><h5>What are the key steps for the dbSNP - EVA transitional process?</h5></a>
                                                <div class="accordion-content" data-tab-content>
                                                    <p>We have outlined the transitional process, including the timeline and more detailed technical information, in this online presentation:</p>
                                                    <div class="card">
                                                        <a href="https://goo.gl/hDazyv" target="_blank"><img src="img/dbSNP/dbSNP_ppt.png" alt="EVA-presentation"></a>

                                                    </div>
                                                </div>
                                            </li>
                                        </div>
                                        <div id="accessions-administred-by-eva">
                                            <li class="accordion-item is-active" data-accordion-item>
                                                <a class="accordion-title"><h5>What metadata accessions are administered by the EVA?</h5></a>
                                                <div class="accordion-content" data-tab-content>
                                                    <p>
                                                        The EVA follows the SRA model for accessioning metadata:
                                                    <ul>
                                                        <ul>
                                                            <li>A submitted project is administered a 'project' accession. These begin "PRJEB" followed by a numerical sequence.</li>
                                                            <li>Each analysis object within a project is administered an 'analysis' accession. These begin "ERZ" followed by a numerical sequence.</li>
                                                        </ul>
                                                    </ul>
                                                    </p>
                                                    <p>
                                                        Both the project and analysis accessions are sent to the submitter once the data has been validated and fully archived at EMBL-EBI. The project and analysis accessions are stable identifiers that are suitable for publication.                                                    
                                                    </p>
                                                </div>
                                            </li>
                                        </div>
                                        <div id="variant-accession-administred-by-eva">
                                            <li class="accordion-item is-active" data-accordion-item>
                                                <a class="accordion-title"><h5>What variant accessions will be administered by the EVA?</h5></a>
                                                <div class="accordion-content" data-tab-content>
                                                    <p>
                                                        The EVA shall follow the same accessioning principles employed by dbSNP: non-human variants that are submitted to the EVA are issued 'Submitted SNP' (SS) identifiers, and these are periodically clustered to form 'Reference SNP' (RS) identifiers.
                                                    </p>

                                                    <h5>Submitted SNP (SS)</h5>
                                                    <p>
                                                        A Submitted SNP (SS) identifies a variant submitted in the context of a genomic study. New SS IDs issued by the EVA will always be linked to a change from a reference allele to a single alternate allele. This means that a SS ID uniquely identifies a combination of:
                                                        <ul>
                                                            <li>Taxonomy (e.g. 9615 - dog)</li>
                                                            <li>Reference sequence (e.g. CanFam3.1)</li>
                                                            <li>EVA study or dbSNP batch (e.g. PRJEB24066)</li>
                                                            <li>Contig or chromosome</li>
                                                            <li>Start position</li>
                                                            <li>Reference allele</li>
                                                            <li>Alternate allele</li>
                                                        </ul>
                                                    </p>
                                                    <p>
                                                        Most pre-existing dbSNP identifiers also follow this rule but, in some cases, multiple alternate alleles in the same location could be assigned a single SS ID depending on the representation at the time of submission.
                                                    </p>
                                                    <p>
                                                        An SS ID is linked to a variant in a particular study. Therefore, the same variant reported across multiple files in the study will still be assigned only one SS ID.
                                                    </p>
                                                    <h6>Submissions</h6>
                                                    <p>
                                                        The following restrictions apply to EVA submissions, although legacy dbSNP identifiers are preserved even if they don’t:
                                                        <ul>
                                                            <li>Variants must have a specific location in a sequence accessioned by INSDC in order to be archived and accessioned.</li>
                                                            <li>Variants must have supporting evidence (genotypes and/or allele frequencies) in order to be archived and accessioned.</li>
                                                            <li>Non-variants (as in gVCF reference blocks) will be archived, but will neither be assigned an SS ID nor displayed in the Variant Browser.</li>
                                                        </ul>
                                                    </p>
                                                    <h6>Reporting</h6>
                                                    <p>
                                                        The list of SS IDs issued for a submission are provided to the submitter in VCF format, as in <a href="files/ss_ids.vcf">this example</a>; please note the SS IDs listed there are mere examples and do not refer to real variants. Multi-allelic variants are split in multiple records, one per alternate allele. Variants are always reported in the forward strand.
                                                    </p>

                                                    <h5>Reference SNP (RS)</h5>
                                                    <p>
                                                        A Reference SNP (RS) identifies a variant class in a genomic location, which can be reported by one or more studies. RS IDs are preserved when remapped to a newer version of the reference sequence they were initially mapped against; this allows to track how they have evolved with the passing of time. This means that a RS ID can be associated with one or more combinations of:
                                                        <ul>
                                                            <li>Taxonomy (e.g. 9615 - dog)</li>
                                                            <li>Reference sequence (e.g. CanFam3.1)</li>
                                                            <li>Contig or chromosome</li>
                                                            <li>Start position</li>
                                                            <li>Variant class</li>
                                                        </ul>
                                                    </p>
                                                    <p>
                                                        The following table lists the mapping between variant classes supported by dbSNP and the EVA, and the Sequence Ontology terms used by the latter.
                                                    </p>

                                                    <table id="help-variant-classes" class="responsive-table hover tablesorter">
                                                        <thead>
                                                            <tr>
                                                                <th>dbSNP class</th>
                                                                <th>EVA class</th>
                                                                <th>Sequence Ontology term</th>
                                                            </tr>
                                                        </thead>
                                                        <tbody>
                                                            <tr>
                                                                <td>SNV</td>
                                                                <td>SNV</td>
                                                                <td><a href="http://www.sequenceontology.org/browser/current_release/term/SO:0001483">SO:0001483</a></td>
                                                            </tr>
                                                            <tr>
                                                                <td rowspan="3">DIV</td>
                                                                <td>deletion</td>
                                                                <td><a href="http://www.sequenceontology.org/browser/current_release/term/SO:0000159">SO:0000159</a></td>
                                                            </tr>
                                                            <tr>
                                                                <td>insertion</td>
                                                                <td><a href="http://www.sequenceontology.org/browser/current_release/term/SO:0000667">SO:0000667</a></td>
                                                            </tr>
                                                            <tr>
                                                                <td>indel</td>
                                                                <td><a href="http://www.sequenceontology.org/browser/current_release/term/SO:1000032">SO:1000032</a></td>
                                                            </tr>
                                                            <tr>
                                                                <td>HETEROZYGOUS</td>
                                                                <td></td>
                                                                <td></td>
                                                            </tr>
                                                            <tr>
                                                                <td>STR / microsatellite</td>
                                                                <td>tandem repeat</td>
                                                                <td><a href="http://www.sequenceontology.org/browser/current_release/term/SO:0000705">SO:0000705</a></td>
                                                            </tr>
                                                            <tr>
                                                                <td>NAMED</td>
                                                                <td>sequence alteration</td>
                                                                <td><a href="http://www.sequenceontology.org/browser/current_release/term/SO:0001059">SO:0001059</a></td>
                                                            </tr>
                                                            <tr>
                                                                <td>NO VARIATION</td>
                                                                <td>no sequence alteration</td>
                                                                <td><a href="http://www.sequenceontology.org/browser/current_release/term/SO:0002073">SO:0002073</a></td>
                                                            </tr>
                                                            <tr>
                                                                <td>MIXED</td>
                                                                <td></td>
                                                                <td></td>
                                                            </tr>
                                                            <tr>
                                                                <td>MNV</td>
                                                                <td>MNV</td>
                                                                <td><a href="http://www.sequenceontology.org/browser/current_release/term/SO:0002007">SO:0002007</a></td>
                                                            </tr>
                                                        </tbody>
                                                    </table>
                                                    <p>
                                                        Please note that due to the different granularities, dbSNP DIV variants will be assigned the appropriate EVA class when imported.<br />
                                                        HETEROZYGOUS and MIXED variants were initially defined but no longer supported in any of the non-human dbSNP databases, thus why the EVA hasn’t defined a mapping.<br />
                                                        Variants of class NO VARIATION will no longer be accessioned by the EVA, but legacy identifiers will be preserved.
                                                    </p>

                                                    <h6>Reporting</h6>
                                                    <p>
                                                        The list of RS IDs issued for a species in a release will be provided in VCF format, as in <a href="files/rs_ids.vcf">this example</a>; multi-allelic variants are split in multiple records, one per alternate allele, even if several of them have the same variant class. Coordinates are mandatory in VCF format, so RefSNP's without them won't be included in the dump. Please note the RS IDs listed in the file are mere examples and do not refer to real variants.
                                                    </p>
                                                </div>
                                            </li>
                                        </div>
                                        <div id="retrieve-existing-dbsnp-ids">
                                            <li class="accordion-item is-active" data-accordion-item>
                                                <a class="accordion-title"><h5>Will it be possible to retrieve existing dbSNP identifiers from the EVA?</h5></a>
                                                <div class="accordion-content" data-tab-content>
                                                    <p>
                                                        Yes. The EVA is committed to the continuation of existing dbSNP identifiers for Submitted SNP (SS) and Reference SNP (RS).
                                                    </p>
                                                    <p>
                                                        The identifiers of those dbSNP variants that satisfy the EVA submission requirements will be retrievable via the
                                                        <a href="?Variant%20Browser&species=mmusculus_grcm38&selectFilter=snp&snp=rs585444580">EVA variant browser</a> and
                                                        <a href="?API">web services API</a>. If you want to learn more about EVA submission requirements, please click <a href="?Submit%20Data">here</a>.
                                                    </p>
                                                    <p>
                                                        dbSNP variants that don't satisfy these requirements <b>will still be imported</b> and searchable via a specific API and web view. We will work to make this experience as intuitive as possible, while keeping our commitment to only make high-quality variants part of the core EVA database.
                                                    </p>
                                                </div>
                                            </li>
                                        </div>
                                        <div id="why-variants-in-different-strand">
                                            <li class="accordion-item is-active" data-accordion-item>
                                                <a class="accordion-title"><h5>Why are some variants in a different strand than the one dbSNP reports?</h5></a>
                                                <div class="accordion-content" data-tab-content>
                                                    <p>
                                                        The EVA accepts variant submission in the forward strand only, whereas dbSNP is not so restrictive. In addition to this, dbSNP registers multiple orientations:
                                                    </p>
                                                    <p>
                                                        <ul>
                                                            <ul>
                                                                <li>Contig to chromosome</li>
                                                                <li>SNP to contig</li>
                                                                <li>SubSNP (ss) to RefSNP (rs)</li>
                                                            </ul>
                                                        </ul>
                                                    </p>
                                                    <p>
                                                        As a result, variants with a 'reverse' orientation will be displayed differently despite the underlying data being equivalent.
                                                    </p>
                                                </div>
                                            </li>
                                        </div>
                                        <div id="cant-find-dbsnp-ids-in-eva-website">
                                            <li class="accordion-item is-active" data-accordion-item>
                                                <a class="accordion-title"><h5>Why can't I find some of the dbSNP identifiers on the EVA website?</h5></a>
                                                <div class="accordion-content" data-tab-content>
                                                    <p>
                                                        We will gradually import the dbSNP data, and your species of interest may not have been yet processed. Please check our <a href="?dbSNP-Import-Progress">status report</a> on the transition process.
                                                    </p>
                                                    <p>
                                                        If a species has been fully imported and your variant of interest is not displayed in our browser, please communicate the issue via <a href="mailto:eva-helpdesk@ebi.ac.uk" target="_top">eva-helpdesk@ebi.ac.uk</a>
                                                    </p>
                                                </div>
                                            </li>
                                        </div>
                                        <div id="eva-builds">
                                            <li class="accordion-item is-active" data-accordion-item>
                                                <a class="accordion-title"><h5>Will there be EVA builds or releases like the dbSNP ones?</h5></a>
                                                <div class="accordion-content" data-tab-content>
                                                    <p>
                                                        Yes. The EVA will create RefSNP dumps in VCF format every 6 months, starting in Q3 2018. These dumps shall contain the most basic information about each RefSNP:
                                                    </p>
                                                    <p>
                                                        <ul>
                                                            <ul>
                                                                <li>Genomic coordinates</li>
                                                                <li>Reference and alternate alleles in the forward strand</li>
                                                                <li>Identifiers of the studies that reported the variant</li>
                                                            </ul>
                                                        </ul>
                                                    </p>
                                                    <p>
                                                        The list of RS IDs issued for a species in a release will be provided in VCF format, as in <a href="files/rs_ids.vcf">this example</a>; multi-allelic variants are split in multiple records, one per alternate allele, even if several of them have the same variant class. Coordinates are mandatory in VCF format, so RefSNP's without them won't be included in the dump. Please note the RS IDs listed in the file are mere examples and do not refer to real variants.
                                                    </p>
                                                    <p>
                                                        The details of a study submission, reported genotypes and frequencies, etc., can then be queried using the existing EVA website and  <a href="?API">web services API</a>.
                                                    </p>
                                                </div>
                                            </li>
                                        </div>
                                    </ul>
                                </div>
                            </ul>
                        </div>
                    </div>
                </div>
                <div id="studyView" class="hide-div">
                    <nav aria-label="You are here:" role="navigation">
                        <ul class="breadcrumbs">
                            <li><a href="?Home">EVA</a></li>
                            <li><a href="?Study Browser">Study Browser</a></li>
                            <li>
                                <span class="show-for-sr">Current: </span> Study View
                            </li>
                        </ul>
                    </nav>
                    <div id="studyViewContent"></div>
                </div>
                <div id="variantView" class="hide-div">
                    <nav aria-label="You are here:" role="navigation">
                        <ul class="breadcrumbs">
                            <li><a href="?Home">EVA</a></li>
                            <li><a href="?Variant Browser">Variant Browser</a></li>
                            <li>
                                <span class="show-for-sr">Current: </span> Variant View
                            </li>
                        </ul>
                    </nav>
                    <h2 id="variantInfo"  data-magellan-target="variantInfo" class="padding-bottom-large"></h2>
                    <div data-sticky-container>
                        <div class="sticky" data-sticky data-options="marginTop:2.65;" style="width:100%" data-top-anchor="summary-grid-div:top">
                                <div data-magellan class="megallan-div expanded button-group">
                                    <a href="#summary-grid-div" class="button">Summary</a>
                                    <a href="#consequence-types-grid-div" class="button">Consequence Type</a>
                                    <a href="#studies-grid-div" class="button">Files</a>
                                    <a href="#population-stats-grid-div" class="button">Population Statistics</a>
                                </div>
                        </div>
                    </div>
                    <div class="columns medium-12 large-12">
                        <div id="summary-grid-div" class="padding-bottom-large" data-magellan-target="summary-grid-div"></div><br/>
                        <div id="summary-grid"></div>
                        <div id="consequence-types-grid-div"  data-magellan-target="consequence-types-grid-div"></div><br/><br/>
                        <div id="consequence-types-grid"></div>
                        <div id="studies-grid-div"  data-magellan-target="studies-grid-div"></div><br/>
                        <div id="studies-grid" style="margin-left:-20px;"></div>
                        <div id="population-stats-grid-div"  data-magellan-target="population-stats-grid-div"></div><br/>
                        <div id="population-stats-grid-view" style="margin-left:-20px;"></div>
                    </div>
                    <div id="variantViewContent"></div>
                </div>
                <div id="geneView" class="hide-div">
                    <nav aria-label="You are here:" role="navigation">
                        <ul class="breadcrumbs">
                            <li><a href="?Home">EVA</a></li>
                            <li><a href="?Clinical Browser">Clinical Browser</a></li>
                            <li>
                                <span class="show-for-sr">Current: </span> Gene View
                            </li>
                        </ul>
                    </nav>
                    <div id="geneViewContent"></div>
                </div>
                <div id="dbSNPImport" class="hide-div">
                    <nav aria-label="You are here:" role="navigation">
                        <ul class="breadcrumbs">
                            <li><a href="?Home">EVA</a></li>
                            <li>
                                <span class="show-for-sr">Current: </span> dbSNP Import Progress
                            </li>
                        </ul>
                    </nav>
                    <div id="dbSNPImportContent"></div>
                </div>
        </div>
    </section>
</div>


<footer>

    <div id="elixir-banner" data-color="grey" data-name="This service" data-description="The European Variation Archive is an ELIXIR Deposition Database" data-more-information-link="https://www.elixir-europe.org/platforms/data/elixir-deposition-databases" data-use-basic-styles="false"></div>
    <div id="global-footer">

        <nav id="global-nav-expanded" class="row">
            <!-- Footer will be automatically inserted by footer.js -->
        </nav>

        <section id="ebi-footer-meta" class="row">
            <!-- Footer meta will be automatically inserted by footer.js -->
        </section>

    </div>

</footer>
</div> <!--! end of #wrapper -->


<!-- JavaScript -->
<!-- JavaScript at the bottom for fast page loading -->
<!-- build:script vendor -->
<script type="text/javascript" src="../vendor/ext-6.0.1/js/ext-all.js"></script>
<script type="text/javascript" src="../bower_components/underscore/underscore.js"></script>
<script type="text/javascript" src="../bower_components/backbone/backbone.js"></script>
<script type="text/javascript" src="../bower_components/jquery/dist/jquery.min.js"></script>
<!-- These dependencies must go AFTER jquery -->
<script type="text/javascript" src="../bower_components/linqjs/linq.min.js"></script>
<script type="text/javascript" src="../bower_components/vkbeautify/vkbeautify.js"></script>
<!-- /build -->
<script type="text/javascript" src="https://www.gstatic.com/charts/loader.js"></script>
<!-- build:script lib -->
<script type="text/javascript" src="../lib/jsorolla/build/1.1.9/genome-viewer/genome-viewer.js"></script>
<!-- /build -->
<!-- build:script eva-js -->
<script type="text/javascript" src="js/eva-manager.js"></script>
<script type="text/javascript" src="js/eva-annotation-model.js"></script>
<script type="text/javascript" src="js/eva-config.js"></script>
<script type="text/javascript" src="js/eva-menu.js"></script>
<script type="text/javascript" src="js/eva-adapter.js"></script>
<script type="text/javascript" src="js/variant-widget/eva-variant-widget-panel.js"></script>
<script type="text/javascript" src="js/study-browser/eva-study-browser-widget-panel.js"></script>
<script type="text/javascript" src="js/study-browser/eva-study-browser-filter-form-panel.js"></script>
<script type="text/javascript" src="js/study-browser/eva-study-browser-type-filter-form-panel.js"></script>
<script type="text/javascript" src="js/study-browser/eva-study-browser-text-search-form-panel.js"></script>
<script type="text/javascript" src="js/study-browser/eva-study-browser-grid.js"></script>
<script type="text/javascript" src="js/beacon-form/eva-beacon-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/eva-variant-widget.js"></script>
<script type="text/javascript" src="js/variant-widget/eva-variant-browser-grid.js"></script>
<script type="text/javascript" src="js/variant-widget/eva-variant-files-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/eva-variant-genotype-grid-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/eva-variant-population-stats-panel.js"></script>
<script type="text/javascript" src="js/views/eva-study-view.js"></script>
<script type="text/javascript" src="js/views/eva-variant-view.js"></script>
<script type="text/javascript" src="js/views/eva-gene-view.js"></script>
<script type="text/javascript" src="js/statistics/eva-statistics.js"></script>
<script type="text/javascript" src="js/statistics/dgva-statistics.js"></script>
<script type="text/javascript" src="js/beacon-form/eva-beacon-form.js"></script>
<script type="text/javascript" src="js/beacon-form/eva-variant-search-form.js"></script>
<script type="text/javascript" src="js/variant-widget/filters/eva-form-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/filters/eva-species-filter-form-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/filters/eva-study-filter-form-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/filters/eva-position-filter-form-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/filters/eva-annotation-version-filter-form-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/filters/eva-population-frequency-filter-form-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/filters/eva-protein-substitutions-score-filter-form-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/filters/eva-consequence-type-filter-form-panel.js"></script>
<script type="text/javascript" src="js/variant-widget/filters/eva-conservation-score-filter-form-panel.js"></script>
<script type="text/javascript" src="js/clinvar/eva-clinical-widget-panel.js"></script>
<script type="text/javascript" src="js/clinvar/eva-clinvar-widget.js"></script>
<script type="text/javascript" src="js/clinvar/eva-clinvar-browser-grid.js"></script>
<script type="text/javascript" src="js/clinvar/filters/eva-clinvar-position-filter-form-panel.js"></script>
<script type="text/javascript" src="js/clinvar/filters/eva-clinvar-filter-form-panel.js"></script>
<script type="text/javascript" src="js/clinvar/filters/eva-clinvar-trait-filter-form-panel.js"></script>
<script type="text/javascript" src="js/clinvar/eva-clinvar-summary-panel.js"></script>
<script type="text/javascript" src="js/clinvar/eva-clinvar-assertion-panel.js"></script>
<script type="text/javascript" src="js/clinvar/eva-clinvar-links-panel.js"></script>
<script type="text/javascript" src="js/clinvar/eva-clinvar-annotation-panel.js"></script>
<script type="text/javascript" src="js/dbSNP-import/dbSNP-import-progress.js"></script>
<script type="text/javascript" src="js/eva.js"></script>
<script type="text/javascript" src="js/eva-google-analytics.js"></script>
<!-- /build -->
<!-- build:script ebi_framework -->
<script src="../lib/EBI-Framework/js/cookiebanner.js"></script>
<script src="../lib/EBI-Framework/js/foot.js"></script>
<script src="../lib/EBI-Framework/js/script.js"></script>
<!-- The Foundation theme JavaScript -->
<script src="../lib/EBI-Framework/libraries/foundation-6/js/foundation.js"></script>
<script src="../lib/EBI-Framework/libraries/foundation-6/js/foundation.magellan.js"></script>
<script src="../lib/EBI-Framework/js/foundationExtendEBI.js"></script>
<script src="../lib/EBI-Framework/libraries/tablesorter/dist/js/jquery.tablesorter.min.js"></script>
<!-- /build -->
<script type="text/JavaScript">$(document).foundation();</script>
<script type="text/JavaScript">$(document).foundationExtendEBI();</script>
<script defer="defer" src="https://ebi.emblstatic.net/web_guidelines/EBI-Framework/v1.3/js/elixirBanner.js"></script>

<script type="text/javascript">
    var eva;
    var start = function () {
        // Bring in the import content.
        var homeDiv = document.querySelector('#home-page');
        var submitDiv = document.querySelector('#submit');
        var feedbackDiv = document.querySelector('#feedback');
        var studyViewDiv = document.querySelector('#studyView');
        var variantViewDiv = document.querySelector('#variantView');
        var geneViewDiv = document.querySelector('#geneView');
        var apiDiv =  document.querySelector('#api');
        var helpDiv = document.querySelector('#help-section');
        var dbSNPImportDiv = document.querySelector('#dbSNPImport');



        var tab = getUrlParameters("");
        var tabId =  decodeURI(tab.id);
        if(tabId == 'eva study'){
            var evastudyView = new EvaStudyView({
                target:'studyViewContent',
                type: 'eva',
                projectId:tab.value
            });

        }
        if(tabId == 'dgva study'){
            var dgvastudyView = new EvaStudyView({
                target:'studyViewContent',
                type: 'dgva',
                projectId:tab.value
            });
        }
        if(tabId == 'variant'){
            var variantSpecies = $.urlParam('species');
            var annotationVersion = $.urlParam('annotationVersion');
            var evaVarinatView = new EvaVariantView({
                target:'variantViewContent',
                position:tab.value,
                species:variantSpecies,
                annotationVersion:annotationVersion
            });
            $('.sticky').foundation('_calc', true);
            $(window).trigger('resize');
        }

        if(tabId == 'gene'){
            var variantSpecies = $.urlParam('species');
            var evaGeneView = new EvaGeneView({
                target:'geneViewContent',
                geneId:tab.value,
                species:variantSpecies
            });
        }

        Ext.getBody().removeCls('x-body');
        var ulLocalNav = document.querySelector('.eva-primary-menu');

        eva = new Eva({
            target: 'content',
            targetMenu: ulLocalNav,
            width: $('#content').width(),
            homeDiv:homeDiv,
            submitDiv:submitDiv,
            feedbackDiv:feedbackDiv,
            studyView:studyViewDiv,
            variantView:variantViewDiv,
            geneView:geneViewDiv,
            apiDiv:apiDiv,
            helpDiv:helpDiv,
            dbSNPImportDiv:dbSNPImportDiv
        });


        if(tab.id){
            eva.draw(tabId);
        }else{
            eva.draw('Home');
        }

        window.onpopstate = function(event) {
            if (event.state) {
                location.reload();
            }
        }

    };

    function getUrlParameters(staticURL) {
        var data = [];

        var currLocation = (staticURL.length) ? staticURL : window.location.search;
        if (currLocation.indexOf("?") >= 0) {
            parArr = currLocation.split("?")[1].split("&");
            if (parArr) {
                for (var i = 0; i < parArr.length; i++) {
                    parr = parArr[i].split("=");
                    data.id = parr[0].replace(/-/g, " ");
                    data.value = parr[1];
                    return data;
                }
            }
        } else {
            return false
        }
    }

    $.urlParam = function(name){

        var results = new RegExp('[?&]' + name + '=([^&#]*)').exec(window.location.href);
        if (results==null){
            return null;
        }
        else{
            return results[1] || 0;
        }
    }

    //Start point
    window.addEventListener('load', function (e) {
        // all imports loaded
        Ext.QuickTips.init();
        Ext.onReady(start);
    });
    
    
</script>
</body>
</html>
