<!DOCTYPE html>
<html>
<head lang="en">
<meta name=viewport content="width=device-width, initial-scale=1">
<meta name="description" content="EMBL-EBI, EVA, Variation, database, archive"><!-- Describe what this page is about -->
<meta name="keywords" content="bioinformatics, europe, institute"><!-- A few keywords that relate to the content of THIS PAGE (not the whol project) -->
<meta name="author" content="EMBL-EBI"><!-- Your [project-name] here -->
<meta charset="utf-8">

<!-- build:style vendor -->
<link type="text/css" rel="stylesheet" href="../vendor/ext-6.0.1/theme/theme-ebi-embl-all.css"/>
<!--<link type="text/css" rel="stylesheet" href="../lib/jsorolla/vendor/jquery.qtip.min.css"/>-->
<link rel="stylesheet" href="../bower_components/bootstrap/dist/css/bootstrap.css">
<!--<link rel="stylesheet" href="../lib/jsorolla/vendor/font-awesome/css/font-awesome.min.css">-->
<link rel="stylesheet" href="css/ebi-complaince/ebi-fluid-embl.css">
<!-- /build -->
<!-- build:style css -->
<!--<link type="text/css" rel="stylesheet" href="../lib/jsorolla/styles/css/style.css"/>-->
<link type="text/css" rel="stylesheet" href="css/eva.css"/>
<!-- /build -->

</head>

<body class="eva">

    <div id="skip-to">
        <ul>
            <li><a href="#content">Skip to main content</a></li>
            <li><a href="#local-nav">Skip to local navigation</a></li>
            <li><a href="#global-nav">Skip to EBI global navigation menu</a></li>
            <li><a href="#global-nav-expanded">Skip to expanded EBI global navigation menu (includes all sub-sections)</a></li>
        </ul>
    </div>

    <div id="wrapper" class="container_24">
        <header>
            <div id="global-masthead" class="masthead grid_24">
                <!--This has to be one line and no newline characters-->
                <a href="//www.ebi.ac.uk/" title="Go to the EMBL-EBI homepage"><img src="img/EMBL_EBI_Logo_white.png" alt="EMBL European Bioinformatics Institute" width="75" height="75"></a>

                <nav>
                    <ul id="global-nav">
                        <!-- set active class as appropriate -->
                        <li class="first active" id="services"><a href="//www.ebi.ac.uk/services">Services</a></li>
                        <li id="research"><a href="//www.ebi.ac.uk/research">Research</a></li>
                        <li id="training"><a href="//www.ebi.ac.uk/training">Training</a></li>
                        <li id="industry"><a href="//www.ebi.ac.uk/industry">Industry</a></li>
                        <li id="about" class="last"><a href="//www.ebi.ac.uk/about">About us</a></li>
                    </ul>
                </nav>
            </div>

            <div id="local-masthead" class="masthead grid_24 nomenu">

                <!-- local-title -->
                <div class="grid_24 alpha" id="local-title">
                    <h1><img src="img/eva_logo.png" alt="" width="75" height="64"><a href="?Home" title="Back to European Variation Archive homepage">European Variation Archive</a></h1>
                </div>
                <!-- /local-title -->

                <!-- local-search -->
                <!--<div class="grid_12 omega">-->
                    <!--<form id="local-search" name="local-search" action="[search-action]" method="post">-->

                        <!--<fieldset>-->

                            <!--<div class="left">-->
                                <!--<label>-->
                                    <!--<input type="text" name="first" id="local-searchbox">-->
                                <!--</label>-->
                                <!--&lt;!&ndash; Include some example searchterms - keep them short and few! &ndash;&gt;-->
                                <!--<span class="examples">Examples: <a href="[search-url-1]">[search-text-1]</a>, <a href="[search-url-1]">[search-text-2]</a>, <a href="[search-url-1]">[search-text-3]</a></span>-->
                            <!--</div>-->

                            <!--<div class="right">-->
                                <!--<input type="submit" name="submit" value="Search" class="submit">-->
                                <!--&lt;!&ndash; If your search is more complex than just a keyword search, you can link to an Advanced Search,-->
                                     <!--with whatever features you want available &ndash;&gt;-->
                                <!--<span class="adv"><a href="../search" id="adv-search" title="Advanced">Advanced</a></span>-->
                            <!--</div>-->

                        <!--</fieldset>-->

                    <!--</form>-->
                <!--</div>-->
                <!-- /local-search -->
                <!-- local-nav -->
                <nav>
                    <ul class="eva-primary-menu">
                        <li class="active"><a>Home</a></li>
                        <li><a>Submit Data</a></li>
                        <li><a>Study Browser</a></li>
                        <li><a>Variant Browser</a></li>
                        <li><a>Clinical Browser</a></li>
                        <li><a>GA4GH</a></li>
                        <!--<li><a>Genome Browser</a></li>-->
                        <li><a>API</a></li>
                        <li><a>FAQ</a></li>
                        <!--<li><a>About</a></li>-->
                        <li><a>Feedback</a></li>
                        <!--<li><a>Feedback</a></li>-->

                    </ul>
                </nav>
                <!-- /local-nav -->

            </div>
        </header>
        <div id="content" role="main" class="grid_24 clearfix">
            <title>European Variation Archive &lt; EMBL-EBI</title>
            <div id="home"  class="hide-div">
                <!--<div class="jumbotron" style="height:120px;"><h2 style="margin-top:-30px;">Beta site</h2><p style="font-size:12px;">EVA is a work in progress. The current website is a beta version and we seek feedback from our users. Please contact <a href="mailto:eva-helpdesk@ebi.ac.uk" target="_top">eva-helpdesk@ebi.ac.uk</a> with all suggestions, bug reports and feature requests for EVA.</p></div>-->
                <div class="row">
                    <div class="col-md-6">
                        <h3><span class="icon icon-generic" data-icon="&"></span>Overview</h3>
                        <div><p> The European Variation Archive is an open-access database of all types of genetic variation data from all species.</p></div>
                        <div><p> ​All users can <a href="?Study Browser "> download ​data </a> from any​ ​study, or <a href="?Submit Data"> submit their own data</a> to the archive. You can also query all variants in the EVA by study, gene, chromosomal location or dbSNP identifier using our <a href="?Variant Browser"> Variant Browser</a>. </p></div>
                        <div><p> We will be adding new features to the EVA on a regular basis, and welcome <a href="mailto:eva-helpdesk@ebi.ac.uk" target="_top">your comments and feedback</a>. </p></div>
                    </div>

                    <div class="col-xs-12 col-sm-12 col-md-6">
                        <h3><span class="icon icon-generic" data-icon="N"></span>News</h3>
                        <div id="twitter-widget"></div>
                    </div>

                </div>

                <div class="row">
                    <div class="col-xs-6 col-sm-6 col-md-6">
                        <h3><span class="icon icon-generic" data-icon="g"></span>Statistics</h3>
                        <div><h4>Short Genetic Variants (&lt;50bp)</h4><div id="eva-statistics"></div></div>
                    </div>
                    <div class="col-xs-6  col-sm-6 col-md-6">
                        <div><br /><br /><h4>Structural Variants (&gt;50bp)</h4><div id="dgva-statistics"></div></div>
                    </div>

                </div>

                <div class="row">
                    <div class="col-md-12">
                        <div><p style="font-size:11px;">This web application makes an intensive use of new web technologies and standards like HTML5. Please see <a href="?About">About</a> for further browser compatibility notes.</p></div>
                    </div>
                </div>
            </div>

            <div  id="submit" class="hide-div">
                <h2>European Variation Archive submissions</h2>
                <div>
                    <p>EVA follows the infrastructure of fellow EMBL-EBI resources European Nucleotide Archive (<a href="http://www.ebi.ac.uk/ena/" target="_blank">ENA</a>) and European Genome-phenome Archive (<a href="http://www.ebi.ac.uk/ega/" target="_blank">EGA</a>)  to accept, archive, and accession <a target="_blank" href="?VCF">VCF files</a>. Submissions consist of VCF file(s) and metadata that describe sample(s), experiment(s), and analysis that produced the variant and/or genotype call(s).</p>
                    <p>EVA works in collaboration with the Database of Genomics Variants Archive (<a href="http://www.ebi.ac.uk/dgva" target="_blank">DGVa</a>) to accession and archive structural variants. DGVa relies on a template based submission process that is explained in detail <a href="http://www.ebi.ac.uk/dgva/data-submission" target="_blank">here</a></p>                    <p>​Data submitted to EVA is brokered to our collaborating databases at <a href="http://www.ncbi.nlm.nih.gov/" target="_blank">NCBI</a>, <a href="http://www.ncbi.nlm.nih.gov/SNP/" target="_blank">dbSNP</a> and <a href="http://www.ncbi.nlm.nih.gov/dbvar" target="_blank">dbVar</a>. It is therefore  unnecessary to submit data to multiple resources. Please contact <a href="mailto:eva-helpdesk@ebi.ac.uk?subject=Sequence submissionto EVA&amp; target="_top">eva-helpdesk@ebi.ac.uk</a> if you would like any further information on this brokering process or collaboration.</p>
                    <div><p>If you have data in a format other than VCF, which cannot be converted to VCF, please contact <a href="mailto:eva-helpdesk@ebi.ac.uk" target="_top">eva-helpdesk@ebi.ac.uk</a>.  Additional submission formats may be supported over time as required by the scientific community.</p></div>
                </div>

                <h3>Key stages of EVA submissions</h3>

                <h4><span class="icon icon-generic" data-icon="M"></span>Contact</h4>
                <div>
                    <!--<p><img src="img/Contact.png" alt="" width="48" height="48"> Contact the EVA Helpdesk via <a href="?submission-start">this webform</a> in order to provide details of your submission.</p>-->
                    <!--<p>Contact the EVA Helpdesk via <a href="?submission-start">this webform</a> in order to provide details of your submission.</p>-->
                    <p>Contact  <a href="mailto:eva-helpdesk@ebi.ac.uk" target="_top">eva-helpdesk@ebi.ac.uk</a> in order to provide details of your submission.</p>
                </div>

                <h4><span class="icon icon-generic" data-icon="n"></span>Receive</h4>
                <div>
                    <p>
                        <!--<img src="img/Receive.png" alt="" width="48" height="48"> &nbsp;Receive your <a href="files/eva_submission_package.zip" target="_blank">submission pack</a>, which will include:-->
                        Download your <a href="files/eva_submission_package.zip" target="_blank" onClick="ga('send', 'event', { eventCategory: 'Download', eventAction: 'Click', eventLabel: 'eva_submission_package.zip'});">submission pack</a>, which will include:
                    <div style="margin-left:10px;">
                        <ul>
                            <li>Details for your submission uploads </li>
                            <li> <a href="?Templates" target="_blank">Templates</a> to capture your associated metadata </li>
                            <li>Key stages for your submission</li>
                        </ul>
                    </div>
                    </p>
                </div>

                <h4><span class="icon icon-functional" data-icon="_"></span>Submit</h4>
                <div>
                    <!--<p><img src="img/Upload.png" alt="" width="48" height="48"> Upload your data files to your private submission upload account or directly to the <a href="mailto:eva-helpdesk@ebi.ac.uk?subject=Sequence submissionto EVA&amp;body=**We will provide you with a submission package metadata templates guides for completing templates**" target="_top">eva-helpdesk@ebi.ac.uk</a>.</p>-->
                    <p>Upload your data files to your private submission upload account or directly to the <a href="mailto:eva-helpdesk@ebi.ac.uk?subject=Sequence submissionto EVA&amp;body=**We will provide you with a submission package metadata templates guides for completing templates**" target="_top">eva-helpdesk@ebi.ac.uk</a>.</p>
                </div>
                <h2>Clinically Relevant Genetic Variants</h2>
                <div class="row">
                    <div class="col-md-12">
                        <p>
                            EVA welcomes submission of clinically relevant variant data from clinical testing labs, locus-specific databases, expert panels and professional societies.
                            EVA works in collaboration with the <a href="http://www.ncbi.nlm.nih.gov/clinvar" target="_blank">ClinVar resource at NCBI</a> to provide the most comprehensive dataset of clinically relevant variants available worldwide - all data in NCBI ClinVar is imported into the EVA database.
                            For clinical data submitted directly to EVA, we shall create a submission to ClinVar on the user's behalf, with user's permission, thus exchanging data between EVA and ClinVar.
                            Therefore, submitters should submit clinically relevant data once either to NCBI ClinVar or to EVA, but should not submit to both.

                        </p>
                        <p>To submit data to EVA please download this  <a href="files/SubmissionTemplate.xlsx" target="_blank"  onClick="ga('send', 'event', { eventCategory: 'Download', eventAction: 'Click', eventLabel: 'SubmissionTemplate.xlsx'});">template</a>, fill in and send by email to <a href="mailto:eva-helpdesk@ebi.ac.uk" target="_top">eva-helpdesk@ebi.ac.uk</a>.
                            <!--A completed template for a fictional study can be found <a href="files/EVA_Submission_template.EVA.V1.0.1_live_mockup_plus_Clinical_Variants.xlsx" target="_blank">here</a>.</p>-->
                        <!--<p>If you would like to discuss any aspect of EVA, or if anything is unclear, please contact us directly at <a href="mailto:eva-helpdesk@ebi.ac.uk" target="_top">eva-helpdesk@ebi.ac.uk</a>. </p>-->
                    </div>
                </div>
            </div>

            <div id="feedback" class="hide-div">
                <h2>Feedback</h2>
                <div>
                    <p>If you have any questions related to the European Variation Archive resource, please <a href="mailto:eva-helpdesk@ebi.ac.uk">contact us</a>.</p>
                    <p>Follow us on Twitter using <a href="https://twitter.com/EBIvariation" target="_blank">@EBIvariation</a></p>
                </div>
                <div id="feedback-form">
                    <!--<h3>Feedback</h3>-->
                    <div class="row">
                        <div class="col-md-2"></div>
                        <div class="col-md-8 feedback-form">
                            <iframe src="https://www.surveymonkey.co.uk/r/eva-feedback" frameborder="0" width="100%"  height="1490"></iframe>
                        </div>
                    </div>
                </div>

            </div>

            <div id="submissionTemplates" class="hide-div">
                <h2>Templates</h2>
                <div>
                    <h3>EVA Submissions</h3>
                    <p>Unless fully described in the header of the VCF file(s), EVA submissions are accompanied by an Excel spreadsheet and/or tab-delimited text files which describe the meta-data. We encourage our users to submit as much metadata as possible.</p>
                    <p>Please download our submission template via this <a href="files/EVA_Submission_template.EVA_.V1.0.1_live.xlsx" target="_blank"  onClick="ga('send', 'event', { eventCategory: 'Download', eventAction: 'Click', eventLabel: 'EVA_Submission_template.EVA_.V1.0.1_live.xlsx'})">link</a>.</p>
                    <p>A completed template for a fictional submission can be downloaded <a href="files/EVA_Submission_template.EVA_.V1.0.1_live_mockup.xlsx" target="_blank" onClick="ga('send', 'event', { eventCategory: 'Download', eventAction: 'Click', eventLabel: 'EVA_Submission_template.EVA_.V1.0.1_live_mockup.xlsx'})">here</a>.</p>
                </div>
                <h3>EVA Clincal Submissions</h3>
                <p>EVA Clinical submissions are described via an Excel spreadsheet and/or tab-delimited text files. Our submissions templates are fully compliant with the data required for ClinVar allowing us to broker the submission of all EVA Clincal data to our collaborating NCBI resource.</p>
                <p>Please download our EVA Clinical submission template via this <a href="files/SubmissionTemplate.xlsx" target="_blank"  onClick="ga('send', 'event', { eventCategory: 'Download', eventAction: 'Click', eventLabel: 'ClinicalSubmissionTemplate.xlsx'})">link</a>.</p>
                <!--<p>A completed EVA Clinical template for a fictional submission can be downloaded <a href="files/EVA_Submission_template.EVA.V1.0.1_live_mockup_plus_Clinical_Variants.xlsx" target="_blank">here</a>.</p>-->
            </div>

            <div id="vcf-format" class="hide-div">
                <h2>EVA VCF Format</h2>
                <p> EVA only accepts VCF files that conform to <a href="http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41" target="_blank">VCF version 4.1 and above.</a></p>
                <h3>VCF Specification</h3>
                <div>
                    <p>All VCF files must be valid, i.e. they must pass validation using a VCF validator, such as vcftools. In addition, all VCF files must include a reference tag (##reference). Any tag that contains white space (e.g. spaces) must be enclosed in double quotes, e.g.</p>
                    <p>##EVA_PipelineDescription="This is a testing pipepline!" OK</p>
                    <p>##EVA_PipelineDescription=This is a testing pipepline! NOT OK</p>
                </div>
                <h3>EVA Specific headers</h3>
                <div>
                    <p>An EVA submission automatically creates the required objects in ENA. A VCF file belongs to an Analysis, which in turn belongs to a Project. Currently each VCF file can only belong to one Analysis, and each Analysis can only contain one VCF file. The Analysis allows the VCF file to be linked to the Samples or SampleGroup (a group of samples).</p>
                </div>
                <h3>Samples</h3>
                <div>
                    <p>If a Sample or SampleGroup already exists in ENA, then the VCF file can reference this object. This can be done in 3 ways:</p>
                    <p>1) Using the ENA Sample accession in the header of the VCF file.</p>
                    <p>2) Use the ##EVA_Sample information line to to link the sample ID in the VCF file to the Sample accession using the following specification:</p>
                    <p>##EVA_Sample=&lt;ID=sample_id,ACCESSION=ena_accession&gt;</p>
                    <p>3) Supply a sample mapping file(s) along with the VCF files, which links the VCF sample IDs to the Sample accession. The filename for the sample mapping file is given using the ##EVA_SampleMappingFile information line in the VCF file. Multiple files can be used, but each sample ID from the VCF must only occur once in all of the specified sample mapping files. The format to specify sample mapping files is:</p>
                    <p>##EVA_SampleMappingFile=&lt;NAME=filename,MD5=md5sum&gt;</p>
                    <p dir="ltr">Samples can be linked to external resources, e.g. CORIELL. This is done using the ##EVA_SampleLink tag. e.g.</p>
                    <p dir="ltr">##EVA_SampleLink=&lt;ID=HG00097,DB=CORIELL,DB_ID=NA18501,LABEL="Link to Coriell"&gt;</p>
                    <p dir="ltr">ID and DB are required (if the VCF ID is the same as the external resource ID). If the external resource ID is different to the VCF ID, this can be specified using the DB_ID sub-tag. LABEL is optional.</p>
                    <p>The ##EVA_Scope tag defines the study type. This tag is optional, and if used must correspond to the number of samples listed in the VCF file. If there is 1 sample, the ##EVA_Scope tag is 'single-isolate', otherwise it is 'multi-isolate'.</p>
                </div>
                <h3>Analysis</h3>
                <div>
                    <p dir="ltr">Each VCF file must belong to an Analysis object which describes the analysis used to produce the VCF file. Currently and Analysis object can only contain 1 VCF file.</p>
                    <p dir="ltr">The following tags are used to describe the analysis:</p>
                    <p dir="ltr">##reference--REQUIRED</p>
                    <p dir="ltr">##EVA_AnalysisTitle--REQUIRED</p>
                    <p dir="ltr">##EVA_PipelineDescription--REQUIRED</p>
                    <p dir="ltr">##EVA_AnalysisCenter--OPTIONAL</p>
                    <p dir="ltr">##EVA_AnalysisDate--OPTIONAL</p>
                    <p dir="ltr">##EVA_CallingAlgorithm--OPTIONAL</p>
                    <p dir="ltr">##EVA_Platform--OPTIONAL</p>
                    <p dir="ltr">##EVA_ExperimentType--REQUIRED</p>
                    <p dir="ltr">##EVA_RunAccession--OPTIONAL</p>
                    <p dir="ltr">##EVA_AnalysisAccession--OPTIONAL</p>
                    <p dir="ltr">##reference</p>
                    <p dir="ltr">This is a 'highly recommended' tag according to VCFv4.1 specifications, and EVA require it for all VCF files. It can refer to either:</p>
                    <p dir="ltr">1) An INSDC assembly name or accession,e.g.</p>
                    <p dir="ltr">##reference=GRCh37</p>
                    <p dir="ltr">or</p>
                    <p dir="ltr">##reference=GCA_000001405.1</p>
                    <p dir="ltr">2) or a location of the reference file, e.g.</p>
                    <p dir="ltr">##reference=file:///seq/references/1000GenomesPilot-NCBI36.fasta</p>
                    <p dir="ltr">or</p>
                    <p dir="ltr">##reference=<a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.2bit">http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.2bit</a></p>
                    <p dir="ltr">EVA also requires INSDC accessions or names for each of the chromosomes (or sequences) referenced in column 1 of the VCF. This should be of the form:</p>
                    <p dir="ltr">##contig=&lt;ID=1,accession=<a href="http://www.ncbi.nlm.nih.gov/nuccore/CM000663.1">CM000663.1</a>,length=249,250,621,species=”Homo sapiens”,taxonomy=9606&gt;</p>
                    <p dir="ltr">or</p>
                    <p dir="ltr">##contig=&lt;ID=1,URL=http://www.ncbi.nlm.nih.gov/nuccore/224384768?report=fasta&gt;</p>
                    <p dir="ltr">##EVA_AnalysisTitle</p>
                    <p dir="ltr">This is the title displayed for the accession Analysis object, e.g.</p>
                    <p dir="ltr">##EVA_AnalysisTitle="1000 Genomes Phase 1 Variation Calls Analysis"</p>
                    <p dir="ltr">##EVA_PipelineDescription</p>
                    <p dir="ltr">Description of analysis pipeline. Free text field, e.g.</p>
                    <p dir="ltr">##EVA_PipelineDescription="Final release set from derived from data from 7 different centers. See PUBMED:23128226"</p>
                    <p dir="ltr">##EVA_AnalysisCenter</p>
                    <p dir="ltr">The center that performed the analysis</p>
                    <p dir="ltr">##EVA_AnalysisCenter="European Bioinformatics Institute"</p>
                    <p dir="ltr">##EVA_AnalysisDate</p>
                    <p dir="ltr">The date the analysis was performed (in DD/MM/YYYY-hh:min:secs-offest, where offset and hh:min:secs are optional, hr:min:secs defaults to 12:00:00, and offset indicates the hour offset from UTC defaulting to 0), e.g. for an analysis performed at 9.30 am, 6 hours before UTC on 30th May 2011:</p>
                    <p dir="ltr">##EVA_AnalysisDate="30/05/2011-09:30-06"</p>
                    <p dir="ltr">##EVA_CallingAlgorithm</p>
                    <p dir="ltr">Details of the calling algorithm, version, and parameters, e.g.</p>
                    <p dir="ltr">##EVA_CallingAlgorithm="GATKv2.3, java -jar GenomeAnalysisTK.jar -T HaplotypeCaller -R reference/human_g1k_v37.fasta -I HG00096.bam -I HG00097.bam --dbsnp dbSNP.vcf -stand_call_conf &nbsp;-stand_emit_conf 10.0 -o output.raw.snps.indels.vcf"</p>
                    <p dir="ltr">##EVA_Platform</p>
                    <p dir="ltr">The platform used, e.g.</p>
                    <p dir="ltr">##EVA_Platform="Illumina Genome Analyser IIe"</p>
                    <p dir="ltr">##EVA_ExperimentType</p>
                    <p dir="ltr">Tag to indicate the type of experiment, choose from:</p>
                    <p dir="ltr">Whole genome, Exome, Array</p><p dir="ltr">e.g.</p>
                    <p dir="ltr">##EVA_ExperimentType="Whole Genome"</p>
                    <p dir="ltr">##EVA_RunAccession</p>
                    <p dir="ltr">Allows the VCF and associated Analysis object to be linked to an existing run accession, e.g.</p>
                    <p dir="ltr">##EVA_RunAccession=DRR000003</p>
                </div>
                <h3>Project</h3>
                <div>
                    <p dir="ltr">A Project object is required to contain the Sample and Analysis objects that pertain to the submitted VCF files. The tags for Project are:</p>
                    <p dir="ltr">##EVA_ProjectName--REQUIRED</p>
                    <p dir="ltr">##EVA_ProjectDescription--REQUIRED</p>
                    <p dir="ltr">##EVA_Material--REQUIRED</p>
                    <p dir="ltr">##EVA_Selection--REQUIRED</p>
                    <p dir="ltr">##EVA_Publication--OPTIONAL</p>
                    <p dir="ltr">##EVA_RelatedProject--OPTIONAL</p>
                    <p dir="ltr">##EVA_ProjectLink--OPTIONAL</p>
                    <p dir="ltr">##EVA_ProjectName</p>
                    <p dir="ltr">This is a unique name for the Project,e.g.</p>
                    <p dir="ltr">##EVA_ProjectName="1000 Genomes Phase 1 Variation Calls"</p>
                    <p dir="ltr">##EVA_ProjectDescription</p>
                    <p dir="ltr">This is a short description describing the project,e.g.</p>
                    <p dir="ltr">##EVA_ProjectDescription="Final release set of the Phase 1 1000 genomes variation data."</p>
                    <p dir="ltr">##EVA_Material</p>
                    <p dir="ltr">The material used, one of:</p>
                    <p dir="ltr">DNA, genomic RNA, transcribed RNA, unknown, other. &nbsp;e.g.</p>
                    <p dir="ltr">##EVA_Material=DNA</p>
                    <p dir="ltr">##EVA_Selection</p>
                    <p dir="ltr">One of the following:</p>
                    <p dir="ltr">genome, partial genome, exome, epigenomics, PCR, clone ends, random, CDNA, tag, unknown, other. e.g.</p>
                    <p dir="ltr">##EVA_Selection=genome</p>
                    <p dir="ltr">##EVA_Publication</p>
                    <p dir="ltr">Any associated publications with the Project, e.g.</p>
                    <p dir="ltr">##EVA_Publication=&lt;DB=PUBMED,ID=23128226,LABEL="Main 1000 Genomes Publication"&gt;</p>
                    <p dir="ltr">DB and ID are required, LABEL is optional.</p>
                    <p dir="ltr">##EVA_RelatedProject</p>
                    <p dir="ltr">Projects can have parent, peer, or child projects. This tag allows one to specify associated projects, e.g.</p>
                    <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=PARENT,ID=28889&gt;</p>
                    <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=PEER,ID=28890&gt;</p>
                    <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=CHILD,ID=28891&gt;</p>
                    <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=CHILD,ID=28892&gt;</p>
                    <p dir="ltr">The RELATIVE and ID sub-tags are both required.</p>
                    <p dir="ltr">##EVA_ProjectLink</p>
                    <p dir="ltr">This tag allows the project to be linked to an external resource, e.g.</p>
                    <p dir="ltr">##EVA_ProjectLink=&lt;URL=http://www.1000genomes.org,LABEL="1000 Genomes website"&gt;</p>
                    <p dir="ltr">##EVA_ProjectLink=&lt;DB=DGVa,ID=estd199&gt;</p>
                    <p dir="ltr">DB and ID can be used for INSDC databases.</p>
                    <p dir="ltr">If DB is present, ID is required. If DB is absent, URL is required and ID must not be included. LABEL is OPTIONAL.</p>
                </div>
                <br /><h4>Below is a valid set of tags for a fictional submission:</h4>
                <div>
                    <p dir="ltr">##fileformat=VCFv4.1</p>
                    <p dir="ltr">##EVA_ProjectName="1000 Genomes Phase 1 Variation Calls"</p>
                    <p dir="ltr">##EVA_ProjectTitle="Test 1"</p>
                    <p dir="ltr">##EVA_ProjectDescription="Test 2"</p>
                    <p dir="ltr">##EVA_Publication=&lt;DB=PUBMED,ID=23128226,LABEL="Main 1000 Genomes Publication"&gt;</p>
                    <p dir="ltr">##EVA_Publication=&lt;DB=ePUB,ID=23128226,LABEL="Main 1000 Genomes Publication"&gt;</p>
                    <p dir="ltr">##EVA_Scope=multi-isolate</p>
                    <p dir="ltr">##EVA_Material=DNA</p>
                    <p dir="ltr">##EVA_Selection=other</p>
                    <p dir="ltr">##EVA_TaxID=9606</p>
                    <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=PARENT,ID=28889&gt;</p>
                    <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=PEER,ID=28890&gt;</p>
                    <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=CHILD,ID=28891&gt;</p>
                    <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=CHILD,ID=28892&gt; ##EVA_ProjectLink=&lt;URL=http://www.1000genomes.org/phase1-analysis-results-directory,LABEL="FTP directory of submitted VCF files"&gt;</p>
                    <p dir="ltr">##EVA_ProjectLink=&lt;URL=http://www.1000genomes.org,LABEL="1000 Genomes website"&gt;</p>
                    <p dir="ltr">##EVA_AnalysisTitle="1000 Genomes Phase 1 Variation Calls VCF Parse TEST"</p>
                    <p dir="ltr">##EVA_FileDescription="Final release set from derived from data from 7 different centers. See PUBMED:23128226"</p>
                    <p dir="ltr">##EVA_SampleMappingFile=&lt;Name=test_file1.txt,MD5=123456789123456789123456789000&gt; ##EVA_ChecksumFile=ch1_test_md5.txt ##EVA_PipelineDescription="This is a testing piepline!"</p>
                    <p dir="ltr">##EVA_CallingAlgorithm="GATKv2.3, java -jar GenomeAnalysisTK.jar -T HaplotypeCaller -R reference/human_g1k_v37.fasta -I HG00096.bam -I HG00097.bam --dbsnp dbSNP.vcf -stand_call_conf [50.0] -stand_emit_conf 10.0 -o output.raw.snps.indels.vcf" ##EVA_Platform="Illumina GA11"</p>
                    <p dir="ltr">##EVA_ExperimentType="Is this a CV"</p>
                    <p dir="ltr">##EVA_RunAccession=&lt;ID=SRA00001,NAME=ARunName&gt;</p>
                    <p dir="ltr">##EVA_AnalysisAccession=&lt;ID=ENAA00001,NAME=AAnalysisName&gt;</p>
                    <p dir="ltr">##EVA_Sample=&lt;ID=HG00096,ACCESSION=ERSAM1&gt;</p>
                    <p dir="ltr">##EVA_Sample=&lt;ID=HG00097,GENDER=Male,PHENOTYPE=Ill,SAMPLETYPE=DNA,SUBJECT=Someone,DISEASE_SITE=Pancreas,TAXID=9606,TITLE="1000 Genomes Sample Number 97",DESCRIPTION="This is a description of HG00097"&gt;</p>
                    <p dir="ltr">##EVA_SampleLink=&lt;ID=HG00097,DB=CORIELL,DB_ID=HG00097,LABEL="This is a label"&gt;</p>
                    <p dir="ltr">&nbsp;</p>
                </div>
                <div>
                    <p dir="ltr">These guidlines are also avaialble in <a href="files/EVA_VCF_1.0.pdf" target="_blank">PDF format.</a> For any additional support, please contact us at <a href="mailto:eva-helpdesk@ebi.ac.uk">eva-helpdesk@ebi.ac.uk</a>.</p>
                </div>
            </div>
            <div id="api" class="hide-div">
                <h2>API</h2>
                <div>
                    <h3>The general structure of a EVA RESTful web service URL is:</h3><p><a href="#">http://www.ebi.ac.uk/eva/webservices/rest/{version}/{category}/IDs/{resource}?{filters}</a></p>
                    <h3>Where:</h3>
                    <p>
                       * <span class="api-span">version</span>: indicates the version of the API, this defines the available filters and JSON schema to be returned. Currently there is only version 'v1'.
                       <br/> * <span class="api-span">category</span>: this defines what objects we want to query. Currently there are five different categories: variants, segments, genes, files and studies.
                       <br/> * <span class="api-span">resource</span>: specifies the resource to be returned, therefore the JSON data model.
                       <br/> * <span class="api-span">filters</span>: each specific endpoint allows different filters.
                    </p>
                    <div>
                        <p>RESTfull web services have been implemented using GET protocol since only queries are allowed so far. Several IDs can be concatenated using comma as separator.<br/>For more detailed information about the API and filters you can visit the <a href="https://github.com/ebivariation/eva/wiki" target="_blank">project wiki</a> and <a href="http://wwwdev.ebi.ac.uk/eva/webservices/api/" target="_blank">Swagger documentation</a>.</p>
                    </div>
                    <div>
                        <h3>Some example of queries include:</h3>
                        <p>* To fetch all the variants in a segment region:
                        <br/> <a href="http://www.ebi.ac.uk/eva/webservices/rest/v1/segments/11:128446-128446/variants?species=hsapiens_grch37" target="_blank">http://www.ebi.ac.uk/eva/webservices/rest/v1/segments/11:128446-128446/variants?species=hsapiens_grch37</a></p>

                        <p><br/> * To fetch all the info of a variant:
                        <br/> <a href="http://www.ebi.ac.uk/eva/webservices/rest/v1/variants/rs666/info?species=hsapiens_grch37" target="_blank">http://www.ebi.ac.uk/eva/webservices/rest/v1/variants/rs666/info?species=hsapiens_grch37</a></p>

                        <p><br/> * To fetch all the Short Genetics Variations studies:
                        <br/> <a href="http://www.ebi.ac.uk/eva/webservices/rest/v1/meta/studies/all" target="_blank">http://www.ebi.ac.uk/eva/webservices/rest/v1/meta/studies/all</a></p>

                        <p><br/> * To fetch all the Structural Variations studies:
                        <br/> <a href="http://www.ebi.ac.uk/eva/webservices/rest/v1/meta/studies/all?structural=true" target="_blank">http://www.ebi.ac.uk/eva/webservices/rest/v1/meta/studies/all?structural=true</a></p>

                        <p><br/> * To fetch all  info of a study:
                        <br/> <a href="http://www.ebi.ac.uk/eva/webservices/rest/v1/studies/PRJEB4019/summary" target="_blank">http://www.ebi.ac.uk/eva/webservices/rest/v1/studies/PRJEB4019/summary</a></p>

                        <p><br/> * To fetch all file information of a study:
                        <br/> <a href="http://www.ebi.ac.uk/eva/webservices/rest/v1/studies/PRJEB4019/files?species=hsapiens_grch37" target="_blank">http://www.ebi.ac.uk/eva/webservices/rest/v1/studies/PRJEB4019/files?species=hsapiens_grch37</a></p>
                    </div>
                </div>
            </div>
            <div id="help-section" class="hide-div">
                <h2>FAQ</h2>
                <div class="panel-group" id="accordion">
                    <div class="panel panel-default">
                        <div class="panel-heading">
                            <h3 class="panel-title study-browser-panel">
                                <a data-toggle="collapse" data-parent="#accordion" href="#collapseOne"> What is the European Variation Archive (EVA)? <i class="indicator glyphicon glyphicon-chevron-up  pull-right"></i></a>
                            </h3>
                        </div>
                        <div id="collapseOne" class="panel-collapse collapse in">
                            <div class="panel-body">
                                <p>The European Variation Archive (EVA) is EMBL-EBI’s open-access genetic variation archive. The EVA accepts submission of all types of genetic variants, ranging from single nucleotide polymorphisms to large structural variants, observed in germline or somatic sources, from any eukaryotic organism. The EVA permits access to these data at two distinct levels:</p>
                                <p> i) The raw variant data as was submitted to the EVA, via the <a href="http://www.ebi.ac.uk/eva/?Study%20Browser&browserType=sgv" target="_blank">EVA Study Browser</a></p>
                                <p> ii) The normalised and processed variant data, via the <a href="http://www.ebi.ac.uk/eva/?Variant%20Browser" target="_blank">EVA Variant Browser</a> and <a href="http://www.ebi.ac.uk/eva/?API" target="_blank">EVA API</a></p>
                            </div>
                        </div>
                    </div>
                    <div class="panel panel-default">
                        <div class="panel-heading">
                            <h3 class="panel-title study-browser-panel">
                                <a data-toggle="collapse" data-parent="#accordion" href="#collapseTwo">Why submit data to the EVA?  <i class="indicator glyphicon glyphicon-chevron-down pull-right"></i></a>
                            </h3>
                        </div>
                        <div id="collapseTwo" class="panel-collapse collapse">
                            <div class="panel-body">
                                <p>The EVA provides to the community a completely free, secure and permanent solution to data sharing. Each project, VCF file and sample that is submitted to the EVA is assigned a unique identifier that is accessible in perpetuity and is therefore able to be referenced in publication, for example.
                                    The <a href="mailto:eva-helpdesk@ebi.ac.uk" target="_top">EVA helpdesk</a> provides support to submitters, and users, to ensure accurate represention and proper integration of the submitted data with other EMBL-EBI resources such as the <a href="https://www.ebi.ac.uk/ega/" target="_blank">EGA</a> and <a href="http://www.ensembl.org/index.html" target="_blank">Ensembl</a>. A final advantage of submitting to the EVA is that variants are brokered to the National Centre for Biotechnology Information (<a href="http://www.ncbi.nlm.nih.gov/" target="_blank">NCBI</a>) on the submitter’s behalf, negating the requirement for independent submissions.</p>
                            </div>
                        </div>
                    </div>
                    <div class="panel panel-default">
                        <div class="panel-heading">
                            <h3 class="panel-title study-browser-panel">
                                <a data-toggle="collapse" data-parent="#accordion" href="#collapseThree"> Is my data suitable for submission to the EVA?  <i class="indicator glyphicon glyphicon-chevron-down pull-right"></i></a>
                            </h3>
                        </div>
                        <div id="collapseThree" class="panel-collapse collapse">
                            <div class="panel-body">
                                <p>The most important consideration is that all data archived at the EVA is open access. As such, there are no restrictions as to who can access the data or how such data is reused. It is the submitter’s responsibility to ensure that the data archived at the EVA complies to this open access policy.</p>
                                <p>
                                    Genetic variants submitted to the EVA must be described in the <a href="http://samtools.github.io/hts-specs/" target="_blank">Variant Call Format (VCF)</a>.
                                    The EVA has developed a custom <a href="https://github.com/ebivariation/vcf-validator" target="_blank">VCF file validator</a> and accepts submission of only VCF files that pass <a href="https://docs.google.com/document/d/1Wcr2fjzwQ7NmxvqZEbiT7GVck8dLyWG2KvzmOxXAVfs/edit" target="_blank">these validation steps</a>.
                                    Furthermore, VCF files submitted to the EVA should provide either genotypes from the individual samples analysed, or aggregated sample summary information, such as allele frequencies.
                                </p>
                                <p>Finally, each submission to the EVA is accompanied by a completed metadata <a href="http://www.ebi.ac.uk/eva/?Templates" target="_blank">template </a>. This metadata template captures the study description, details of the sample(s) analysed and the experimental methodology. Submission of as much metadata as possible is strongly encouraged as this information is extremely useful for downstream analysis and is directly related to the frequency at which datasets archived at EVA are reused.</p>
                            </div>
                        </div>
                    </div>
                    <div class="panel panel-default">
                        <div class="panel-heading">
                            <h3 class="panel-title study-browser-panel">
                                <a data-toggle="collapse" data-parent="#accordion" href="#collapseFour">What happens to my data once submitted?  <i class="indicator glyphicon glyphicon-chevron-down  pull-right"></i></a>
                            </h3>
                        </div>
                        <div id="collapseFour" class="panel-collapse collapse">
                            <div class="panel-body">
                                <p> All  VCF files and novel samples that are submitted to EVA are permanently and securely archived at the <a href="http://www.ebi.ac.uk/ena" target="_blank">European Nucleotide Archive </a> and <a href="http://www.ebi.ac.uk/biosamples/" target="_blank">BioSamples </a>, respectively. The EVA provides access to all submitted data via the <a href="http://www.ebi.ac.uk/eva/?Study%20Browser&browserType=sgv" target="_blank">EVA Study Browser</a>.</p>
                                <p> Variants within VCF files submitted to EVA are normalised, annotated and used for statistical calculations (via methodologies described below) and these EVA processed data are available via the <a href="http://www.ebi.ac.uk/eva/?Variant%20Browser" target="_blank">EVA Variant Browser</a> and <a href="http://www.ebi.ac.uk/eva/?API" target="_blank">EVA API</a>.
                                <p> Finally, the EVA brokers all submitted data to the NCBI: <a href="http://www.ncbi.nlm.nih.gov/projects/SNP/" target="_blank">dbSNP</a>, for short variants (<50bp); <a href="http://www.ncbi.nlm.nih.gov/dbvar" target="_blank">dbVar</a> for large variants (>50bp) or <a href="http://www.ncbi.nlm.nih.gov/clinvar/" target="_blank">ClinVar</a>, for variants associated with both a phenotype and clinical significance.</p>
                            </div>
                        </div>
                    </div>
                    <div class="panel panel-default">
                        <div class="panel-heading">
                            <h3 class="panel-title study-browser-panel">
                                <a data-toggle="collapse" data-parent="#accordion" href="#collapseFive">What are the EVA normalisation and variant processing steps?  <i class="indicator glyphicon glyphicon-chevron-down pull-right"></i></a>
                            </h3>
                        </div>
                        <div id="collapseFive" class="panel-collapse collapse">
                            <div class="panel-body">
                                <div>
                                    <h5>Normalisation</h5>
                                    <p>
                                        Variants submitted to the EVA have been determined by a number of different algorithms and software packages. As a result, the VCF files generated by these differing methodologies describe variants in a number of different ways. The primary processing step of the EVA is to normalise variant representation following two basic rules:
                                    <ol>
                                        <li>Each variant is shifted to be left-aligned</li>
                                        <li>The Start and End positions represent exactly the range where the variation occurs (which could, in the case of insertions, result in the reference allele being recorded as ‘empty’)</li>
                                    </ol>
                                    </p>
                                    <p>Examples of our variant normalisation process can be seen <a href="https://docs.google.com/presentation/d/1WqSiT5AEEQF9jdIewdYIp-I0G5ozkFP3IikfCJZO1dc/edit#slide=id.ge1548f905_0_592" target="_blank">here</a></p>

                                </div>
                                <div>
                                    <h5>Annotation</h5>
                                    <p>
                                        Once variants have been normalised, the EVA uses the <a href="http://www.ensembl.org/info/docs/tools/vep/" target="_blank">Variant Effect Predictor (VEP) </a>  of Ensembl to annotate variant consequences.
                                        The variant consequences are described using Sequence Ontology terms and both the VEP version and Ensembl gene build used are described via the “i” help bubbles on the <a href="http://www.ebi.ac.uk/eva/?Variant%20Browser" target="_blank">EVA Variant Browser</a>.
                                        <br /><br /> N.B. Variants that have been mapped to a reference genome sequence that is not supported by Ensembl are not annotated.
                                    </p>
                                </div>
                                <div>
                                    <h5>Statistical calculations</h5>
                                    <p>The EVA adopts the classical definition of allele frequency (AF): ‘a measure of the relative frequency of an allele at a genetic locus in a given population’.
                                        The AF value(s) stored by the EVA for each variant is (are) study specific - i.e. the same variant reported in two distinct studies shall be given two allele frequencies, one for each study. There are two methodologies by which the EVA is able to determine allele frequency values, dependent on the datatype of the study in question:</p>
                                    <p><span style="font-size:14px;font-weight:bold;">Variants associated with genotypes:</span></p>
                                    <p>
                                        For variants associated with genotypes, the EVA determines the AF values via the calculation:
                                        <br /><br />AF = (number of alternate allele observations (AC)) / (number of observations (AN))
                                        <br /><br />The result of this calculation allows the EVA to also store the minor allele frequency (MAF) for each variant (defined as the minimum of the reference or alternative allele frequency) and the MAF allele (the allele associated with the MAF).
                                    </p>
                                    <p><span style="font-size:14px;font-weight:bold;">Variants not associated with genotypes:</span></p>
                                    <p>
                                        For variants that are not associated with genotypes, the EVA is dependent on the AF value(s) estimated from the primary data and provided in the submitted VCF files(s).
                                        AF values that are specifically provided in the submitted aggregated VCF file(s) are directly stored. In cases where no AF is provided then the EVA uses the AC and AN values in the submitted aggregated VCF file(s) to calculate AF value(s) via the calculation:
                                        <br /><br />AF = AC / AN
                                    </p>
                                    <p><span style="font-size:14px;font-weight:bold;">Population / sample cohort allele frequency values:</span></p>
                                    <p>
                                        The EVA accepts submission of <a href="http://pngu.mgh.harvard.edu/~purcell/plink/data.shtml#ped" target="_blank">pedigree </a> files, or structured samples (using “derived_from” and/or “subject” layers), to define populations and cohorts within studies.
                                        In cases where such information is associated with variants that have genotypes then the EVA calculates intra-study population/cohort specific AF values via the method described above, with the caveat that the (total number of populations/cohorts):(total number of samples) ratio must be less than 1:10.
                                        For studies that do not contain genotypes but instead provide intra-study population/cohort AF values in the submitted aggregated VCF file(s), or AC and AN values, then these are directly stored, or calculated by the EVA using the method described above, again with the caveat that a ratio of 1:10 (total number of populations/cohorts):(total number of samples) must not be exceeded.

                                        <br /><br />*NB: there are a low number of variants for which the EVA is unable to determine any allele frequency value(s) as the submitted VCF file(s) contain neither genotypes nor AF or AC and AN values. The EVA discourages submission of variants that cannot be associated with an AF.
                                    </p>
                                </div>
                            </div>
                        </div>
                    </div>
                    <div class="panel panel-default">
                        <div class="panel-heading">
                            <h3 class="panel-title study-browser-panel">
                                <a data-toggle="collapse" data-parent="#accordion" href="#collapseSix"> How can I consume variant data from the  EVA? <i class="indicator glyphicon glyphicon-chevron-down pull-right"></i></a>
                            </h3>
                        </div>
                        <div id="collapseSix" class="panel-collapse collapse">
                            <div class="panel-body">
                                <p> All EVA files are hosted on our <a href="ftp://ftp.ebi.ac.uk/pub/databases/eva/">FTP site</a> and are freely downloadable. Each EVA project FTP site allows users to download the original files as they were submitted to the EVA and/or the EVA ‘normalised’ files. Please note that due to processing time required not all EVA projects shall have ‘normalised’ files just yet.</p>
                                <p> Additionally users can download query results from the <a href="https://www.ebi.ac.uk/eva/?Variant%20Browser&species=hsapiens_grch37&selectFilter=region&studies=PRJEB4019%2CPRJEB6930%2CPRJEB5439%2CPRJEB5829%2CPRJEB8652%2CPRJEB8650%2CPRJEB8639%2CPRJEB6042%2CPRJX00001%2CPRJEB8705%2CPRJEB8661%2CPRJEB7895%2CPRJEB7217%2CPRJEB7218%2CPRJEB6041&region=1%3A3000000-3100000">EVA variant browser</a> directly and all of our data is available programmatically via our <a href="https://www.ebi.ac.uk/eva/?API">API</a>.</p>
                            </div>
                        </div>
                    </div>
                    <div class="panel panel-default">
                        <div class="panel-heading">
                            <h3 class="panel-title study-browser-panel">
                                <a data-toggle="collapse" data-parent="#accordion" href="#collapseSeven">What data is shown in the EVA Clinical Browser? <i class="indicator glyphicon glyphicon-chevron-down pull-right"></i></a>
                            </h3>
                        </div>
                        <div id="collapseSeven" class="panel-collapse collapse">
                            <div class="panel-body">
                               <p>
                                   The EVA Clinical Browser displays variant data imported from the NCBI resource <a href="http://www.ncbi.nlm.nih.gov/clinvar/" target="_blank">ClinVar</a>, where each variant is associated with both a phenotype and a clinical significance assigned using the guidlines from the <a href="http://www.ncbi.nlm.nih.gov/pubmed/18414213" target="_blank">American College of Medical Genetics and Genomics</a>.
                                   Importantly, the variants shown via the EVA Clinical Browser have been annotated against the human GRCh37 genome using both the VEP and GENCODE Basic gene build of Ensembl version 78.
                               </p>
                            </div>
                        </div>
                    </div>
                    <div class="panel panel-default">
                        <div class="panel-heading">
                            <h3 class="panel-title study-browser-panel">
                                <a data-toggle="collapse" data-parent="#accordion" href="#collapseEight">With whom does the EVA collaborate?  <i class="indicator glyphicon glyphicon-chevron-down pull-right"></i></a>
                            </h3>
                        </div>
                        <div id="collapseEight" class="panel-collapse collapse">
                            <div class="panel-body">
                                <div><img src="img/EVA_Related_Projects.png"><br/><br/><br/></div>
                                <h5> The EVA & GEUVADIS European Exome Variant Server </h5>
                                <div>
                                  <p>
                                      The ​​EVA & GE​UVADIS European Exome Variant Server ​(<a href="http://www.ncbi.nlm.nih.gov/pubmed/24037378" target="_blank">GEEVS;</a>) work in collaboration to ​coordinate​​ common ​data formats ​for ​data​ exchange​​.
                                      As part of this collaboration, we fully endorse the variant calling protocol detailed on the <a href="http://geevs.crg.eu/home" target="_blank">GEEVS</a> website as adherence to this protocol for variant calling permits direct comparison and/or aggregation of results from different datasets.
                                  </p>
                                </div>
                                <h5>The EVA & Príncipe Felipe Research Cent​er​</h5>
                                <p>Some of the technical and analytical features of the EVA were developed in collaboration with the department of Computational Genomics led by Joaquin Dopazo at the Principe Felipe Research Centre <a href="http://bioinfo.cipf.es/" target="_blank">Computational Genomics Department (CIPF)</a>.</p>
                                <h4 style="color:#996a44;">Past Collaborations</h4>
                                <h5>EVA & The Genomics and Bioinformatics Platform of Andalusia​</h5>
                                <div><p>Early development of the EVA was carried out in collaboration with the Bioinformatics Department at the <a href="http://www.gbpa.es" target="_blank">Genomics and Bioinformatics Platform of Andalusia</a>.</p></div>
                            </div>
                        </div>
                    </div>
                    <div class="panel panel-default">
                        <div class="panel-heading">
                            <h3 class="panel-title study-browser-panel">
                                <a data-toggle="collapse" data-parent="#accordion" href="#collapseNine">Which browsers does the EVA website support? <i class="indicator glyphicon glyphicon-chevron-down pull-right"></i></a>
                            </h3>
                        </div>
                        <div id="collapseNine" class="panel-collapse collapse">
                            <div class="panel-body">
                                <p>
                                    The EVA website employs HTML5 technologies and standards. Chrome (version 18 or higher), Firefox (version 12 or higher), Safari (version 6 or higher) , Opera (version 12 or higher) and Internet Explorer (version 10 or higher) are fully supported, however older versions of these browsers may give rise to errors.<br />
                                    The EVA website also supports mobile versions of web browsers, with limited functionality. Please report all errors to the <a href="mailto:eva-helpdesk@ebi.ac.uk" target="_top">EVA helpdesk</a>.
                                </p>
                            </div>
                        </div>
                    </div>
                </div>
            </div>

            <div id="studyView" class="hide-div"></div>
            <div id="variantView" class="hide-div"></div>
            <div id="geneView" class="hide-div"></div>
            <div id="clinical" class="hide-div"></div>
        </div>
        <footer>

            <div id="global-footer" class="grid_24">

                <nav id="global-nav-expanded">

                    <div class="grid_4 alpha">
                        <h3 class="embl-ebi"><a href="//www.ebi.ac.uk/" title="EMBL-EBI">EMBL-EBI</a></h3>
                    </div>

                    <div class="grid_4">
                        <h3 class="services"><a href="//www.ebi.ac.uk/services">Services</a></h3>
                    </div>

                    <div class="grid_4">
                        <h3 class="research"><a href="//www.ebi.ac.uk/research">Research</a></h3>
                    </div>

                    <div class="grid_4">
                        <h3 class="training"><a href="//www.ebi.ac.uk/training">Training</a></h3>
                    </div>

                    <div class="grid_4">
                        <h3 class="industry"><a href="//www.ebi.ac.uk/industry">Industry</a></h3>
                    </div>

                    <div class="grid_4 omega">
                        <h3 class="about"><a href="//www.ebi.ac.uk/about">About us</a></h3>
                    </div>

                </nav>

                <section id="ebi-footer-meta">
                    <p class="address">EMBL-EBI, Wellcome Trust Genome Campus, Hinxton, Cambridgeshire, CB10 1SD, UK &nbsp; &nbsp; +44 (0)1223 49 44 44</p>
                    <p class="legal">Copyright &copy; EMBL-EBI 2013 | EBI is an outstation of the <a href="http://www.embl.org">European Molecular Biology Laboratory</a> | <a href="/about/privacy">Privacy</a> | <a href="/about/cookies">Cookies</a> | <a href="/about/terms-of-use">Terms of use</a></p>
                </section>

            </div>

        </footer>
    </div> <!--! end of #wrapper -->

    <!-- JavaScript at the bottom for fast page loading -->
    <!-- build:script browser-detect -->
    <script type="text/javascript" src="js/browser-detect.js"></script>
    <!-- /build -->
    <!-- build:script vendor -->
    <script type="text/javascript" src="../vendor/ext-6.0.1/js/ext-all.js"></script>
    <script type="text/javascript" src="../bower_components/underscore/underscore.js"></script>
    <script type="text/javascript" src="../bower_components/backbone/backbone.js"></script>
    <script type="text/javascript" src="../bower_components/jquery/dist/jquery.js"></script>
    <!-- These dependencies must go AFTER jquery -->
    <!--<script type="text/javascript" src="../lib/jsorolla/vendor/bootstrap-scoped-dist/js/bootstrap.min.js"></script>-->
    <script type="text/javascript" src="../bower_components/highcharts-release/highcharts.src.js"></script>
    <script type="text/javascript" src="../bower_components/bootstrap/dist/js/bootstrap.js"></script>
    <!--<script type="text/javascript" src="../lib/jsorolla/vendor/jquery.cookie.js"></script>-->
    <!--<script type="text/javascript" src="../lib/jsorolla/vendor/jquery.sha1.js"></script>-->
    <!--<script type="text/javascript" src="../lib/jsorolla/vendor/purl.min.js"></script>-->
    <!--<script type="text/javascript" src="../lib/jsorolla/vendor/jquery.qtip.min.js"></script>-->
    <script type="text/javascript" src="../bower_components/linqjs/linq.min.js"></script>
    <script type="text/javascript" src="../bower_components/autotrack/autotrack.js"></script>
    <!--<script type="text/javascript" src="../bower_components/vkbeautify/vkbeautify.js"></script>-->
    <!-- /build -->
    <!-- build:script lib -->
    <script type="text/javascript" src="../lib/jsorolla/build/1.1.9/genome-viewer/genome-viewer.js"></script>
    <!--<script type="text/javascript" src="../lib/jsorolla/build/1.1.9/genome-viewer/gv-config.js"></script>-->
    <!-- /build -->
    <!-- build:script eva-js -->
    <script type="text/javascript" src="js/eva-manager.js"></script>
    <script type="text/javascript" src="js/eva-config.js"></script>
    <script type="text/javascript" src="js/eva-menu.js"></script>
    <script type="text/javascript" src="js/eva-adapter.js"></script>
    <script type="text/javascript" src="js/variant-widget/eva-variant-widget-panel.js"></script>
    <script type="text/javascript" src="js/study-browser/eva-study-browser-widget-panel.js"></script>
    <script type="text/javascript" src="js/study-browser/eva-study-browser-filter-form-panel.js"></script>
    <script type="text/javascript" src="js/study-browser/eva-study-browser-type-filter-form-panel.js"></script>
    <script type="text/javascript" src="js/study-browser/eva-study-browser-text-search-form-panel.js"></script>
    <script type="text/javascript" src="js/study-browser/eva-study-browser-grid.js"></script>
    <script type="text/javascript" src="js/variant-widget/eva-genome-viewer-panel.js"></script>
    <script type="text/javascript" src="js/beacon-form/eva-beacon-panel.js"></script>
    <script type="text/javascript" src="js/variant-widget/eva-variant-widget.js"></script>
    <script type="text/javascript" src="js/variant-widget/eva-variant-browser-grid.js"></script>
    <!--<script type="text/javascript" src="js/variant-widget/eva-variant-rawdata-panel.js"></script>-->
    <script type="text/javascript" src="js/variant-widget/eva-variant-stats-panel.js"></script>
    <script type="text/javascript" src="js/variant-widget/eva-variant-genotype-grid-panel.js"></script>
    <script type="text/javascript" src="js/variant-widget/eva-variant-population-stats-panel.js"></script>
    <!--<script type="text/javascript" src="js/variant-widget/eva-variant-transcript-grid.js"></script>-->
    <script type="text/javascript" src="js/views/eva-study-view.js"></script>
    <script type="text/javascript" src="js/views/eva-variant-view.js"></script>
    <script type="text/javascript" src="js/views/eva-gene-view.js"></script>
    <script type="text/javascript" src="js/views/eva-iobio-view.js"></script>
    <script type="text/javascript" src="js/statistics/eva-statistics.js"></script>
    <script type="text/javascript" src="js/statistics/dgva-statistics.js"></script>
    <script type="text/javascript" src="js/beacon-form/eva-beacon-form.js"></script>
    <script type="text/javascript" src="js/beacon-form/eva-variant-search-form.js"></script>
    <script type="text/javascript" src="js/eva-submission-form.js"></script>
    <script type="text/javascript" src="js/variant-widget/filters/eva-form-panel.js"></script>
    <script type="text/javascript" src="js/variant-widget/filters/eva-species-filter-form-panel.js"></script>
    <script type="text/javascript" src="js/variant-widget/filters/eva-study-filter-form-panel.js"></script>
    <script type="text/javascript" src="js/variant-widget/filters/eva-position-filter-form-panel.js"></script>
    <script type="text/javascript" src="js/variant-widget/filters/eva-population-frequency-filter-form-panel.js"></script>
    <script type="text/javascript" src="js/variant-widget/filters/eva-protein-substitutions-score-filter-form-panel.js"></script>
    <script type="text/javascript" src="js/variant-widget/filters/eva-consequence-type-filter-form-panel.js"></script>
    <script type="text/javascript" src="js/variant-widget/filters/eva-conservation-score-filter-form-panel.js"></script>
    <script type="text/javascript" src="js/clinvar/eva-clinical-widget-panel.js"></script>
    <script type="text/javascript" src="js/clinvar/eva-clinvar-widget.js"></script>
    <script type="text/javascript" src="js/clinvar/eva-clinvar-browser-grid.js"></script>
    <script type="text/javascript" src="js/clinvar/filters/eva-clinvar-position-filter-form-panel.js"></script>
    <script type="text/javascript" src="js/clinvar/filters/eva-clinvar-filter-form-panel.js"></script>
    <script type="text/javascript" src="js/clinvar/filters/eva-clinvar-trait-filter-form-panel.js"></script>
    <script type="text/javascript" src="js/clinvar/eva-clinvar-summary-panel.js"></script>
    <script type="text/javascript" src="js/clinvar/eva-clinvar-assertion-panel.js"></script>
    <script type="text/javascript" src="js/clinvar/eva-clinvar-links-panel.js"></script>
    <script type="text/javascript" src="js/clinvar/eva-clinvar-annotation-panel.js"></script>
    <script type="text/javascript" src="js/eva.js"></script>
    <!-- /build -->
    <!-- build:script ebi-web-guidelines-js -->
    <script defer="defer" src="js/ebi-web-guidelines/modernizr.custom.49274.js"></script>
    <script defer="defer" src="js/ebi-web-guidelines/cookiebanner.js"></script>
    <script defer="defer" src="js/ebi-web-guidelines/foot.js"></script>
    <!-- /build -->
    <!-- end scripts-->

    <script type="text/javascript">
        var eva;
        loadGoogleAnalytics();
        var start = function () {
            // Bring in the import content.
//            var link = document.querySelector('link[href="content.html"]');
              var homeDiv = document.querySelector('#home');
              var submitDiv = document.querySelector('#submit');
//              var aboutDiv = document.querySelector('#about-us');
              var feedbackDiv = document.querySelector('#feedback');
              var templatesDiv = document.querySelector('#submissionTemplates');
              var vcfDiv = document.querySelector('#vcf-format');
              var clinicalDiv = document.querySelector('#clinical');
              var studyViewDiv = document.querySelector('#studyView');
              var variantViewDiv = document.querySelector('#variantView');
              var geneViewDiv = document.querySelector('#geneView');
              var beaconDiv =  document.querySelector('#beacon');
              var apiDiv =  document.querySelector('#api');
              var faqDiv = document.querySelector('#help-section');

            var tab = getUrlParameters("");
            var tabId =  decodeURI(tab.id);
            if(tabId == 'eva-study'){
                var evastudyView = new EvaStudyView({
                    target:'studyView',
                    type: 'eva',
                    projectId:tab.value
                });

            }
            if(tabId == 'dgva-study'){
                var dgvastudyView = new EvaStudyView({
                    target:'studyView',
                    type: 'dgva',
                    projectId:tab.value
                });
            }
            if(tabId == 'variant'){

                var varianttSpecies = $.urlParam('species');
                var evaVarinatView = new EvaVariantView({
                    target:'variantView',
                    position:tab.value,
                    species:varianttSpecies
                });
            }

            if(tabId == 'gene'){
                var varianttSpecies = $.urlParam('species');
                var evaGeneView = new EvaGeneView({
                    target:'geneView',
                    geneId:tab.value,
                    species:varianttSpecies
                });
            }

            if(tabId == 'submission-start'){
                var evaSubmission= new EvaSubmissionForm({
                    target:'submissionForm'
                });
            }
            if(tabId == 'eva-iobio'){
                var evaIobioView = new EvaIobioView({
                    target:'evaIobioView',
                    url: $.urlParam('url')
                });
            }

            Ext.getBody().removeCls('x-body');
            var ulLocalNav = document.querySelector('.eva-primary-menu');

            eva = new Eva({
                target: 'content',
                targetMenu: ulLocalNav,
                width: $('#content').width(),
                homeDiv:homeDiv,
                submitDiv:submitDiv,
//                aboutDiv:aboutDiv,
                feedbackDiv:feedbackDiv,
                templatesDiv:templatesDiv,
                vcfDiv:vcfDiv,
                studyView:studyViewDiv,
                variantView:variantViewDiv,
                geneView:geneViewDiv,
                beacon:beaconDiv,
                clinicalDiv:clinicalDiv,
                apiDiv:apiDiv,
                faqDiv:faqDiv
            });
            eva.draw();

            if(tab.id){
                eva.select(tabId);
            }else{
                eva.select('Home');
            }

            window.onpopstate = function(event) {
                if(event && event.state) {
                    location.reload();
                }
            }
        };


        function getUrlParameters(staticURL) {
            var data = [];

            var currLocation = (staticURL.length) ? staticURL : window.location.search;
            if (currLocation.indexOf("?") >= 0) {
                parArr = currLocation.split("?")[1].split("&");
                if (parArr) {
                    for (var i = 0; i < parArr.length; i++) {
                        parr = parArr[i].split("=");
                        data.id = parr[0];
                        data.value = parr[1];
                        return data;
                    }
                }
            } else {
                return false
            }
        }

        $.urlParam = function(name){

            var results = new RegExp('[?&]' + name + '=([^&#]*)').exec(window.location.href);
            if (results==null){
                return null;
            }
            else{
                return results[1] || 0;
            }

        }

        function loadGoogleAnalytics() {
            var trackingID;
            if (document.location.hostname == 'www.ebi.ac.uk') {
                trackingID = 'UA-82688737-1';
            } else if (document.location.hostname == 'wwwdev.ebi.ac.uk') {
                trackingID = 'UA-82375858-1';
            } else {
                trackingID = 'UA-82386453-1';
            }

            (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
                        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
                    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
            })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
            ga('create', trackingID, 'auto');
            ga('require', 'socialWidgetTracker');

        }

        function toggleChevron(e) {
            $(e.target)
                    .prev('.panel-heading')
                    .find("i.indicator")
                    .toggleClass('glyphicon-chevron-down glyphicon-chevron-up');
        }
        $('#accordion').on('hidden.bs.collapse', toggleChevron);
        $('#accordion').on('shown.bs.collapse', toggleChevron);

        //Start point
        window.addEventListener('load', function (e) {
            // all imports loaded
            Ext.QuickTips.init();
            Ext.onReady(start);
        });



    </script>

</body>
</html>
