<div>
    <h3>EVA VCF Format</h3>
    <p> EVA only accepts VCF files that conform to <a href="http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41" target="_blank">VCF version 4.1 and above.</a></p>
    <h3>VCF Specification</h3>
    <div>
        <p>All VCF files must be valid, i.e. they must pass validation using a VCF validator, such as vcftools. In addition, all VCF files must include a reference tag (##reference). Any tag that contains white space (e.g. spaces) must be enclosed in double quotes, e.g.</p>
        <p>##EVA_PipelineDescription="This is a testing pipepline!" OK</p>
        <p>##EVA_PipelineDescription=This is a testing pipepline! NOT OK</p>
    </div>
    <h3>EVA Specific headers</h3>
    <div>
        <p>An EVA submission automatically creates the required objects in ENA. A VCF file belongs to an Analysis, which in turn belongs to a Project. Currently each VCF file can only belong to one Analysis, and each Analysis can only contain one VCF file. The Analysis allows the VCF file to be linked to the Samples or SampleGroup (a group of samples).</p>
    </div>
    <h3>Samples</h3>
    <div>
        <p>If a Sample or SampleGroup already exists in ENA, then the VCF file can reference this object. This can be done in 3 ways:</p>
        <p>1) Using the ENA Sample accession in the header of the VCF file.</p>
        <p>2) Use the ##EVA_Sample information line to to link the sample ID in the VCF file to the Sample accession using the following specification:</p>
        <p>##EVA_Sample=&lt;ID=sample_id,ACCESSION=ena_accession&gt;</p>
        <p>3) Supply a sample mapping file(s) along with the VCF files, which links the VCF sample IDs to the Sample accession. The filename for the sample mapping file is given using the ##EVA_SampleMappingFile information line in the VCF file. Multiple files can be used, but each sample ID from the VCF must only occur once in all of the specified sample mapping files. The format to specify sample mapping files is:</p>
        <p>##EVA_SampleMappingFile=&lt;NAME=filename,MD5=md5sum&gt;</p>
        <p dir="ltr">Samples can be linked to external resources, e.g. CORIELL. This is done using the ##EVA_SampleLink tag. e.g.</p>
        <p dir="ltr">##EVA_SampleLink=&lt;ID=HG00097,DB=CORIELL,DB_ID=NA18501,LABEL="Link to Coriell"&gt;</p>
        <p dir="ltr">ID and DB are required (if the VCF ID is the same as the external resource ID). If the external resource ID is different to the VCF ID, this can be specified using the DB_ID sub-tag. LABEL is optional.</p>
        <p>The ##EVA_Scope tag defines the study type. This tag is optional, and if used must correspond to the number of samples listed in the VCF file. If there is 1 sample, the ##EVA_Scope tag is 'single-isolate', otherwise it is 'multi-isolate'.</p>
    </div>
    <h3>Analysis</h3>
    <div>
        <p dir="ltr">Each VCF file must belong to an Analysis object which describes the analysis used to produce the VCF file. Currently and Analysis object can only contain 1 VCF file.</p>
        <p dir="ltr">The following tags are used to describe the analysis:</p>
        <p dir="ltr">##reference--REQUIRED</p>
        <p dir="ltr">##EVA_AnalysisTitle--REQUIRED</p>
        <p dir="ltr">##EVA_PipelineDescription--REQUIRED</p>
        <p dir="ltr">##EVA_AnalysisCenter--OPTIONAL</p>
        <p dir="ltr">##EVA_AnalysisDate--OPTIONAL</p>
        <p dir="ltr">##EVA_CallingAlgorithm--OPTIONAL</p>
        <p dir="ltr">##EVA_Platform--OPTIONAL</p>
        <p dir="ltr">##EVA_ExperimentType--REQUIRED</p>
        <p dir="ltr">##EVA_RunAccession--OPTIONAL</p>
        <p dir="ltr">##EVA_AnalysisAccession--OPTIONAL</p>
        <p dir="ltr">##reference</p>
        <p dir="ltr">This is a 'highly recommended' tag according to VCFv4.1 specifications, and EVA require it for all VCF files. It can refer to either:</p>
        <p dir="ltr">1) An INSDC assembly name or accession,e.g.</p>
        <p dir="ltr">##reference=GRCh37</p>
        <p dir="ltr">or</p>
        <p dir="ltr">##reference=GCA_000001405.1</p>
        <p dir="ltr">2) or a location of the reference file, e.g.</p>
        <p dir="ltr">##reference=file:///seq/references/1000GenomesPilot-NCBI36.fasta</p>
        <p dir="ltr">or</p>
        <p dir="ltr">##reference=<a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.2bit">http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.2bit</a></p>
        <p dir="ltr">EVA also requires INSDC accessions or names for each of the chromosomes (or sequences) referenced in column 1 of the VCF. This should be of the form:</p>
        <p dir="ltr">##contig=&lt;ID=1,accession=<a href="http://www.ncbi.nlm.nih.gov/nuccore/CM000663.1">CM000663.1</a>,length=249,250,621,species=”Homo sapiens”,taxonomy=9606&gt;</p>
        <p dir="ltr">or</p>
        <p dir="ltr">##contig=&lt;ID=1,URL=http://www.ncbi.nlm.nih.gov/nuccore/224384768?report=fasta&gt;</p>
        <p dir="ltr">##EVA_AnalysisTitle</p>
        <p dir="ltr">This is the title displayed for the accession Analysis object, e.g.</p>
        <p dir="ltr">##EVA_AnalysisTitle="1000 Genomes Phase 1 Variation Calls Analysis"</p>
        <p dir="ltr">##EVA_PipelineDescription</p>
        <p dir="ltr">Description of analysis pipeline. Free text field, e.g.</p>
        <p dir="ltr">##EVA_PipelineDescription="Final release set from derived from data from 7 different centers. See PUBMED:23128226"</p>
        <p dir="ltr">##EVA_AnalysisCenter</p>
        <p dir="ltr">The center that performed the analysis</p>
        <p dir="ltr">##EVA_AnalysisCenter="European Bioinformatics Institute"</p>
        <p dir="ltr">##EVA_AnalysisDate</p>
        <p dir="ltr">The date the analysis was performed (in DD/MM/YYYY-hh:min:secs-offest, where offset and hh:min:secs are optional, hr:min:secs defaults to 12:00:00, and offset indicates the hour offset from UTC defaulting to 0), e.g. for an analysis performed at 9.30 am, 6 hours before UTC on 30th May 2011:</p>
        <p dir="ltr">##EVA_AnalysisDate="30/05/2011-09:30-06"</p>
        <p dir="ltr">##EVA_CallingAlgorithm</p>
        <p dir="ltr">Details of the calling algorithm, version, and parameters, e.g.</p>
        <p dir="ltr">##EVA_CallingAlgorithm="GATKv2.3, java -jar GenomeAnalysisTK.jar -T HaplotypeCaller -R reference/human_g1k_v37.fasta -I HG00096.bam -I HG00097.bam --dbsnp dbSNP.vcf -stand_call_conf &nbsp;-stand_emit_conf 10.0 -o output.raw.snps.indels.vcf"</p>
        <p dir="ltr">##EVA_Platform</p>
        <p dir="ltr">The platform used, e.g.</p>
        <p dir="ltr">##EVA_Platform="Illumina Genome Analyser IIe"</p>
        <p dir="ltr">##EVA_ExperimentType</p>
        <p dir="ltr">Tag to indicate the type of experiment, choose from:</p>
        <p dir="ltr">Whole genome, Exome, Array</p><p dir="ltr">e.g.</p>
        <p dir="ltr">##EVA_ExperimentType="Whole Genome"</p>
        <p dir="ltr">##EVA_RunAccession</p>
        <p dir="ltr">Allows the VCF and associated Analysis object to be linked to an existing run accession, e.g.</p>
        <p dir="ltr">##EVA_RunAccession=DRR000003</p>
    </div>
    <h3>Project</h3>
    <div>
        <p dir="ltr">A Project object is required to contain the Sample and Analysis objects that pertain to the submitted VCF files. The tags for Project are:</p>
        <p dir="ltr">##EVA_ProjectName--REQUIRED</p>
        <p dir="ltr">##EVA_ProjectDescription--REQUIRED</p>
        <p dir="ltr">##EVA_Material--REQUIRED</p>
        <p dir="ltr">##EVA_Selection--REQUIRED</p>
        <p dir="ltr">##EVA_Publication--OPTIONAL</p>
        <p dir="ltr">##EVA_RelatedProject--OPTIONAL</p>
        <p dir="ltr">##EVA_ProjectLink--OPTIONAL</p>
        <p dir="ltr">##EVA_ProjectName</p>
        <p dir="ltr">This is a unique name for the Project,e.g.</p>
        <p dir="ltr">##EVA_ProjectName="1000 Genomes Phase 1 Variation Calls"</p>
        <p dir="ltr">##EVA_ProjectDescription</p>
        <p dir="ltr">This is a short description describing the project,e.g.</p>
        <p dir="ltr">##EVA_ProjectDescription="Final release set of the Phase 1 1000 genomes variation data."</p>
        <p dir="ltr">##EVA_Material</p>
        <p dir="ltr">The material used, one of:</p>
        <p dir="ltr">DNA, genomic RNA, transcribed RNA, unknown, other. &nbsp;e.g.</p>
        <p dir="ltr">##EVA_Material=DNA</p>
        <p dir="ltr">##EVA_Selection</p>
        <p dir="ltr">One of the following:</p>
        <p dir="ltr">genome, partial genome, exome, epigenomics, PCR, clone ends, random, CDNA, tag, unknown, other. e.g.</p>
        <p dir="ltr">##EVA_Selection=genome</p>
        <p dir="ltr">##EVA_Publication</p>
        <p dir="ltr">Any associated publications with the Project, e.g.</p>
        <p dir="ltr">##EVA_Publication=&lt;DB=PUBMED,ID=23128226,LABEL="Main 1000 Genomes Publication"&gt;</p>
        <p dir="ltr">DB and ID are required, LABEL is optional.</p>
        <p dir="ltr">##EVA_RelatedProject</p>
        <p dir="ltr">Projects can have parent, peer, or child projects. This tag allows one to specify associated projects, e.g.</p>
        <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=PARENT,ID=28889&gt;</p>
        <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=PEER,ID=28890&gt;</p>
        <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=CHILD,ID=28891&gt;</p>
        <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=CHILD,ID=28892&gt;</p>
        <p dir="ltr">The RELATIVE and ID sub-tags are both required.</p>
        <p dir="ltr">##EVA_ProjectLink</p>
        <p dir="ltr">This tag allows the project to be linked to an external resource, e.g.</p>
        <p dir="ltr">##EVA_ProjectLink=&lt;URL=http://www.1000genomes.org,LABEL="1000 Genomes website"&gt;</p>
        <p dir="ltr">##EVA_ProjectLink=&lt;DB=DGVa,ID=estd199&gt;</p>
        <p dir="ltr">DB and ID can be used for INSDC databases.</p>
        <p dir="ltr">If DB is present, ID is required. If DB is absent, URL is required and ID must not be included. LABEL is OPTIONAL.</p>
    </div>
    <br /><h4>Below is a valid set of tags for a fictional submission:</h4>
    <div>
        <p dir="ltr">##fileformat=VCFv4.1</p>
        <p dir="ltr">##EVA_ProjectName="1000 Genomes Phase 1 Variation Calls"</p>
        <p dir="ltr">##EVA_ProjectTitle="Test 1"</p>
        <p dir="ltr">##EVA_ProjectDescription="Test 2"</p>
        <p dir="ltr">##EVA_Publication=&lt;DB=PUBMED,ID=23128226,LABEL="Main 1000 Genomes Publication"&gt;</p>
        <p dir="ltr">##EVA_Publication=&lt;DB=ePUB,ID=23128226,LABEL="Main 1000 Genomes Publication"&gt;</p>
        <p dir="ltr">##EVA_Scope=multi-isolate</p>
        <p dir="ltr">##EVA_Material=DNA</p>
        <p dir="ltr">##EVA_Selection=other</p>
        <p dir="ltr">##EVA_TaxID=9606</p>
        <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=PARENT,ID=28889&gt;</p>
        <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=PEER,ID=28890&gt;</p>
        <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=CHILD,ID=28891&gt;</p>
        <p dir="ltr">##EVA_RelatedProject=&lt;RELATIVE=CHILD,ID=28892&gt; ##EVA_ProjectLink=&lt;URL=http://www.1000genomes.org/phase1-analysis-results-directory,LABEL="FTP directory of submitted VCF files"&gt;</p>
        <p dir="ltr">##EVA_ProjectLink=&lt;URL=http://www.1000genomes.org,LABEL="1000 Genomes website"&gt;</p>
        <p dir="ltr">##EVA_AnalysisTitle="1000 Genomes Phase 1 Variation Calls VCF Parse TEST"</p>
        <p dir="ltr">##EVA_FileDescription="Final release set from derived from data from 7 different centers. See PUBMED:23128226"</p>
        <p dir="ltr">##EVA_SampleMappingFile=&lt;Name=test_file1.txt,MD5=123456789123456789123456789000&gt; ##EVA_ChecksumFile=ch1_test_md5.txt ##EVA_PipelineDescription="This is a testing piepline!"</p>
        <p dir="ltr">##EVA_CallingAlgorithm="GATKv2.3, java -jar GenomeAnalysisTK.jar -T HaplotypeCaller -R reference/human_g1k_v37.fasta -I HG00096.bam -I HG00097.bam --dbsnp dbSNP.vcf -stand_call_conf [50.0] -stand_emit_conf 10.0 -o output.raw.snps.indels.vcf" ##EVA_Platform="Illumina GA11"</p>
        <p dir="ltr">##EVA_ExperimentType="Is this a CV"</p>
        <p dir="ltr">##EVA_RunAccession=&lt;ID=SRA00001,NAME=ARunName&gt;</p>
        <p dir="ltr">##EVA_AnalysisAccession=&lt;ID=ENAA00001,NAME=AAnalysisName&gt;</p>
        <p dir="ltr">##EVA_Sample=&lt;ID=HG00096,ACCESSION=ERSAM1&gt;</p>
        <p dir="ltr">##EVA_Sample=&lt;ID=HG00097,GENDER=Male,PHENOTYPE=Ill,SAMPLETYPE=DNA,SUBJECT=Someone,DISEASE_SITE=Pancreas,TAXID=9606,TITLE="1000 Genomes Sample Number 97",DESCRIPTION="This is a description of HG00097"&gt;</p>
        <p dir="ltr">##EVA_SampleLink=&lt;ID=HG00097,DB=CORIELL,DB_ID=HG00097,LABEL="This is a label"&gt;</p>
        <p dir="ltr">&nbsp;</p>
    </div>
    <div>
        <p dir="ltr">These guidlines are also avaialble in <a href="files/EVA_VCF_1.0.pdf" target="_blank">PDF format.</a> For any additional support, please contact us at eva-helpdesk@ebi.ac.uk.</p>
    </div>
</div>